<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Glenn Hickey" />
  <meta name="author" content="David Heller" />
  <meta name="author" content="Jean Monlong" />
  <meta name="author" content="Benedict Paten" />
  <meta name="dcterms.date" content="2019-02-04" />
  <meta name="keywords" content="structural variation, pangenome, variant genotyping" />
  <title>Genotyping structural variation in variation graphs with the vg toolkit</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="github-pandoc.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!-- Insert Analytics Script Below -->
  <script>
  </script>
  <!-- End Analytics Script -->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Genotyping structural variation in variation graphs with the vg toolkit</h1>
</header>
<p><small><em> This manuscript (<a href="https://jmonlong.github.io/manu-vgsv/v/6ac424a22768cb973d7e9abb1890a5edb956e84c/">permalink</a>) was automatically generated from <a href="https://github.com/jmonlong/manu-vgsv/tree/6ac424a22768cb973d7e9abb1890a5edb956e84c">jmonlong/manu-vgsv@6ac424a</a> on February 4, 2019. </em></small></p>
<h2 id="authors">Authors</h2>
<p><a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Glenn Hickey<sup>1,☯</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> David Heller<sup>1,☯</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Jean Monlong<sup>1,☯</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Benedict Paten<sup>1,†</sup></p>
<p><sup>☯</sup> — These authors contributed equally to this work</p>
<p><sup>†</sup> — To whom correspondence should be addressed: bpaten@ucsc.edu <small></p>
<ol type="1">
<li>UC Santa Cruz Genomics Institute, University of California, Santa Cruz, California, USA</li>
</ol>
<p></small></p>
<h2 id="abstract" class="page_break_before">Abstract</h2>
<h2 id="introduction" class="page_break_before">Introduction</h2>
<p>Structural variation (SV) represents genomic mutation involving 50 bp or more and can take several forms, such as for example deletions, insertions, inversions, or translocations. Although whole-genome sequencing (WGS) made it possible to assess virtually any type of structural variation, many challenges remain. In particular, SV-supporting reads are difficult to map to reference genomes. Multi-mapping, caused by widespread repeated sequences in the genome, is another issue because it often resembles SV-supporting signal. As a result, many SV detection algorithms have been developed and multiple methods must usually be combined to minimize false positives. Several large-scale projects used this ensemble approach, cataloging tens of thousands of SV in humans<span class="citation" data-cites="qA6dWFP py6BC5kj">[<a href="#ref-qA6dWFP">1</a>,<a href="#ref-py6BC5kj">2</a>]</span>. SV detection from short-read sequencing remains laborious and of lower accuracy, explaining why these variants and their impact have been under-studied as compared to single-nucleotide variants (SNVs) and small insertions/deletions (indels).</p>
<p>Over the last few years, exciting developments in sequencing technologies and library preparation made it possible to produce long reads or retrieve long-range information over kilobases of sequence. These approaches are maturing to the point were it is feasible to analyze the human genome. This multi-kbp information is particularly useful for SV detection and de novo assembly. In the last few years, several studies using long-read or linked-read sequencing have produced large catalogs of structural variation, the majority of which were novel and sequence-resolved<span class="citation" data-cites="z91V6jjU rs7e40wC PRx3qEIm 121OWcTA4 3NNFS6U2">[<a href="#ref-z91V6jjU">3</a>,<a href="#ref-rs7e40wC">4</a>,<a href="#ref-PRx3qEIm">5</a>,<a href="#ref-121OWcTA4">6</a>,<a href="#ref-3NNFS6U2">7</a>]</span>. These technologies are also enabling high-quality de novo genome assemblies to be produced<span class="citation" data-cites="z91V6jjU 6KbgcueR">[<a href="#ref-z91V6jjU">3</a>,<a href="#ref-6KbgcueR">8</a>]</span>, as well as large blocks of haplotype-resolved sequences<span class="citation" data-cites="Pu6SY37C">[<a href="#ref-Pu6SY37C">9</a>]</span>. These technological advances promise to expand the amount of known genomic variation in humans in the near future.</p>
<p>In parallel, the reference genome is evolving from a linear reference to a graph-based reference that contains known genomic variation<span class="citation" data-cites="Qa8mx6Ll 10jxt15v0 11Jy8B61m">[<a href="#ref-Qa8mx6Ll">10</a>,<a href="#ref-10jxt15v0">11</a>,<a href="#ref-11Jy8B61m">12</a>]</span>. By having variants in the graph, mapping rates are increased and variants are more uniformly covered, including indels and variants in complex regions<span class="citation" data-cites="10jxt15v0">[<a href="#ref-10jxt15v0">11</a>]</span>. Both the mapping and variant calling become variant-aware and benefit in term of accuracy and sensitivity. In addition, different variant types are called simultaneously by a unified framework. Graphs have also been used locally, i.e. to call variants at the region level. GraphTyper<span class="citation" data-cites="ohTIiqfV">[<a href="#ref-ohTIiqfV">13</a>]</span> and BayesTyper<span class="citation" data-cites="14Uxmwbxm">[<a href="#ref-14Uxmwbxm">14</a>]</span> both construct variation graphs of small regions and use them for variant genotyping. Here again, the graph-approach showed clear advantages over standard approaches that use the linear reference. Other SV genotyping approaches compare read mapping in the reference genome and a sequence modified with the SV. For example SMRT-SV was designed to genotype SVs identified on PacBio reads<span class="citation" data-cites="rs7e40wC">[<a href="#ref-rs7e40wC">4</a>]</span>, SVTyper uses paired-end mapping and split-read mapping information<span class="citation" data-cites="AltPnocw">[<a href="#ref-AltPnocw">15</a>]</span>, and Delly provides a genotyping feature in addition to its discovery mode<span class="citation" data-cites="nLvQCjXU">[<a href="#ref-nLvQCjXU">16</a>]</span>.</p>
<h2 id="results">Results</h2>
<h3 id="structural-variation-in-vg">Structural variation in vg</h3>
<p>In addition to SNV and short indels, vg can handle large deletions and insertions (and inversion?) (Figure <a href="#fig:cartoon">1a</a>). As a proof-of-concept we simulated genomes and SVs of varying sizes. Some errors were added at the breakpoints to investigate their effect on genotyping. In all simulations, vg performed better than SVtyper<span class="citation" data-cites="AltPnocw">[<a href="#ref-AltPnocw">15</a>]</span> and Delly<span class="citation" data-cites="nLvQCjXU">[<a href="#ref-nLvQCjXU">16</a>]</span> (Figure <a href="#fig:sim">1b</a>). The recall was particularly higher than other methods at low sequencing depth. vg was also more robust to errors around the breakpoints, performing almost as well as in the absence of errors.</p>
<figure>
<img src="images/VGSVcartoon.jpg" alt="Figure 1a: Large deletions and insertions in variation graphs" id="fig:cartoon" data-tag="1a" /><figcaption><span>Figure 1a:</span> <strong>Large deletions and insertions in variation graphs</strong></figcaption>
</figure>
<figure>
<img src="images/sv-vg-simulation-f1.png" alt="Figure 1b: Simulation experiment." id="fig:sim" data-tag="1b" /><figcaption><span>Figure 1b:</span> <strong>Simulation experiment</strong>.</figcaption>
</figure>
<h3 id="hgsvc">HGSVC</h3>
<p>Chaisson et al.<span class="citation" data-cites="vQTymKCj">[<a href="#ref-vQTymKCj">17</a>]</span> provide a high-quality SV catalog of three samples, obtained using a consensus from different sequencing, phasing and variant caling technologies.</p>
<h4 id="whole-genome-simulation">(Whole-genome) Simulation</h4>
<p>The phasing information in the HGSVC VCF was used to extract two haplotypes for sample HG00514, and 30X pairend-end reads were simulated using vg sim. The reads were used to call VCFs then compared back to the original HGSVC calls.</p>
<table>
<thead>
<tr class="header">
<th>Graph</th>
<th>type</th>
<th>TP</th>
<th>TP.baseline</th>
<th>FP</th>
<th>FN</th>
<th>precision</th>
<th>recall</th>
<th>F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>HGSVC-Construct</td>
<td>Total</td>
<td>24451</td>
<td>24089</td>
<td>3119</td>
<td>2617</td>
<td>0.8854</td>
<td>0.902</td>
<td>0.8936</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>14596</td>
<td>14264</td>
<td>775</td>
<td>1421</td>
<td>0.9485</td>
<td>0.9094</td>
<td>0.9285</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>9855</td>
<td>9825</td>
<td>2344</td>
<td>1196</td>
<td>0.8074</td>
<td>0.8915</td>
<td>0.8474</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>HGSVC-1KG-Construct</td>
<td>Total</td>
<td>24172</td>
<td>23815</td>
<td>3236</td>
<td>2891</td>
<td>0.8804</td>
<td>0.8917</td>
<td>0.886</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>14540</td>
<td>14111</td>
<td>836</td>
<td>1574</td>
<td>0.9441</td>
<td>0.8996</td>
<td>0.9213</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>9632</td>
<td>9704</td>
<td>2400</td>
<td>1317</td>
<td>0.8017</td>
<td>0.8805</td>
<td>0.8393</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>HGSVC-Bayestyper</td>
<td>Total</td>
<td>13895</td>
<td>14362</td>
<td>123</td>
<td>12344</td>
<td>0.9915</td>
<td>0.5378</td>
<td>0.6974</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>8473</td>
<td>8757</td>
<td>102</td>
<td>6928</td>
<td>0.9885</td>
<td>0.5583</td>
<td>0.7136</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>5422</td>
<td>5605</td>
<td>21</td>
<td>5416</td>
<td>0.9963</td>
<td>0.5086</td>
<td>0.6734</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>SVPOP-Construct</td>
<td>Total</td>
<td>10548</td>
<td>11559</td>
<td>5990</td>
<td>15147</td>
<td>0.6587</td>
<td>0.4328</td>
<td>0.5224</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>7733</td>
<td>8223</td>
<td>2266</td>
<td>7462</td>
<td>0.784</td>
<td>0.5243</td>
<td>0.6284</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>2815</td>
<td>3336</td>
<td>3724</td>
<td>7685</td>
<td>0.4725</td>
<td>0.3027</td>
<td>0.369</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>SVPOP-1KG-Construct</td>
<td>Total</td>
<td>10403</td>
<td>11369</td>
<td>6750</td>
<td>15337</td>
<td>0.6275</td>
<td>0.4257</td>
<td>0.5073</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>7497</td>
<td>7934</td>
<td>2198</td>
<td>7751</td>
<td>0.7831</td>
<td>0.5058</td>
<td>0.6146</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>2906</td>
<td>3435</td>
<td>4552</td>
<td>7586</td>
<td>0.4301</td>
<td>0.3117</td>
<td>0.3615</td>
</tr>
</tbody>
</table>
<p>When restricting the comparisons to regions not identified as tandem repeats or segmental duplications in the Genome Browser:</p>
<table>
<thead>
<tr class="header">
<th>Graph</th>
<th>type</th>
<th>TP</th>
<th>TP.baseline</th>
<th>FP</th>
<th>FN</th>
<th>precision</th>
<th>recall</th>
<th>F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>HGSVC-Construct</td>
<td>Total</td>
<td>5901</td>
<td>5822</td>
<td>452</td>
<td>253</td>
<td>0.928</td>
<td>0.9584</td>
<td>0.943</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>4026</td>
<td>3952</td>
<td>98</td>
<td>172</td>
<td>0.9758</td>
<td>0.9583</td>
<td>0.967</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>1875</td>
<td>1870</td>
<td>354</td>
<td>81</td>
<td>0.8408</td>
<td>0.9585</td>
<td>0.8958</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>HGSVC-1KG-Construct</td>
<td>Total</td>
<td>5880</td>
<td>5785</td>
<td>486</td>
<td>290</td>
<td>0.9225</td>
<td>0.9523</td>
<td>0.9372</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>4024</td>
<td>3922</td>
<td>123</td>
<td>202</td>
<td>0.9696</td>
<td>0.951</td>
<td>0.9602</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>1856</td>
<td>1863</td>
<td>363</td>
<td>88</td>
<td>0.8369</td>
<td>0.9549</td>
<td>0.892</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>HGSVC-Bayestyper</td>
<td>Total</td>
<td>3805</td>
<td>3863</td>
<td>8</td>
<td>2212</td>
<td>0.9979</td>
<td>0.6359</td>
<td>0.7768</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>2401</td>
<td>2430</td>
<td>6</td>
<td>1694</td>
<td>0.9975</td>
<td>0.5892</td>
<td>0.7408</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>1404</td>
<td>1433</td>
<td>2</td>
<td>518</td>
<td>0.9986</td>
<td>0.7345</td>
<td>0.846</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>SVPOP-Construct</td>
<td>Total</td>
<td>3565</td>
<td>3856</td>
<td>390</td>
<td>2219</td>
<td>0.9081</td>
<td>0.6347</td>
<td>0.7472</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>3091</td>
<td>3246</td>
<td>239</td>
<td>878</td>
<td>0.9314</td>
<td>0.7871</td>
<td>0.8532</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>474</td>
<td>610</td>
<td>151</td>
<td>1341</td>
<td>0.8016</td>
<td>0.3127</td>
<td>0.4499</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>SVPOP-1KG-Construct</td>
<td>Total</td>
<td>3574</td>
<td>3817</td>
<td>562</td>
<td>2258</td>
<td>0.8717</td>
<td>0.6283</td>
<td>0.7303</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>3066</td>
<td>3180</td>
<td>253</td>
<td>944</td>
<td>0.9263</td>
<td>0.7711</td>
<td>0.8416</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>508</td>
<td>637</td>
<td>309</td>
<td>1314</td>
<td>0.6734</td>
<td>0.3265</td>
<td>0.4398</td>
</tr>
</tbody>
</table>
<h4 id="whole-genome-real-reads">(Whole-genome) Real reads</h4>
<table>
<thead>
<tr class="header">
<th>Graph</th>
<th>type</th>
<th>TP</th>
<th>TP.baseline</th>
<th>FP</th>
<th>FN</th>
<th>precision</th>
<th>recall</th>
<th>F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>HGSVC-Construct</td>
<td>Total</td>
<td>18436</td>
<td>18500</td>
<td>6575</td>
<td>8206</td>
<td>0.7378</td>
<td>0.6927</td>
<td>0.7145</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>10984</td>
<td>10600</td>
<td>3542</td>
<td>5085</td>
<td>0.7495</td>
<td>0.6758</td>
<td>0.7107</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>7452</td>
<td>7900</td>
<td>3033</td>
<td>3121</td>
<td>0.7226</td>
<td>0.7168</td>
<td>0.7197</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>HGSVC-1KG-Construct</td>
<td>Total</td>
<td>17802</td>
<td>17946</td>
<td>6221</td>
<td>8760</td>
<td>0.7426</td>
<td>0.672</td>
<td>0.7055</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>10647</td>
<td>10262</td>
<td>3304</td>
<td>5423</td>
<td>0.7564</td>
<td>0.6543</td>
<td>0.7017</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>7155</td>
<td>7684</td>
<td>2917</td>
<td>3337</td>
<td>0.7248</td>
<td>0.6972</td>
<td>0.7107</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>HGSVC-Bayestyper</td>
<td>Total</td>
<td>4342</td>
<td>4840</td>
<td>1048</td>
<td>21866</td>
<td>0.822</td>
<td>0.1812</td>
<td>0.2969</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>1786</td>
<td>1883</td>
<td>309</td>
<td>13802</td>
<td>0.859</td>
<td>0.1201</td>
<td>0.2107</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>2556</td>
<td>2957</td>
<td>739</td>
<td>8064</td>
<td>0.8001</td>
<td>0.2683</td>
<td>0.4018</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>SVPOP-Construct</td>
<td>Total</td>
<td>9091</td>
<td>9931</td>
<td>10235</td>
<td>16775</td>
<td>0.4925</td>
<td>0.3719</td>
<td>0.4238</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>6972</td>
<td>7420</td>
<td>6706</td>
<td>8265</td>
<td>0.5253</td>
<td>0.4731</td>
<td>0.4978</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>2119</td>
<td>2511</td>
<td>3529</td>
<td>8510</td>
<td>0.4157</td>
<td>0.2278</td>
<td>0.2943</td>
</tr>
</tbody>
</table>
<p>When restricting the comparisons to regions not identified as tandem repeats or segmental duplications in the Genome Browser:</p>
<table>
<thead>
<tr class="header">
<th>Graph</th>
<th>type</th>
<th>TP</th>
<th>TP.baseline</th>
<th>FP</th>
<th>FN</th>
<th>precision</th>
<th>recall</th>
<th>F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>HGSVC-Construct</td>
<td>Total</td>
<td>5197</td>
<td>5244</td>
<td>854</td>
<td>831</td>
<td>0.86</td>
<td>0.8632</td>
<td>0.8616</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>3708</td>
<td>3626</td>
<td>459</td>
<td>498</td>
<td>0.8876</td>
<td>0.8792</td>
<td>0.8834</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>1489</td>
<td>1618</td>
<td>395</td>
<td>333</td>
<td>0.8038</td>
<td>0.8293</td>
<td>0.8164</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>HGSVC-1KG-Construct</td>
<td>Total</td>
<td>5103</td>
<td>5155</td>
<td>865</td>
<td>920</td>
<td>0.8563</td>
<td>0.8486</td>
<td>0.8524</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>3642</td>
<td>3555</td>
<td>464</td>
<td>569</td>
<td>0.8845</td>
<td>0.862</td>
<td>0.8731</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>1461</td>
<td>1600</td>
<td>401</td>
<td>351</td>
<td>0.7996</td>
<td>0.8201</td>
<td>0.8097</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>HGSVC-Bayestyper</td>
<td>Total</td>
<td>1560</td>
<td>1731</td>
<td>274</td>
<td>4344</td>
<td>0.8633</td>
<td>0.2849</td>
<td>0.4284</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>883</td>
<td>901</td>
<td>69</td>
<td>3223</td>
<td>0.9289</td>
<td>0.2185</td>
<td>0.3538</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>677</td>
<td>830</td>
<td>205</td>
<td>1121</td>
<td>0.8019</td>
<td>0.4254</td>
<td>0.5559</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>SVPOP-Construct</td>
<td>Total</td>
<td>3251</td>
<td>3480</td>
<td>941</td>
<td>2595</td>
<td>0.7872</td>
<td>0.5728</td>
<td>0.6631</td>
</tr>
<tr class="even">
<td></td>
<td>INS</td>
<td>2859</td>
<td>3009</td>
<td>780</td>
<td>1115</td>
<td>0.7941</td>
<td>0.7296</td>
<td>0.7605</td>
</tr>
<tr class="odd">
<td></td>
<td>DEL</td>
<td>392</td>
<td>471</td>
<td>161</td>
<td>1480</td>
<td>0.7453</td>
<td>0.2414</td>
<td>0.3647</td>
</tr>
</tbody>
</table>
<h3 id="genotyping-sv-using-vg-and-de-novo-assemblies">Genotyping SV using vg and de novo assemblies</h3>
<p>We investigated whether genome graphs derived from genome-genome alignments yield advantages for SV genotyping. To this end, we analyzed public sequencing datasets for 12 yeast strains from two clades (S. cerevisiae and S. paradoxus) <span class="citation" data-cites="7f5OKa5O">[<a href="#ref-7f5OKa5O">18</a>]</span>. From these datasets, we generated two different types of genome graphs. The first graph type (in the following called <em>construct graph</em>) was created from a linear reference genome of the S.c. S288C strain and a set of SVs relative to this reference strain in VCF format. We compiled the SV set using the output of three methods for SV detection from genome assemblies: Assemblytics <span class="citation" data-cites="krO7WgVi">[<a href="#ref-krO7WgVi">19</a>]</span>, AsmVar <span class="citation" data-cites="oVaXIwl5">[<a href="#ref-oVaXIwl5">20</a>]</span> and paftools <span class="citation" data-cites="172cJaw4Q">[<a href="#ref-172cJaw4Q">21</a>]</span>. All three methods were run to detect SVs between the reference strain S.c. S288C and each of the other 11 strains. Merging the results from the three methods and the 11 strains provided us with a high-sensitivity set of SVs occuring in the two yeast clades. We used this set to construct the <em>construct graph</em>. The second graph (in the following called <em>cactus graph</em>) was derived from a multiple genome alignment of all 12 strains using our Cactus tool <span class="citation" data-cites="1FgS53pXi">[<a href="#ref-1FgS53pXi">22</a>]</span>. While the <em>construct graph</em> is still mainly linear and highly dependent on the reference genome, the cactus graph is completely unbiased in that regard.</p>
<figure>
<img src="images/mapping_comparison_all.png" alt="Figure 1: Mappability comparison. The fraction of reads mapped (with mapping quality &gt; 0) to the cactus graph (y-axis) and the construct graph (x-axis) are compared" id="fig:mapping-comp" style="width:80.0%" /><figcaption><span>Figure 1:</span> <strong>Mappability comparison.</strong> The fraction of reads mapped (with mapping quality &gt; 0) to the cactus graph (y-axis) and the construct graph (x-axis) are compared</figcaption>
</figure>
<p>In a first step, we tested our hypothesis that the <em>cactus graph</em> has higher mappability due to its better representation of sequence diversity among the yeast strains. When mapping short Illumina reads from the 12 strains to both graphs, we indeed observed a higher fraction of reads mapped to the <em>cactus graph</em> than to the <em>construct graph</em> (see Fig. <a href="#fig:mapping-comp">1</a>). Only for the reference strain S.c. S288C, both graphs exhibited similar mappability. This suggests that not the higher sequence content in the <em>cactus graph</em> alone (XX Mb compared to XX Mb in the <em>construct graph</em>) drives the improvement in mappability. Instead, our measurements suggest that genetic distance to the reference strain increases the advantage of the <em>cactus graph</em> over the <em>construct graph</em>. Consequently. the gap is largest for strains in the S. paradoxus clade and smaller for reads from strains in the S. cerevisiae clade.</p>
<figure>
<img src="images/david-yeast-jan29.png" alt="Figure 2: SV genotyping comparison. SV genotype recall from the cactus graph (y-axis) and construct graph (x-axis) are compared. Colors and shapes represent the 12 strains and two clades, respectively" id="fig:geno-comp" style="width:80.0%" /><figcaption><span>Figure 2:</span> <strong>SV genotyping comparison.</strong> SV genotype recall from the <em>cactus graph</em> (y-axis) and <em>construct graph</em> (x-axis) are compared. Colors and shapes represent the 12 strains and two clades, respectively</figcaption>
</figure>
<p>Next, we compared the SV genotype performance of both graphs. To facilitate a fair evaluation of genotype performance, we combined all SVs that were detected by at least two of the three SV callers (Assemblytics, AsmVar and paftools) into a truth set. This truth set is a subset of the SV set used for construction of the <em>construct graph</em> which is important because only variants already present in the graph can be genotyped.</p>
<p>Figure <a href="#fig:geno-comp">2</a> shows the results of our analysis. Depending on the clade, the <em>cactus graph</em> reaches either a substantially higher SV genotyping recall than the <em>construct graph</em> (S. paradoxus) or a substantially lower recall (S. cerevisiae).</p>
<h2 id="methods">Methods</h2>
<h2 id="discussion">Discussion</h2>
<h2 id="references" class="page_break_before">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs">
<div id="ref-qA6dWFP">
<p>1. <strong>An integrated map of structural variation in 2,504 human genomes</strong><br />
Peter H. SudmantTobias Rausch, Eugene J. Gardner, Robert E. Handsaker, Alexej Abyzov, John Huddleston, Yan Zhang, Kai Ye, Goo Jun, … Jan O. Korbel<br />
<em>Nature</em> (2015-10) <a href="https://doi.org/73c">https://doi.org/73c</a><br />
DOI: <a href="https://doi.org/10.1038/nature15394">10.1038/nature15394</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26432246">26432246</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4617611">PMC4617611</a></p>
</div>
<div id="ref-py6BC5kj">
<p>2. <strong>Whole-genome sequence variation, population structure and demographic history of the Dutch population</strong><br />
Laurent C FrancioliAndroniki Menelaou, Sara L Pulit, Freerk van Dijk, Pier Francesco Palamara, Clara C Elbers, Pieter BT Neerincx, Kai Ye, Victor Guryev, … Cisca Wijmenga<br />
<em>Nature Genetics</em> (2014-06-29) <a href="https://doi.org/f6bxm8">https://doi.org/f6bxm8</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3021">10.1038/ng.3021</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24974849">24974849</a></p>
</div>
<div id="ref-z91V6jjU">
<p>3. <strong>Resolving the complexity of the human genome using single-molecule sequencing</strong><br />
Mark J. P. Chaisson, John Huddleston, Megan Y. Dennis, Peter H. Sudmant, Maika Malig, Fereydoun Hormozdiari, Francesca Antonacci, Urvashi Surti, Richard Sandstrom, Matthew Boitano, … Evan E. Eichler<br />
<em>Nature</em> (2014-11-10) <a href="https://doi.org/w69">https://doi.org/w69</a><br />
DOI: <a href="https://doi.org/10.1038/nature13907">10.1038/nature13907</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25383537">25383537</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4317254">PMC4317254</a></p>
</div>
<div id="ref-rs7e40wC">
<p>4. <strong>Discovery and genotyping of structural variation from long-read haploid genome sequence data</strong><br />
John Huddleston, Mark J.P. Chaisson, Karyn Meltz Steinberg, Wes Warren, Kendra Hoekzema, David Gordon, Tina A. Graves-Lindsay, Katherine M. Munson, Zev N. Kronenberg, Laura Vives, … Evan E. Eichler<br />
<em>Genome Research</em> (2016-11-28) <a href="https://doi.org/f9x79h">https://doi.org/f9x79h</a><br />
DOI: <a href="https://doi.org/10.1101/gr.214007.116">10.1101/gr.214007.116</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27895111">27895111</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411763">PMC5411763</a></p>
</div>
<div id="ref-PRx3qEIm">
<p>5. <strong>Mapping and phasing of structural variation in patient genomes using nanopore sequencing</strong><br />
Mircea Cretu Stancu, Markus J. van Roosmalen, Ivo Renkens, Marleen M. Nieboer, Sjors Middelkamp, Joep de Ligt, Giulia Pregno, Daniela Giachino, Giorgia Mandrile, Jose Espejo Valle-Inclan, … Wigard P. Kloosterman<br />
<em>Nature Communications</em> (2017-11-06) <a href="https://doi.org/gftpt9">https://doi.org/gftpt9</a><br />
DOI: <a href="https://doi.org/10.1038/s41467-017-01343-4">10.1038/s41467-017-01343-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29109544">29109544</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5673902">PMC5673902</a></p>
</div>
<div id="ref-121OWcTA4">
<p>6. <strong>Genome-wide reconstruction of complex structural variants using read clouds</strong><br />
Noah Spies, Ziming Weng, Alex Bishara, Jennifer McDaniel, David Catoe, Justin M Zook, Marc Salit, Robert B West, Serafim Batzoglou, Arend Sidow<br />
<em>Nature Methods</em> (2017-07-17) <a href="https://doi.org/gbnhwk">https://doi.org/gbnhwk</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.4366">10.1038/nmeth.4366</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28714986">28714986</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5578891">PMC5578891</a></p>
</div>
<div id="ref-3NNFS6U2">
<p>7. <strong>Characterizing the Major Structural Variant Alleles of the Human Genome</strong><br />
Peter A. Audano, Arvis Sulovari, Tina A. Graves-Lindsay, Stuart Cantsilieris, Melanie Sorensen, AnneMarie E. Welch, Max L. Dougherty, Bradley J. Nelson, Ankeeta Shah, Susan K. Dutcher, … Evan E. Eichler<br />
<em>Cell</em> (2019-01) <a href="https://doi.org/gfthvz">https://doi.org/gfthvz</a><br />
DOI: <a href="https://doi.org/10.1016/j.cell.2018.12.019">10.1016/j.cell.2018.12.019</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30661756">30661756</a></p>
</div>
<div id="ref-6KbgcueR">
<p>8. <strong>Nanopore sequencing and assembly of a human genome with ultra-long reads</strong><br />
Miten Jain, Sergey Koren, Karen H Miga, Josh Quick, Arthur C Rand, Thomas A Sasani, John R Tyson, Andrew D Beggs, Alexander T Dilthey, Ian T Fiddes, … Matthew Loose<br />
<em>Nature Biotechnology</em> (2018-01-29) <a href="https://doi.org/gczffw">https://doi.org/gczffw</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.4060">10.1038/nbt.4060</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29431738">29431738</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5889714">PMC5889714</a></p>
</div>
<div id="ref-Pu6SY37C">
<p>9. <strong>Phased diploid genome assembly with single-molecule real-time sequencing</strong><br />
Chen-Shan Chin, Paul Peluso, Fritz J Sedlazeck, Maria Nattestad, Gregory T Concepcion, Alicia Clum, Christopher Dunn, Ronan O’Malley, Rosa Figueroa-Balderas, Abraham Morales-Cruz, … Michael C Schatz<br />
<em>Nature Methods</em> (2016-10-17) <a href="https://doi.org/f9fv4w">https://doi.org/f9fv4w</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.4035">10.1038/nmeth.4035</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27749838">27749838</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5503144">PMC5503144</a></p>
</div>
<div id="ref-Qa8mx6Ll">
<p>10. <strong>Genome graphs and the evolution of genome inference</strong><br />
Benedict Paten, Adam M. Novak, Jordan M. Eizenga, Erik Garrison<br />
<em>Genome Research</em> (2017-03-30) <a href="https://doi.org/f95nhd">https://doi.org/f95nhd</a><br />
DOI: <a href="https://doi.org/10.1101/gr.214155.116">10.1101/gr.214155.116</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28360232">28360232</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411762">PMC5411762</a></p>
</div>
<div id="ref-10jxt15v0">
<p>11. <strong>Variation graph toolkit improves read mapping by representing genetic variation in the reference</strong><br />
Erik Garrison, Jouni Sirén, Adam M Novak, Glenn Hickey, Jordan M Eizenga, Eric T Dawson, William Jones, Shilpa Garg, Charles Markello, Michael F Lin, … Richard Durbin<br />
<em>Nature Biotechnology</em> (2018-08-20) <a href="https://doi.org/gd2zqs">https://doi.org/gd2zqs</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.4227">10.1038/nbt.4227</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30125266">30125266</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6126949">PMC6126949</a></p>
</div>
<div id="ref-11Jy8B61m">
<p>12. <strong>Fast and accurate genomic analyses using genome graphs</strong><br />
Goran Rakocevic, Vladimir Semenyuk, Wan-Ping Lee, James Spencer, John Browning, Ivan J. Johnson, Vladan Arsenijevic, Jelena Nadj, Kaushik Ghose, Maria C. Suciu, … Deniz Kural<br />
<em>Nature Genetics</em> (2019-01-14) <a href="https://doi.org/gftd46">https://doi.org/gftd46</a><br />
DOI: <a href="https://doi.org/10.1038/s41588-018-0316-4">10.1038/s41588-018-0316-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30643257">30643257</a></p>
</div>
<div id="ref-ohTIiqfV">
<p>13. <strong>Graphtyper enables population-scale genotyping using pangenome graphs</strong><br />
Hannes P Eggertsson, Hakon Jonsson, Snaedis Kristmundsdottir, Eirikur Hjartarson, Birte Kehr, Gisli Masson, Florian Zink, Kristjan E Hjorleifsson, Aslaug Jonasdottir, Adalbjorg Jonasdottir, … Bjarni V Halldorsson<br />
<em>Nature Genetics</em> (2017-09-25) <a href="https://doi.org/gbx7v6">https://doi.org/gbx7v6</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3964">10.1038/ng.3964</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28945251">28945251</a></p>
</div>
<div id="ref-14Uxmwbxm">
<p>14. <strong>Accurate genotyping across variant classes and lengths using variant graphs</strong><br />
Jonas Andreas SibbesenLasse Maretty, Anders Krogh<br />
<em>Nature Genetics</em> (2018-06-18) <a href="https://doi.org/gdndnz">https://doi.org/gdndnz</a><br />
DOI: <a href="https://doi.org/10.1038/s41588-018-0145-5">10.1038/s41588-018-0145-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29915429">29915429</a></p>
</div>
<div id="ref-AltPnocw">
<p>15. <strong>SpeedSeq: ultra-fast personal genome analysis and interpretation</strong><br />
Colby Chiang, Ryan M Layer, Gregory G Faust, Michael R Lindberg, David B Rose, Erik P Garrison, Gabor T Marth, Aaron R Quinlan, Ira M Hall<br />
<em>Nature Methods</em> (2015-08-10) <a href="https://doi.org/gcpgfh">https://doi.org/gcpgfh</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.3505">10.1038/nmeth.3505</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26258291">26258291</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4589466">PMC4589466</a></p>
</div>
<div id="ref-nLvQCjXU">
<p>16. <strong>DELLY: structural variant discovery by integrated paired-end and split-read analysis</strong><br />
T. Rausch, T. Zichner, A. Schlattl, A. M. Stutz, V. Benes, J. O. Korbel<br />
<em>Bioinformatics</em> (2012-09-07) <a href="https://doi.org/f38r2c">https://doi.org/f38r2c</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/bts378">10.1093/bioinformatics/bts378</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22962449">22962449</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3436805">PMC3436805</a></p>
</div>
<div id="ref-vQTymKCj">
<p>17. <strong>Multi-platform discovery of haplotype-resolved structural variation in human genomes</strong><br />
Mark J.P. Chaisson, Ashley D. Sanders, Xuefang Zhao, Ankit Malhotra, David Porubsky, Tobias Rausch, Eugene J. Gardner, Oscar Rodriguez, Li Guo, Ryan L. Collins, … Charles Lee<br />
<em>Cold Spring Harbor Laboratory</em> (2017-09-23) <a href="https://doi.org/gftxhc">https://doi.org/gftxhc</a><br />
DOI: <a href="https://doi.org/10.1101/193144">10.1101/193144</a></p>
</div>
<div id="ref-7f5OKa5O">
<p>18. <strong>Contrasting evolutionary genome dynamics between domesticated and wild yeasts</strong><br />
Jia-Xing Yue, Jing Li, Louise Aigrain, Johan Hallin, Karl Persson, Karen Oliver, Anders Bergström, Paul Coupland, Jonas Warringer, Marco Cosentino Lagomarsino, … Gianni Liti<br />
<em>Nature Genetics</em> (2017-04-17) <a href="https://doi.org/f93kpp">https://doi.org/f93kpp</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3847">10.1038/ng.3847</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28416820">28416820</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5446901">PMC5446901</a></p>
</div>
<div id="ref-krO7WgVi">
<p>19. <strong>Assemblytics: a web analytics tool for the detection of variants from an assembly</strong><br />
Maria Nattestad, Michael C. Schatz<br />
<em>Bioinformatics</em> (2016-06-17) <a href="https://doi.org/f9c485">https://doi.org/f9c485</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/btw369">10.1093/bioinformatics/btw369</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27318204">27318204</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6191160">PMC6191160</a></p>
</div>
<div id="ref-oVaXIwl5">
<p>20. <strong>Discovery, genotyping and characterization of structural variation and novel sequence at single nucleotide resolution from de novo genome assemblies on a population scale</strong><br />
Siyang LiuShujia Huang, Junhua Rao, Weijian Ye, Anders Krogh, Jun Wang<br />
<em>GigaScience</em> (2015-12) <a href="https://doi.org/f75r4n">https://doi.org/f75r4n</a><br />
DOI: <a href="https://doi.org/10.1186/s13742-015-0103-4">10.1186/s13742-015-0103-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26705468">26705468</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4690232">PMC4690232</a></p>
</div>
<div id="ref-172cJaw4Q">
<p>21. <strong>Minimap2: pairwise alignment for nucleotide sequences</strong><br />
Heng Li<br />
<em>Bioinformatics</em> (2018-05-10) <a href="https://doi.org/gdhbqt">https://doi.org/gdhbqt</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/bty191">10.1093/bioinformatics/bty191</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29750242">29750242</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6137996">PMC6137996</a></p>
</div>
<div id="ref-1FgS53pXi">
<p>22. <strong>Cactus: Algorithms for genome multiple sequence alignment</strong><br />
B. Paten, D. Earl, N. Nguyen, M. Diekhans, D. Zerbino, D. Haussler<br />
<em>Genome Research</em> (2011-06-10) <a href="https://doi.org/bk4697">https://doi.org/bk4697</a><br />
DOI: <a href="https://doi.org/10.1101/gr.123356.111">10.1101/gr.123356.111</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21665927">21665927</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3166836">PMC3166836</a></p>
</div>
</div>
<script>
// AnchorJS minified version below.
// Source https://github.com/bryanbraun/anchorjs/blob/064abdd0987f305933ec4982af6d0c1cf2fd0814/anchor.js

/**
 * AnchorJS - v4.0.0 - 2017-06-02
 * https://github.com/bryanbraun/anchorjs
 * Copyright (c) 2017 Bryan Braun; Licensed MIT
 */
!function(A,e){"use strict";"function"==typeof define&&define.amd?define([],e):"object"==typeof module&&module.exports?module.exports=e():(A.AnchorJS=e(),A.anchors=new A.AnchorJS)}(this,function(){"use strict";function A(A){function e(A){A.icon=A.hasOwnProperty("icon")?A.icon:"",A.visible=A.hasOwnProperty("visible")?A.visible:"hover",A.placement=A.hasOwnProperty("placement")?A.placement:"right",A.class=A.hasOwnProperty("class")?A.class:"",A.truncate=A.hasOwnProperty("truncate")?Math.floor(A.truncate):64}function t(A){var e;if("string"==typeof A||A instanceof String)e=[].slice.call(document.querySelectorAll(A));else{if(!(Array.isArray(A)||A instanceof NodeList))throw new Error("The selector provided to AnchorJS was invalid.");e=[].slice.call(A)}return e}function n(){if(null===document.head.querySelector("style.anchorjs")){var A,e=document.createElement("style");e.className="anchorjs",e.appendChild(document.createTextNode("")),void 0===(A=document.head.querySelector('[rel="stylesheet"], style'))?document.head.appendChild(e):document.head.insertBefore(e,A),e.sheet.insertRule(" .anchorjs-link {   opacity: 0;   text-decoration: none;   -webkit-font-smoothing: antialiased;   -moz-osx-font-smoothing: grayscale; }",e.sheet.cssRules.length),e.sheet.insertRule(" *:hover > .anchorjs-link, .anchorjs-link:focus  {   opacity: 1; }",e.sheet.cssRules.length),e.sheet.insertRule(" [data-anchorjs-icon]::after {   content: attr(data-anchorjs-icon); }",e.sheet.cssRules.length),e.sheet.insertRule(' @font-face {   font-family: "anchorjs-icons";   src: url(data:n/a;base64,AAEAAAALAIAAAwAwT1MvMg8yG2cAAAE4AAAAYGNtYXDp3gC3AAABpAAAAExnYXNwAAAAEAAAA9wAAAAIZ2x5ZlQCcfwAAAH4AAABCGhlYWQHFvHyAAAAvAAAADZoaGVhBnACFwAAAPQAAAAkaG10eASAADEAAAGYAAAADGxvY2EACACEAAAB8AAAAAhtYXhwAAYAVwAAARgAAAAgbmFtZQGOH9cAAAMAAAAAunBvc3QAAwAAAAADvAAAACAAAQAAAAEAAHzE2p9fDzz1AAkEAAAAAADRecUWAAAAANQA6R8AAAAAAoACwAAAAAgAAgAAAAAAAAABAAADwP/AAAACgAAA/9MCrQABAAAAAAAAAAAAAAAAAAAAAwABAAAAAwBVAAIAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAMCQAGQAAUAAAKZAswAAACPApkCzAAAAesAMwEJAAAAAAAAAAAAAAAAAAAAARAAAAAAAAAAAAAAAAAAAAAAQAAg//0DwP/AAEADwABAAAAAAQAAAAAAAAAAAAAAIAAAAAAAAAIAAAACgAAxAAAAAwAAAAMAAAAcAAEAAwAAABwAAwABAAAAHAAEADAAAAAIAAgAAgAAACDpy//9//8AAAAg6cv//f///+EWNwADAAEAAAAAAAAAAAAAAAAACACEAAEAAAAAAAAAAAAAAAAxAAACAAQARAKAAsAAKwBUAAABIiYnJjQ3NzY2MzIWFxYUBwcGIicmNDc3NjQnJiYjIgYHBwYUFxYUBwYGIwciJicmNDc3NjIXFhQHBwYUFxYWMzI2Nzc2NCcmNDc2MhcWFAcHBgYjARQGDAUtLXoWOR8fORYtLTgKGwoKCjgaGg0gEhIgDXoaGgkJBQwHdR85Fi0tOAobCgoKOBoaDSASEiANehoaCQkKGwotLXoWOR8BMwUFLYEuehYXFxYugC44CQkKGwo4GkoaDQ0NDXoaShoKGwoFBe8XFi6ALjgJCQobCjgaShoNDQ0NehpKGgobCgoKLYEuehYXAAAADACWAAEAAAAAAAEACAAAAAEAAAAAAAIAAwAIAAEAAAAAAAMACAAAAAEAAAAAAAQACAAAAAEAAAAAAAUAAQALAAEAAAAAAAYACAAAAAMAAQQJAAEAEAAMAAMAAQQJAAIABgAcAAMAAQQJAAMAEAAMAAMAAQQJAAQAEAAMAAMAAQQJAAUAAgAiAAMAAQQJAAYAEAAMYW5jaG9yanM0MDBAAGEAbgBjAGgAbwByAGoAcwA0ADAAMABAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAH//wAP) format("truetype"); }',e.sheet.cssRules.length)}}this.options=A||{},this.elements=[],e(this.options),this.isTouchDevice=function(){return!!("ontouchstart"in window||window.DocumentTouch&&document instanceof DocumentTouch)},this.add=function(A){var i,o,s,c,r,a,h,l,u,d,f,g,p=[];if(e(this.options),"touch"===(g=this.options.visible)&&(g=this.isTouchDevice()?"always":"hover"),A||(A="h2, h3, h4, h5, h6"),0===(i=t(A)).length)return this;for(n(),o=document.querySelectorAll("[id]"),s=[].map.call(o,function(A){return A.id}),r=0;r<i.length;r++)if(this.hasAnchorJSLink(i[r]))p.push(r);else{if(i[r].hasAttribute("id"))c=i[r].getAttribute("id");else if(i[r].hasAttribute("data-anchor-id"))c=i[r].getAttribute("data-anchor-id");else{u=l=this.urlify(i[r].textContent),h=0;do{void 0!==a&&(u=l+"-"+h),a=s.indexOf(u),h+=1}while(-1!==a);a=void 0,s.push(u),i[r].setAttribute("id",u),c=u}d=c.replace(/-/g," "),(f=document.createElement("a")).className="anchorjs-link "+this.options.class,f.href="#"+c,f.setAttribute("aria-label","Anchor link for: "+d),f.setAttribute("data-anchorjs-icon",this.options.icon),"always"===g&&(f.style.opacity="1"),""===this.options.icon&&(f.style.font="1em/1 anchorjs-icons","left"===this.options.placement&&(f.style.lineHeight="inherit")),"left"===this.options.placement?(f.style.position="absolute",f.style.marginLeft="-1em",f.style.paddingRight="0.5em",i[r].insertBefore(f,i[r].firstChild)):(f.style.paddingLeft="0.375em",i[r].appendChild(f))}for(r=0;r<p.length;r++)i.splice(p[r]-r,1);return this.elements=this.elements.concat(i),this},this.remove=function(A){for(var e,n,i=t(A),o=0;o<i.length;o++)(n=i[o].querySelector(".anchorjs-link"))&&(-1!==(e=this.elements.indexOf(i[o]))&&this.elements.splice(e,1),i[o].removeChild(n));return this},this.removeAll=function(){this.remove(this.elements)},this.urlify=function(A){var t=/[& +$,:;=?@"#{}|^~[`%!'<>\]\.\/\(\)\*\\]/g;return this.options.truncate||e(this.options),A.trim().replace(/\'/gi,"").replace(t,"-").replace(/-{2,}/g,"-").substring(0,this.options.truncate).replace(/^-+|-+$/gm,"").toLowerCase()},this.hasAnchorJSLink=function(A){var e=A.firstChild&&(" "+A.firstChild.className+" ").indexOf(" anchorjs-link ")>-1,t=A.lastChild&&(" "+A.lastChild.className+" ").indexOf(" anchorjs-link ")>-1;return e||t||!1}}return A});

// Enable links for selected headers
var anchors = new AnchorJS();
anchors.add("h2, h3, h4")
</script>
<!-- Enable Hypothesis annotation. https://web.hypothes.is/ -->
<script src="https://hypothes.is/embed.js" async></script>
</body>
</html>
