## Abstract {.page_break_before}

Variation graphs have been proposed to represent human genomes and improve on different aspects of genomics analysis, such as read mapping and variant calling.
Structural variants (SVs) have been associated with diseases but they remain under-studied due to their complexity and certain technological challenges.
Still, thousands of SVs have been characterized and better catalogs are now being generated thanks to new technologies.
There is now an opportunity to integrate previously neglected SVs into the unified framework of variation graphs.
We first show that the vg toolkit is capable of genotyping insertions, deletions and inversions, even in the presence of small errors in the variant definition.
We then benchmarked vg across three high-quality sequence-resolved SV catalogs generated by recent studies.
vg was compared to state-of-the-art SV genotypers using simulated and real Illumina short reads.
On real data, vg produced the best genotype predictions systematically in all datasets.
Finally, we found that better graphs could be constructed directly from de novo assembly alignment.
We experimented with assemblies from 12 yeast strains and showed that SV genotyping was improved compared to graphs built from intermediate SV catalogs in the VCF format.
Our results demonstrate the power of variation graphs for SV genotyping.
Beyond single nucleotide variants and short insertions/deletions, the vg toolkit now includes SV in its unified variant calling framework and provides a natural solution to integrate high-quality SV catalogs and assemblies.

