## Introduction {.page_break_before}

Structural variation (SV) represents genomic mutation involving 50 bp or more and can take several forms, such as for example deletions, insertions, inversions, or translocations.
SVs have long been associated with developmental disorders, cancer and other complex diseases and phenotypes[@doi:10.1038/nrg3373].
However, SVs have been under-studied for technological reasons and due to their complexity as compared to other types of genomic variation.
Although whole-genome sequencing (WGS) made it possible to assess virtually any type of structural variation, many challenges remain.
SV-supporting reads are generally difficult to map to reference genomes, in part because most SVs are larger than the sequencing reads.
Repeated sequences in the genome often confuse read mapping algorithms, which can produce mappings that seem to support an SV.
In practice, large-scale projects had to combine several methods to achieve better accuracy.
This methodology has been used to compile catalogs with tens of thousands of SVs in humans[@tag:kgp2015; @tag:gonl].
Overall, SV detection from short-read sequencing remains laborious and of lower accuracy than small variant detection.
This explains why these variants and their impact have been under-studied as compared to single-nucleotide variants (SNVs) and small insertions/deletions (indels).

Over the last few years, exciting developments in sequencing technologies and library preparation have made it possible to produce long reads or retrieve long-range information over kilobases of sequence.
This is particularly useful for SV detection and de novo assembly.
Several recent studies using long-read or linked-read sequencing have produced large catalogs of structural variation, the majority of which was novel and sequence-resolved[@tag:chaissonCHM; @tag:smrtsv; @doi:10.1038/s41467-017-01343-4; @doi:10.1038/nmeth.4366; @tag:audano2019].
These technologies are also enabling the production of high-quality de novo genome assemblies[@tag:chaissonCHM; @doi:10.1038/nbt.4060], and large blocks of haplotype-resolved sequences[@doi:10.1038/nmeth.4035].
Such technical advances promise to expand the amount of known genomic variation in humans in the near future.
However, their cost prohibits their use in large-scale studies that require hundreds or thousands of samples, such as disease association studies.

At the same time, the reference genome is evolving from a linear reference to a graph-based reference that incorporates known genomic variation[@tag:patenGRrev; @tag:vgnbt; @doi:10.1038/s41588-018-0316-4].
By including variants in the graph, both read mapping and variant calling become variant-aware and benefit in term of accuracy and sensitivity[@tag:vgnbt].
In addition, different variant types are called simultaneously by a unified framework.
vg was the first openly available tool that scaled to multi-gigabase genomes and provides read mapping, variant calling and haplotype modeling[@tag:vgnbt].
In vg, graphs can be built from both variant catalogs in the VCF format or assembly alignment.
Other genome graph implementations have also been used specifically to genotype variants.
Using a sliding-window approach, GraphTyper realigns reads to a graph build from known SNVs and short indels[@tag:graphtyper].
BayesTyper build graphs of both short variants and SVs, and genotypes variants based on the khmer distribution of sequencing reads[@tag:bayestyper].
Here again, the graph-based approaches showed clear advantages over standard methods that use the linear reference.

Other SV genotyping approaches typically compare read mapping to the reference genome and to a sequence modified with the SV. 
For example SMRT-SV was designed to genotype SVs identified on PacBio reads[@tag:smrtsv].
SVTyper uses paired-end mapping and split-read mapping information and can genotype deletions, duplications, inversions and translocations[@tag:svtyper].
Delly provides a genotyping feature in addition to its discovery mode and can genotype all types of SVs although the VCF needs special formatting for some[@tag:delly].
SMRT-SV2 is a machine learning tool that was trained to genotype SVs from the alignment of read to the reference genome augmented with SV-containing sequences as alternate contigs[@tag:audano2019].


We show that the unified variant calling framework implemented in vg is capable of genotyping deletions, insertions and inversions.
We compare vg with state-of-the-art SV genotypers: SVTyper[@tag:svtyper], Delly[@tag:delly], BayesTyper[@tag:bayestyper] and SMRT-SV2[@tag:audano2019].
On simulation, vg is robust to small errors in the breakpoint location and outperforms most other methods on shallow sequencing experiments.
Starting from SVs discovered in recent long-read sequencing studies[@tag:audano2019;@tag:hgsvc;@doi:10.1038/sdata.2016.25;@doi:10.1101/281006], we evaluated the genotyping accuracy when using simulated or real Illumina reads.
Across all three datasets that we tested, vg is the best performing SV genotyper on real short-read data for all SV types and sizes.
Going further, we show that building graphs from the alignment of de novo assemblies leads to better genotyping performance.

