## Introduction {.page_break_before}

Structural variation (SV) represents genomic mutation involving 50 bp or more and can take several forms, such as for example deletions, insertions, inversions, or translocations.
SVs have long been associated with developmental disorders, cancer and other complex diseases and phenotypes[@doi:10.1038/nrg3373].
Many types of cancer often harbor large or complex rearrangements[@doi:10.1101/gr.143677.112].
However, SVs have been under-studied for technological reasons and due to their complexity as compared to other types of genomic variation.
Although whole-genome sequencing (WGS) made it possible to assess virtually any type of structural variation, many challenges remain.
SV-supporting reads are generally difficult to map to reference genomes, in part because most SVs are larger than the sequencing reads.
Repeated sequences in the genome often confuse read mapping algorithms, which can produce mappings that seem to support an SV.
In practice, large-scale projects had to combine several methods to achieve better accuracy.
This methodology has been used to compile catalogs with tens of thousands of SVs in humans[@tag:kgp2015; @tag:gonl].
Overall, SV detection from short-read sequencing remains laborious and of lower accuracy than small variant detection.
This explains why these variants and their impact have been under-studied as compared to single-nucleotide variants (SNVs) and small insertions/deletions (indels).

Over the last few years, exciting developments in sequencing technologies and library preparation have made it possible to produce long reads or retrieve long-range information over kilobases of sequence.
This is particularly useful for SV detection and de novo assembly.
Several recent studies using long-read or linked-read sequencing have produced large catalogs of structural variation, the majority of which was novel and sequence-resolved[@tag:chaissonCHM; @tag:smrtsv; @doi:10.1038/s41467-017-01343-4; @doi:10.1038/nmeth.4366; @tag:audano2019].
These technologies are also enabling the production of high-quality de novo genome assemblies[@tag:chaissonCHM; @doi:10.1038/nbt.4060], and large blocks of haplotype-resolved sequences[@doi:10.1038/nmeth.4035].
These technological advances promise to expand the amount of known genomic variation in humans in the near future.
However, their cost prohibits their use in large-scale studies that require hundreds or thousands of samples, such as disease association studies.

At the same time, the reference genome is evolving from a linear reference to a graph-based reference that incorporates known genomic variation[@tag:patenGRrev; @tag:vgnbt; @doi:10.1038/s41588-018-0316-4].
By including variants in the graph, both mapping and variant calling become variant-aware and benefit in term of accuracy and sensitivity[@tag:vgnbt].
Both the mapping and variant calling become variant-aware and benefit in term of accuracy and sensitivity.
In addition, different variant types are called simultaneously by a unified framework.
vg was the first openly available tool that scaled to multi-gigabase genomes and provides read mapping, variant calling and haplotype modeling from graphs built from both variant catalogs in the VCF format or assembly alignment[@tag:vgnbt].
Graphs have also been used specifically to genotype variants.
Using a sliding-window approach, GraphTyper realigns reads to a graph build from known SNVs and short indels[@tag:graphtyper].
BayesTyper build graphs of both short variants and SVs, and perform genotyping from the khmer distribution of sequencing reads[@tag:bayestyper].
Here again, the graph-approach showed clear advantages over standard approaches that use the linear reference.

Other SV genotyping approaches compare read mapping in the reference genome and a sequence modified with the SV. 
For example SMRT-SV was designed to genotype SVs identified on PacBio reads[@tag:smrtsv].
SVTyper uses paired-end mapping and split-read mapping information and can genotype deletions, duplications, inversions and translocations[@tag:svtyper]
Delly provides a genotyping feature in addition to its discovery mode and can genotype all types of SVs although the VCF needs special formatting for some[@tag:delly].


We show that vg's graph implementation is capable of genotyping SVs as part of its unified variant calling framework.
State of the art SV genotypers were compared on simulated and real sequencing data on which vg showed the best performance.
vg is capable of genotyping deletions, insertions and inversions.
Going further, we show that building graphs from de novo assembly alignment lead to better genotyping performance.

