## Introduction {.page_break_before}

Structural variation (SV) represents genomic mutation involving 50 bp or more and can take several forms, such as for example deletions, insertions, inversions, or translocations.
Although whole-genome sequencing (WGS) made it possible to assess virtually any type of structural variation, many challenges remain.
In particular, SV-supporting reads are difficult to map to reference genomes.
Multi-mapping, caused by widespread repeated sequences in the genome, is another issue because it often resembles SV-supporting signal.
As a result, many SV detection algorithms have been developed and multiple methods must usually be combined to minimize false positives.
Several large-scale projects used this ensemble approach, cataloging tens of thousands of SV in humans[@tag:kgp2015; @tag:gonl].
SV detection from short-read sequencing remains laborious and of lower accuracy, explaining why these variants and their impact have been under-studied as compared to single-nucleotide variants (SNVs) and small insertions/deletions (indels).

Over the last few years, exciting developments in sequencing technologies and library preparation made it possible to produce long reads or retrieve long-range information over kilobases of sequence.
These approaches are maturing to the point were it is feasible to analyze the human genome.
This multi-kbp information is particularly useful for SV detection and de novo assembly.
In the last few years, several studies using long-read or linked-read sequencing have produced large catalogs of structural variation, the majority of which were novel and sequence-resolved[@tag:chaissonCHM; @tag:smrtsv; @doi:10.1038/s41467-017-01343-4; @doi:10.1038/nmeth.4366; @tag:audano2019].
These technologies are also enabling high-quality de novo genome assemblies to be produced[@tag:chaissonCHM; @doi:10.1038/nbt.4060], as well as large blocks of haplotype-resolved sequences[@doi:10.1038/nmeth.4035].
These technological advances promise to expand the amount of known genomic variation in humans in the near future.

In parallel, the reference genome is evolving from a linear reference to a graph-based reference that contains known genomic variation[@tag:patenGRrev; @tag:vgnbt; @doi:10.1038/s41588-018-0316-4].
By having variants in the graph, mapping rates are increased and variants are more uniformly covered, including indels and variants in complex regions[@tag:vgnbt].
Both the mapping and variant calling become variant-aware and benefit in term of accuracy and sensitivity.
In addition, different variant types are called simultaneously by a unified framework.
Graphs have also been used locally, i.e. to call variants at the region level.
GraphTyper[@tag:graphtyper] and BayesTyper[@tag:bayestyper] both construct variation graphs of small regions and use them for variant genotyping.
Here again, the graph-approach showed clear advantages over standard approaches that use the linear reference.
Other SV genotyping approaches compare read mapping in the reference genome and a sequence modified with the SV. 
For example SMRT-SV was designed to genotype SVs identified on PacBio reads[@tag:smrtsv], SVTyper uses paired-end mapping and split-read mapping information[@tag:svtyper], and Delly provides a genotyping feature in addition to its discovery mode[@tag:delly].


