## Introduction {.page_break_before}

A structural variant (SV) is a genomic mutation involving 50 or more base pairs.
SVs can take several forms such as deletions, insertions, inversions, or translocations.
It is intuitive that their greater size relative to smaller events such as single nucleotide variants (SNVs) and small insertions and deletions (indels) would imply that SVs can have a larger impact on phenotype.
Indeed, SVs have long been associated with developmental disorders, cancer and other complex diseases and phenotypes[@doi:10.1038/nrg3373], but remain much more poorly studied than their smaller mutational counterparts.
This discrepancy stems from technological limitation: short read sequencing has provided the basis of most modern genome sequencing studies due to its high base-level accuracy and relatively low cost. However, it is poorly suited for discovering SVs, which often lie in repeat-rich regions and whose lengths can far exceed read sizes.
The central obstacle is in mapping short reads to the human reference genome.
It is generally difficult or impossible to unambiguously map a short read if the sample whose genome is being analyzed differs substantially from the reference at the read's location.
For example, if the read derives from the middle of a large insertion relative to the reference, there is no sequence in the reference that corresponds to a correct mapping.
The best result a read mapper could hope to produce would be to leave it unmapped.
In some cases, SVs can be inferred indirectly from short read mappings via split reads, discordant read pairs, or read depth, but the accuracy of such approaches remains limited[@tag:hgsvc].
Assembling deeply sequenced short reads and aligning the contigs or scaffolds back to the reference provides an effective means of SV discovery at increased computational and sequencing cost[@tag:denmark]. 

In practice, large-scale projects must often combine short read mapping with several other methods to achieve better accuracy.
This approach has been used to compile catalogs with tens of thousands of SVs in humans[@tag:kgp2015; @tag:gonl].
Recent developments in sequencing technologies and library preparation continue to pave the way for better SV detection.
Several recent studies using long-read or linked-read sequencing have produced large catalogs of structural variation, the majority of which was novel and sequence-resolved[@tag:chaissonCHM; @tag:smrtsv; @doi:10.1038/s41467-017-01343-4; @doi:10.1038/nmeth.4366; @tag:audano2019].
These technologies are also enabling the production of high-quality de novo genome assemblies[@tag:chaissonCHM; @doi:10.1038/nbt.4060], and large blocks of haplotype-resolved sequences[@doi:10.1038/nmeth.4035].
Such technical advances promise to expand the amount of known genomic variation in humans in the near future.
However, their cost currently prohibits their use in large-scale studies that require hundreds or thousands of samples, such as disease association studies.

Representing structural variation poses an additional challenge.
Standard formats such as VCF are unwieldy when used for SVs due to ambiguity in their specification and limited support across tools.
Incorporating SVs into a linear pan-genome reference via alt contigs involves custom pipelines and non-standard metadata formats linking the contigs back to the reference.
They often contain redundant sequence resulting in increased mapping ambiguity.
These issues make the alt contigs difficult to use, maintain, and scale as SV catalogs continue to expand.
[todo: cite? even just grch38 is a pretty good case study of how awful this is]

Pan-genome graph reference representations offer a solution to such problems[@tag:patenGRrev]. 
The graph structure can represent SVs as succinctly and directly as point mutations. 
Moreover, including known variants in the reference makes both read mapping and variant calling variant-aware.
This leads to benefits in terms of accuracy and sensitivity[@tag:vgnbt; @tag:bakeoff; @doi:10.1038/s41588-018-0316-4].
In addition, different variant types can be called simultaneously and scored consistently across types by a unified framework.

vg is the first openly available variation graph tool to scale to multi-gigabase genomes.
It provides read mapping, variant calling and haplotype modeling[@tag:vgnbt].
In addition, vg can build graphs both from variant catalogs in the VCF format and from assembly alignments.
Other tools have used genome graphs specifically to genotype variants.
GraphTyper realigns mapped reads to a graph built from known SNVs and short indels using a sliding-window approach[@tag:graphtyper].
BayesTyper first builds a set of graphs from known variants including SVs, then genotypes variants by comparing the distribution k-mers in the sequencing reads with the k-mers of putative haplotype paths in the graph[@tag:bayestyper].
These graph-based approaches showed clear advantages over standard methods that use only the linear reference.

Other SV genotyping approaches are typically based on reads mapped to the reference genome or a reference sequence modified with the SVs. 
For example SMRT-SV was designed to genotype SVs that were previously identified on PacBio reads[@tag:smrtsv].
SVTyper uses paired-end mapping and split-read mapping information and can genotype deletions, duplications, inversions and translocations[@tag:svtyper].
Delly provides a genotyping feature in addition to its discovery mode and can genotype all types of SVs, although the VCF needs special formatting for some[@tag:delly].
SMRT-SV2 is a machine learning tool that was trained to genotype SVs from the mappings of reads to the reference genome augmented with SV-containing sequences as alternate contigs[@tag:audano2019].
[todo: I think we need to expand on the description of these methods, especially smrt-sv.  I will take a stab in a future pass]

We show that the unified variant calling framework implemented in vg is capable of genotyping known deletions, insertions and inversions.
We compare vg with state-of-the-art SV genotypers: SVTyper[@tag:svtyper], Delly[@tag:delly], BayesTyper[@tag:bayestyper] and SMRT-SV2[@tag:audano2019].
On simulated variants, vg is robust to small errors in the breakpoint location, and it outperforms most other methods on shallow sequencing experiments.
Starting from SVs discovered in recent long-read sequencing studies[@tag:audano2019;@tag:hgsvc;@doi:10.1038/sdata.2016.25;@doi:10.1101/281006], we evaluated the genotyping accuracy using simulated and real Illumina reads.
Across all three datasets that we tested, vg is the best performing SV genotyper on real short-read data for all SV types.
In addition, we show that building graphs from the alignment of de novo assemblies leads to better genotyping performance.

