## Results

### Structural variation in vg

In addition to SNV and short indels, vg can handle large deletions and insertions (and inversion?) (Figure {@fig:cartoon}).
As a proof-of-concept we simulated genomes and SVs with a size distribution matching real SVs[@tag:hgsvc].
Some errors were added at the breakpoints to investigate their effect on genotyping.
In all simulations, vg performed as well or better than SVtyper[@tag:svtyper] and Delly[@tag:delly] (Figure {@fig:sim}).
Overall, vg was more robust to errors around the breakpoints.
For insertions, vg performed almost as well as in the presence of errors in contrast to Delly and BayesTyper whose performance dropped significantly.
For deletions, the F1 dropped for all methods but remained high for vg and SVtyper.

![**Large deletions and insertions in variation graphs**](images/VGSVcartoon.jpg){#fig:cartoon}

![**Simulation experiment**. For each experiment (method, depth and input VCF with/without erros), the deciles of the call qualities were used as threshold and the maximum F1 is reported on the y-axis.](images/simerror.svg){#fig:sim width=80%}

### HGSVC

Chaisson et al.[@tag:hgsvc] provide a high-quality SV catalog of three samples, obtained using a consensus from different sequencing, phasing and variant caling technologies. 



#### (Whole-genome) Simulation

The phasing information in the HGSVC VCF was used to extract two haplotypes for sample HG00514, and 30X pairend-end reads were simulated using vg sim.  The reads were used to call VCFs then compared back to the original HGSVC calls (Figure {@fig:hgsvc-sim} and Table {@tbl:hgsvc-sim}).

When restricting the comparisons to regions not identified as tandem repeats or segmental duplications in the Genome Browser (Table {@tbl:hgsvc-sim-nonrepeat}).


![**HGSVC simulated reads**. ](images/hgsvc-sim.svg){#fig:hgsvc-sim width=80%}

![**HGSVC real reads**. ](images/hgsvc-real.svg){#fig:hgsvc-real width=80%}

#### (Whole-genome) Real reads

Figure {@fig:hgsvc-real}. 
Tables {@tbl:hgsvc-real} and {@tbl:hgsvc-real-nonrepeat} for results over the genome or when restricting the comparisons to regions not identified as tandem repeats or segmental duplications in the Genome Browser.

The input SVs could also be derived from short-reads. 
For example, SVs were first discovered by Manta(REF) from Illumina reads and then genotyped.
For BayesTyper, the precision drops for insertions while the recall increases for deletions (Figure {@fig:hgsvc-real-manta}).


 
### Genotyping SV using vg and de novo assemblies

We investigated whether genome graphs derived from genome-genome alignments yield advantages for SV genotyping.
To this end, we analyzed public sequencing datasets for 12 yeast strains from two clades (S. cerevisiae and S. paradoxus) [@doi:10.1038/ng.3847].
From these datasets, we generated two different types of genome graphs.
The first graph type (in the following called *construct graph*) was created from a linear reference genome of the S.c. S288C strain and a set of SVs relative to this reference strain in VCF format.
We compiled the SV set using the output of three methods for SV detection from genome assemblies: Assemblytics [@doi:10.1093/bioinformatics/btw369], AsmVar [@doi:10.1186/s13742-015-0103-4] and paftools [@doi:10.1093/bioinformatics/bty191].
All three methods were run to detect SVs between the reference strain S.c. S288C and each of the other 11 strains.
Merging the results from the three methods and the 11 strains provided us with a high-sensitivity set of SVs occuring in the two yeast clades.
We used this set to construct the *construct graph*.
The second graph (in the following called *cactus graph*) was derived from a multiple genome alignment of all 12 strains using our Cactus tool [@doi:10.1101/gr.123356.111].
While the *construct graph* is still mainly linear and highly dependent on the reference genome, the cactus graph is completely unbiased in that regard.

![**Mapping quality comparison.** The fraction of reads mapped (stratified by mapping quality threshold) to the cactus graph (y-axis) and the construct graph (x-axis) are compared](images/yeast-mapping-quality.svg){#fig:mapping-qual-comp width=80%}

In a first step, we tested our hypothesis that the *cactus graph* has higher mappability due to its better representation of sequence diversity among the yeast strains.
Fig. @fig:mapping-qual-comp shows the fraction of Illumina reads from the 12 strains that was mapped with a mapping quality above a certain threshold to the *cactus graph* and to the *construct graph*.
Generally, more reads were mapped to the *cactus graph* than to the *construct graph* regardless of the chosen mapping quality threshold.
Only for the reference strain S.c. S288C, both graphs exhibited similar mappability.
This suggests that not the higher sequence content in the *cactus graph*  alone (XX Mb compared to XX Mb in the *construct graph*) drives the improvement in mappability.
Instead, our measurements suggest that genetic distance to the reference strain increases the advantage of the *cactus graph* over the *construct graph*.
Consequently. the benefit of the *cactus graph* is largest for strains in the S. paradoxus clade and smaller for reads from strains in the S. cerevisiae clade.

![**Mapping identity comparison.** The fraction of reads mapped (stratified by percent identity threshold) to the cactus graph (y-axis) and the construct graph (x-axis) are compared](images/yeast-mapping-identity.svg){#fig:mapping-id-comp width=80%}

When we explored the mapping identity of the short reads on the graphs, we observed a similar trend (see Fig. @fig:mapping-id-comp).
For strains in the S. paradoxus clade, the *cactus graph* enabled substantially more mappings with high percent identity than the *construct graph*.
With strains in the S. cerevisiae clade, the difference was smaller, at least for a percent identity threshold up to 90%.
When comparing read fractions with perfect identity (i.e. percent identity threshold = 100%), the *cactus graph* clearly outperforms the *construct graph* on 11 out of 12 samples.
The only exception again is the reference strain S288C.

![**SV genotyping comparison.** Average mapping identity of short reads mapped to the *cactus graph* (y-axis) and *construct graph* (x-axis) is compared. Colors and shapes represent the 11 non-reference strains and two clades, respectively](images/yeast-genotyping-identity.svg){#fig:geno-comp-identity width=80%}

Next, we compared the SV genotype performance of both graphs.
We mapped short reads from the 11 non-reference strains to both graphs and called variants using vg's variant calling module.
To compare the callsets from both graphs, we generated a sample graph for each callset using the reference genome and the callset.
Each sample graph is a graph representation of the respective callset.
If a given callset is correct, we would expect that reads from the same sample can be mapped confidently and with high identity to the corresponding sample graph.
Therefore, we compared the average mapping identity of the short reads on both types of sample graphs (see Figure {@fig:geno-comp-identity}).
Similar to the results of our mapping analysis above, the *cactus graph* clearly outperformed the *construct graph* for strains in the S. paradoxus clade.
With strains in the S. cerevisiae clade, both graphs were on a par.

![**SV genotyping comparison.** Average mapping quality of short reads mapped to the *cactus graph* (y-axis) and *construct graph* (x-axis) is compared. Colors and shapes represent the 11 non-reference strains and two clades, respectively](images/yeast-genotyping-quality.svg){#fig:geno-comp-quality width=80%}

![**SV genotyping comparison.** Average alignment score of short reads mapped to the *cactus graph* (y-axis) and *construct graph* (x-axis) is compared. Colors and shapes represent the 11 non-reference strains and two clades, respectively](images/yeast-genotyping-score.svg){#fig:geno-comp-score width=80%}

This trend was confirmed when we looked at two our measures, average mapping quality and average alignment score (see Figures {@fig:geno-comp-quality} and {@fig:geno-comp-score}).

