## Methods

### Comparison of SVs

The variants are first normalized with `bcftools norm` to ensure consistent representation between called variants and baseline variants.
We then implemented an overlap-based strategy to compare SVs and compute evaluation metrics.

For deletions and inversions, the affected region in the reference genome is overlapped and matched between the two sets od SVs.
First, we select pairs of SVs with at least 10% reciprocal overlap.
Then for each variant we compute the proportion of its region that is covered by an overlapping variant in the other set.
If this coverage proportion is higher than 80%, the variant is considered *covered*.
True positives are covered variants from the call set or the truth set.
False positives are variants from the call set that are not covered.
False negative are variants from the truth set that are not covered.

For insertions, we select pairs of insertions that are located no farther than XX bp from each other.
We then align the inserted sequences using a Smith-Waterman alignment.
Then for each insertion we compute the proportion of its inserted sequence that aligns a matched variant in the other set.
As for deletions/inversions, this coverage proportion is used to annotate variants as true positives, false positives and false negatives.

sveval accepts VCF files with symbolic or explicit representation of the SVs.
If the explicit representation is used, multi-allelic variants are split and their sequences right-trimmed.
When inversions are considered, the reverse-complement of the ALT sequence of variants larger than 10 bp is aligned with the REF sequence and classified as an inversion if more than 80% of the sequence aligns.

We assess either the calling performance (absence/presence of a SV) or the genotyping performance.
For the *calling* evaluation, both heterozygous and homozygous alternate SVs are compared using the approach described above.
To compute genotype-level metrics, the heterozygous and homozygous SVs are compared separately.
Before splitting the variants by genotype, consecutive heterozygous variants are first stitched together if located at less that 20 bp from each other, and then merged into homozygous variants if their reciprocal overlap is at least 80%.

### Simulation experiment

We simulated a synthetic genome with SVs.
The genomic sequence was simulated and each SV separated by XX bp from each other. 
XXXX Insertions, deletions and inversions were added. 
The size of deletions and insertions followed the distribution of real SVs from the HGSVC catalog.
We used the same size distribution as deletions for inversions.
Then a VCF file for three simulated samples was produced with random genotypes (homozygous reference, heterozygous, homozygous alternate).
We created another VCF file containing errors in the SV definition.
The one or both breakpoints of deletions were shifted between 1 and 10 bp.
The location and sequence of insertions were also modified, either shifted or deleted at the flanks, again up to 10 bp. 
The different methods were tested using as input either the true VCF or the VCF that contained errors.
For vg, a graph was constructed from the VCF file, indexed and used to map simulated reads and call variants using toil-vg (see above/below/LINK).
Other methods take a VCF as input. COMMANDS

Paired-end reads were simulated using `vg sim` on graph that contained the true SVs.
Different read depth were tested: 1x, 3x, 7x, 10x, 13x, 20x.
Real Illumina reads from XX (see Glenn's simulation) were used to model base qualities and sequencing errors.

The genotypes called in each experiment (method/VCF with or without errors/sequencing depth) were compared to the true SV genotypes to compute the precision, recall and F1 score (see above/LINK).

### Breakpoint fine-tuning using graph augmentation

vg can call variants after augmenting the graph with the read alignments to discover new variants (see toil-vg methods maybe).
We tested if this approach could fine-tune the breakpoint location of SVs in the graph.
We started with the graph that contained approximate SVs (1-10 bp errors in breakpoint location) and 20x simulated reads from the simulation experiment (see above/LINK).
The variants called after graph augmentation were compared with the true SVs and considered fine-tuned if the breakpoints matched exactly.

### HGSVC

#### (Whole-genome) Simulation

The phasing information in the HGSVC VCF was used to extract two haplotypes for sample HG00514, and 30X pairend-end reads were simulated using vg sim.  The reads were used to call VCFs then compared back to the original HGSVC calls.

When restricting the comparisons to regions not identified as tandem repeats or segmental duplications in the Genome Browser (Table {@tbl:hgsvc}).

#### (Whole-genome) Real reads

Illumina reads were downloaded from ...

### De novo assembly alignment from 12 yeast strains
