## Discussion


*Potential topics for the discussion.*

#### Providing a resource to be used by large-scale sequencing project

As a result of this study we provide a variation graph containing XX millions of SNVs and indels from the 1000 Genomes Project as well as XX thousands of SVs derived from long-read sequencing.
This variation graph could serve as a richer reference for large scale projects that use short-read sequencing.
More and more large-scale projects are sequencing the genomes of thousands or hundreds of thousands of individuals, e.g the Pancancer Analysis of Whole Genomes, the Genomics England initiative, and the TOPMed consortium(REFS).
These large WGS studies will provide a deeper look into the mechanism of common diseases and, in some cases, will be used directly in a clinical setting.
Clinicians and researchers are eager to use these growing WGS resources to interrogate the importance of SVs in disease at a scale never achieved before, either to get a more complete picture of the genetic factors of a disease or to produce a more comprehensive clinical report.
As sequencing reaches the clinic, whole-genome sequencing will become routine for many patients.
Clinicians will rely on variant calling and interpretation for diagnosis and treatment.
For variant interpretation in particular, a comprehensive characterization of the genomic variation and a haplotype view will be extremely valuable. % Unclear

#### Easier to use

Some methods require additional information or special VCF formatting [@doi:10.1101/558247], vg can take explicit or symbolic VCFs, or assembly alignment as input.
SVTyper was designed to use VCFs created by Lumpy.
The genotyping module from Delly was implemented for variants found by its discovery module.
SMRT-SV requires a VCF with information about XXX.
Nebula, khmer-based genotyper requires reads from a sample containing the SV during khmer selection[@doi:10.1101/566620].

#### Assemblies are the future

Our results suggest that constructing a graph from de novo assembly alignment is more representative of the sequencing reads and leads to better SV genotyping.
De novo assemblies for human are becoming more and more common, for example from optimized mate-pair libraries[@doi:10.1038/nature23264] or long-read sequencing[@doi:10.1038/nature23264].
For an optimal representation of the genomic variation, we expect the future graphs to include information from numerous de novo assembly alignment.

#### Future improvements in vg

The vg toolkit is in active development.
Read mapping is an area of constant improvement, both in term of computational efficiency and accuracy.
For example, haplotype information can be modeled in variation graph and, in the future, assist read mapping and variant calling.
These upcoming developments will directly benefit SV genotyping with vg.

#### Limitations

Copy number variants (CNVs) are currently represented as deletions or insertions. 
For this reason duplications are represented as additional sequence rather than encoded as a loop in the graph.
While this is sufficient to represent single copy changes, such as deletions or single tandem duplications, CNVs with multi-copies states are not addressed by the current implementation.
The calling algorithm would need to be extended to model copy number in order to assess these variants.

Near-breakpoint resolution is necessary.
Simulations have shown that SV genotyping with vg is robust to errors up to 10 bp in breakpoint location.

The genotyping evaluation of invertions is limited by the lack of existing gold-standards.
We showed that vg is capable of genotyping simple inversions from simulation or the few discovered in ?? datasets.
However most inversions are more complex and involve small insertions/deletions around their breakpoints.
While these complex variants are difficult to represent in the VCF format, they would naturally be represented through the alignment of de novo assemblies.
For example, in our experiment with yeast assemblies, we identified XX variants that contains at least XX inverted bases.
