<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Glenn Hickey" />
  <meta name="author" content="David Heller" />
  <meta name="author" content="Jean Monlong" />
  <meta name="author" content="Jonas Andreas Sibbesen" />
  <meta name="author" content="Jouni Siren" />
  <meta name="author" content="Jordan Eizenga" />
  <meta name="author" content="Eric Dawson" />
  <meta name="author" content="Erik Garrison" />
  <meta name="author" content="Adam Novak" />
  <meta name="author" content="Benedict Paten" />
  <meta name="dcterms.date" content="2019-05-28" />
  <meta name="keywords" content="structural variation, pangenome, variant genotyping" />
  <title>Genotyping structural variation in variation graphs with the vg toolkit</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="github-pandoc.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!-- Insert Analytics Script Below -->
  <script>
  </script>
  <!-- End Analytics Script -->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Genotyping structural variation in variation graphs with the vg toolkit</h1>
</header>
<p><small><em> This manuscript (<a href="https://jmonlong.github.io/manu-vgsv/v/dc06adada54739cf7fd46d433966b34bf985806e/">permalink</a>) was automatically generated from <a href="https://github.com/jmonlong/manu-vgsv/tree/dc06adada54739cf7fd46d433966b34bf985806e">jmonlong/manu-vgsv@dc06ada</a> on May 28, 2019. </em></small></p>
<h2 id="authors">Authors</h2>
<p><a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Glenn Hickey<sup>1,☯</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> David Heller<sup>1,2,☯</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Jean Monlong<sup>1,☯</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Jonas Andreas Sibbesen<sup>1</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Jouni Siren<sup>1</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Jordan Eizenga<sup>1</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Eric Dawson<sup>3</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Erik Garrison<sup>1</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Adam Novak<sup>1</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Benedict Paten<sup>1,†</sup></p>
<p><sup>☯</sup> — These authors contributed equally to this work</p>
<p><sup>†</sup> — To whom correspondence should be addressed: bpaten@ucsc.edu <small></p>
<ol type="1">
<li>UC Santa Cruz Genomics Institute, University of California, Santa Cruz, California, USA</li>
<li>Max Planck Institute for Molecular Genetics, Berlin, Germany</li>
<li>Department of Genetics, University of Cambridge, Cambridge, UK</li>
</ol>
<p></small></p>
<h2 id="abstract" class="page_break_before">Abstract</h2>
<p>Structural variants (SVs) are significant components of genetic diversity and have been associated with diseases, but the technological challenges surrounding their representation and identification make them difficult to study relative to point mutations. Still, thousands of SVs have been characterized, and catalogs continue to improve with new technologies. In parallel, variation graphs have been proposed to represent human pangenomes, offering reduced reference bias and better mapping accuracy than linear reference genomes. We contend that variation graphs provide an effective means for leveraging SV catalogs for short-read SV genotyping experiments. In this work, we extend vg (a software toolkit for working with variation graphs) to support SV genotyping. We show that it is capable of genotyping insertions, deletions and inversions, even in the presence of small errors in the location of the SVs breakpoints. We then benchmark vg against state-of-the-art SV genotypers using three high-quality sequence-resolved SV catalogs generated by recent studies ranging up to 97,368 variants in size. We find that vg systematically produces the best genotype predictions in all datasets. In addition, we use assemblies from 12 yeast strains to show that graphs constructed directly from aligned <em>de novo</em> assemblies can improve genotyping compared to graphs built from intermediate SV catalogs in the VCF format. Our results demonstrate the power of variation graphs for SV genotyping. Beyond single nucleotide variants and short insertions/deletions, the vg toolkit now incorporates SVs in its unified variant calling framework and provides a natural solution to integrate high-quality SV catalogs and assemblies.</p>
<h2 id="introduction" class="page_break_before">Introduction</h2>
<p>A structural variant (SV) is a genomic mutation involving 50 or more base pairs. SVs can take several forms such as deletions, insertions, inversions, translocations or other complex events.<br />
Due to their greater size, SVs often have a larger impact on phenotype than smaller events such as single nucleotide variants (SNVs) and small insertions and deletions (indels)<span class="citation" data-cites="2gpKwL67">[<a href="#ref-2gpKwL67">1</a>]</span>. Indeed, SVs have long been associated with developmental disorders, cancer and other complex diseases and phenotypes<span class="citation" data-cites="ebc66eBr">[<a href="#ref-ebc66eBr">2</a>]</span>.</p>
<p>Despite their importance, SVs remain much more poorly studied than their smaller mutational counterparts. This discrepancy stems from technological limitations. Short read sequencing has provided the basis of most modern genome sequencing studies due to its high base-level accuracy and relatively low cost, however, it is poorly suited for discovering SVs. The central obstacle is in mapping short reads to the human reference genome. It is generally difficult or impossible to unambiguously map a short read if the sample whose genome is being analyzed differs substantially from the reference at the read’s location. The large size of SVs virtually guarantees that short reads derived from them will not map to the linear reference genome. For example, if a read corresponds to sequence in the middle of a large reference-relative insertion, then there is no location in the reference that corresponds to a correct mapping. The best result a read mapper could hope to produce would be to leave it unmapped. Moreover, SVs often lie in repeat-rich regions, which further frustrate read mapping algorithms.</p>
<p>Short reads can be more effectively used to genotype known SVs. This is important, as even though efforts to catalog SVs with other technologies have been highly successful, their cost currently prohibits their use in large-scale studies that require hundreds or thousands of samples such as disease association studies. Traditional SV genotypers start from reads that were mapped to a reference genome, extracting aberrant mapping that might support the presence of the SV of interest. State-of-art methods like SVTyper<span class="citation" data-cites="AltPnocw">[<a href="#ref-AltPnocw">3</a>]</span> and Delly<span class="citation" data-cites="nLvQCjXU">[<a href="#ref-nLvQCjXU">4</a>]</span> typically focus on split reads and paired reads mapped too close or too far from each other. These discordant reads are tallied and remapped to the reference sequence modified with the SV of interest in order to genotype deletions, insertions, duplications, inversions and translocations. SMRT-SV v2 uses a different approach: the reference genome is augmented with SV-containing sequences as alternate contigs and the resulting mappings are evaluated with a machine learning model trained for this purpose<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>.</p>
<p>The catalog of known SVs in human is quickly expanding. Several large-scale projects have used short-read sequencing and extensive discovery pipelines on large cohorts, compiling catalogs with tens of thousands of SVs in humans<span class="citation" data-cites="qA6dWFP py6BC5kj">[<a href="#ref-qA6dWFP">6</a>,<a href="#ref-py6BC5kj">7</a>]</span>, using split read and discordant pair based methods like SVTyper<span class="citation" data-cites="AltPnocw">[<a href="#ref-AltPnocw">3</a>]</span> and Delly<span class="citation" data-cites="nLvQCjXU">[<a href="#ref-nLvQCjXU">4</a>]</span> to find SVs using short read sequencing. More recent studies using long-read or linked-read sequencing have produced large catalogs of structural variation, the majority of which was novel and sequence-resolved<span class="citation" data-cites="z91V6jjU rs7e40wC PRx3qEIm 121OWcTA4 3NNFS6U2">[<a href="#ref-PRx3qEIm">10</a>,<a href="#ref-121OWcTA4">11</a>,<a href="#ref-3NNFS6U2">5</a>,<a href="#ref-z91V6jjU">8</a>,<a href="#ref-rs7e40wC">9</a>]</span>. These technologies are also enabling the production of high-quality <em>de novo</em> genome assemblies<span class="citation" data-cites="z91V6jjU 6KbgcueR">[<a href="#ref-6KbgcueR">12</a>,<a href="#ref-z91V6jjU">8</a>]</span>, and large blocks of haplotype-resolved sequences<span class="citation" data-cites="Pu6SY37C">[<a href="#ref-Pu6SY37C">13</a>]</span>. Such technical advances promise to expand the amount of known genomic variation in humans in the near future, and further power SV genotyping studies. Representing known structural variation in the wake of increasingly larger datasets poses a considerable challenge, however. VCF, the standard format for representing small variants, is unwieldy when used for SVs due its unsuitability for expressing nested or complex variants. Another strategy consists in incorporating SVs into a linear pangenome reference via alt contigs, but it also has serious drawbacks. Alt contigs tend to increase mapping ambiguity. In addition, it is unclear how to scale this approach as SV catalogs grow.</p>
<p>Pangenomic graph reference representations offer an attractive approach for storing genetic variation of all types<span class="citation" data-cites="Qa8mx6Ll">[<a href="#ref-Qa8mx6Ll">14</a>]</span>. These graphical data structures can seamlessly represent both SVs and point mutations using the same semantics. Moreover, including known variants in the reference makes both read mapping and variant calling variant-aware. This leads to benefits in terms of accuracy and sensitivity<span class="citation" data-cites="10jxt15v0 DuODeStx 11Jy8B61m">[<a href="#ref-10jxt15v0">15</a>,<a href="#ref-DuODeStx">16</a>,<a href="#ref-11Jy8B61m">17</a>]</span>. The coherency of this model allows different variant types to be called and scored simultaneously in a unified framework.</p>
<p>vg is the first openly available variation graph tool to scale to multi-gigabase genomes. It provides read mapping, variant calling and visualization tools<span class="citation" data-cites="10jxt15v0">[<a href="#ref-10jxt15v0">15</a>]</span>. In addition, vg can build graphs both from variant catalogs in the VCF format and from assembly alignments.</p>
<p>Other tools have used genome graphs or pangenomes to genotype variants. GraphTyper realigns mapped reads to a graph built from known SNVs and short indels using a sliding-window approach<span class="citation" data-cites="ohTIiqfV">[<a href="#ref-ohTIiqfV">18</a>]</span>. BayesTyper first builds a set of graphs from known variants including SVs, then genotypes variants by comparing the distribution of k-mers in the sequencing reads with the k-mers of haplotype candidate paths in the graph<span class="citation" data-cites="14Uxmwbxm">[<a href="#ref-14Uxmwbxm">19</a>]</span>. SMRT-SV v2 uses a different approach: the reference genome is augmented with SV-containing sequences as alternate contigs and the resulting mappings are evaluated with a machine learning model trained for this purpose<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>. These graph-based approaches showed clear advantages over standard methods that use only the linear reference.</p>
<p>In this work, we present a SV genotyping framework based on the variation graph model and implemented in the vg toolkit. We show that this method is capable of genotyping known deletions, insertions and inversions, and that its performance is not inhibited by small errors in the specification of SV allele breakpoints. We evaluated the genotyping accuracy of our approach using simulated and real Illumina reads and a pangenome built from SVs discovered in recent long-read sequencing studies<span class="citation" data-cites="3NNFS6U2 vQTymKCj 14neTdqfN 16GvGhO20">[<a href="#ref-vQTymKCj">20</a>,<a href="#ref-14neTdqfN">21</a>,<a href="#ref-16GvGhO20">22</a>,<a href="#ref-3NNFS6U2">5</a>]</span>, We also compared vg’s performance with state-of-the-art SV genotypers: SVTyper<span class="citation" data-cites="AltPnocw">[<a href="#ref-AltPnocw">3</a>]</span>, Delly<span class="citation" data-cites="nLvQCjXU">[<a href="#ref-nLvQCjXU">4</a>]</span>, BayesTyper<span class="citation" data-cites="14Uxmwbxm">[<a href="#ref-14Uxmwbxm">19</a>]</span> and SMRT-SV v2<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>. Across these three datasets that we tested, which range in size from 26k to 97k SVs, vg is the best performing SV genotyper on real short-read data for all SV types. Finally, we demonstrate that a pangenome graph built from the alignment of <em>de novo</em> assemblies of diverse <em>Saccharomyces cerevisiae</em> strains improves SV genotyping performance.</p>
<h2 id="results">Results</h2>
<h3 id="structural-variation-in-vg">Structural variation in vg</h3>
<p>We used vg to implement a straightforward SV genotyping pipeline. Reads are mapped to the graph and used to compute the read support for each node and edge (see <a href="#supplementary-information">Supplementary Information</a> for a description of the graph formalism). Sites of variation are then identified using the snarl (aka “bubble”) decomposition as described in <span class="citation" data-cites="xJlNnKH2">[<a href="#ref-xJlNnKH2">23</a>]</span>, each resulting site being represented as a subgraph of the larger graph. For each site, we determine the two most supported paths (haplotypes), and use their relative support in the read evidence to produce a genotype at that site (Figure <a href="#fig:1">1</a>a). We describe the pipeline in more detail in <a href="#toil-vg">Methods</a>. We rigorously evaluated the accuracy of our method on a variety of datasets, and present these results in the remainder of this section.</p>
<figure>
<img src="images/panel1.png" alt="Figure 1: Structural variation in vg. a) vg uses the read coverage over possible paths to genotype variants in a bubble or more complex snarl. The cartoon depicts the case of an heterozygous insertion and an homozygous deletion. The algorithm is described in more details in Methods. b) Simulation experiment. Each subplot shows a comparison of genotyping accuracy for four SV calling methods. Results are separated between types of variation (insertions, deletions, and inversions). The experiments were also repeated with small random errors introduced to the VCF to simulate breakpoint uncertainty. For each experiment, the y-axis shows the maximum F1 across different minimum quality thresholds. SVTyper cannot genotype insertions, hence the missing line in the top panels." id="fig:1" /><figcaption><span>Figure 1:</span> <strong>Structural variation in vg.</strong> a) vg uses the read coverage over possible paths to genotype variants in a bubble or more complex snarl. The cartoon depicts the case of an heterozygous insertion and an homozygous deletion. The algorithm is described in more details in <a href="#toil-vg-call">Methods</a>. b) Simulation experiment. Each subplot shows a comparison of genotyping accuracy for four SV calling methods. Results are separated between types of variation (insertions, deletions, and inversions). The experiments were also repeated with small random errors introduced to the VCF to simulate breakpoint uncertainty. For each experiment, the y-axis shows the maximum F1 across different minimum quality thresholds. SVTyper cannot genotype insertions, hence the missing line in the top panels.</figcaption>
</figure>
<h3 id="simulated-dataset">Simulated dataset</h3>
<p>As a proof of concept, we simulated genomes and different types of SVs with a size distribution matching real SVs<span class="citation" data-cites="vQTymKCj">[<a href="#ref-vQTymKCj">20</a>]</span>. We compared vg against SVTyper, Delly, and BayesTyper across different levels of sequencing depth. We also added some errors (1-10bp) to the location of the breakpoints to investigate their effect on genotyping accuracy (see <a href="#simulation-experiment">Methods</a>). The results are shown in Figure <a href="#fig:1">1</a>b.</p>
<p>When using the correct breakpoints, vg tied with Delly as the best genotyper for deletions, and with BayesTyper as the best genotyper for insertions. For inversions, vg was the second best genotyper after BayesTyper. The differences between the methods were the most visible at lower sequencing depth. In the presence of 1-10 bp errors in the breakpoint locations, the performance of Delly and BayesTyper dropped significantly (Figure <a href="#fig:1">1</a>b). The dramatic drop for BayesTyper can be explained by its k-mer-based approach that requires precise breakpoints. In contrast, vg was only slightly affected by the presence of errors. For vg, the F1 scores for all SV types decreased no more than 0.07. Overall, these results show that vg is capable of genotyping SVs and is robust to breakpoint inaccuracies in the input VCF.</p>
<h3 id="hgsvc-dataset">HGSVC dataset</h3>
<p>72,485 structural variants from The Human Genome Structural Variation Consortium (HGSVC) were used to benchmark the genotyping performance of vg against the three other SV genotyping methods. This high-quality SV catalog was generated from three samples using a consensus from different sequencing, phasing, and variant calling technologies<span class="citation" data-cites="vQTymKCj">[<a href="#ref-vQTymKCj">20</a>]</span>. The three individual samples represent different human populations: Han Chinese (HG00514), Puerto-Rican (HG00733), and Yoruban Nigerian (NA19240). We used these SVs to construct a graph with vg and as input for the other genotypers. Using short sequencing reads, the SVs were genotyped and compared with the genotypes in the original catalog (see <a href="#hgsvc-analysis">Methods</a>).</p>
<p>First we compared the methods using simulated reads for HG00514. This represents the ideal situation where the SV catalog exactly matches the SVs supported by the reads. While vg outperformed Delly and SVTyper, BayesTyper showed the best F1 score and precision-recall trade-off (Figures <a href="#fig:2">2</a> and <a href="#fig:hgsvc-sim-geno">S1</a>, Table <a href="#tbl:hgsvc">S1</a>). When restricting the comparisons to regions not identified as tandem repeats or segmental duplications, the genotyping predictions were significantly better for all methods, with vg almost as good as BayesTyper on deletions (F1 of 0.944 vs 0.955). We observed similar results when evaluating the presence of an SV call instead of the exact genotype (Figures <a href="#fig:2">2</a> and <a href="#fig:hgsvc-sim">S2</a>). Overall, both graph-based methods, vg and BayesTyper, outperformed the other two methods tested.</p>
<figure>
<img src="images/hgsvc-giab-best-f1.png" alt="Figure 2: Structural variants from the HGSVC and Genome in a Bottle datasets. HGSVC: Simulated and real reads were used to genotype SVs and compared with the high-quality calls from Chaisson et al.[20]. Reads were simulated from the HG00514 individual. Using real reads, the three HG00514, HG00733, and NA19240 individuals were tested. GiaB: Real reads from the HG002 individual were used to genotype SVs and compared with the high-quality calls from the Genome in a Bottle consortium[21,22]. Maximum F1 score for each method (color), across the whole genome or focusing on non-repeat regions (x-axis). We evaluated the ability to predict the presence of an SV (transparent bars) and the exact genotype (solid bars). Results are separated across panels by variant type: insertions and deletions. SVTyper cannot genotype insertions, hence the missing bars in the top panels." id="fig:2" /><figcaption><span>Figure 2:</span> <strong>Structural variants from the HGSVC and Genome in a Bottle datasets</strong>. HGSVC: Simulated and real reads were used to genotype SVs and compared with the high-quality calls from Chaisson et al.<span class="citation" data-cites="vQTymKCj">[<a href="#ref-vQTymKCj">20</a>]</span>. Reads were simulated from the HG00514 individual. Using real reads, the three HG00514, HG00733, and NA19240 individuals were tested. GiaB: Real reads from the HG002 individual were used to genotype SVs and compared with the high-quality calls from the Genome in a Bottle consortium<span class="citation" data-cites="14neTdqfN 16GvGhO20">[<a href="#ref-14neTdqfN">21</a>,<a href="#ref-16GvGhO20">22</a>]</span>. Maximum F1 score for each method (color), across the whole genome or focusing on non-repeat regions (x-axis). We evaluated the ability to predict the presence of an SV (transparent bars) and the exact genotype (solid bars). Results are separated across panels by variant type: insertions and deletions. SVTyper cannot genotype insertions, hence the missing bars in the top panels.</figcaption>
</figure>
<p>We then repeated the analysis using real Illumina reads from the three HGSVC samples to benchmark the methods on a more realistic experiment. Here, vg clearly outperformed other approaches (Figures <a href="#fig:2">2</a> and <a href="#fig:hgsvc-real-geno">S3</a>). In non-repeat regions and across the whole genome, the F1 scores and precision-recall AUC were higher for vg compared to other methods. For example, for deletions in non-repeat regions, the F1 score for vg was 0.801 while the second best method, Delly, had a F1 score of 0.692. We observed similar results when evaluating the presence of an SV call instead of the exact genotype (Figures <a href="#fig:2">2</a> and <a href="#fig:hgsvc-real">S4</a>). Figure <a href="#fig:hgsvc-ex">3</a> shows examples of an exonic deletion and an exonic insertion that were correctly genotyped by vg but not by the other methods.</p>
<figure>
<img src="images/panel7.png" alt="Figure 3: Exonic SVs in the HGSVC dataset correctly genotyped by vg. Visualizations of the HGSVC graph as augmented by reads aligned by vg at two loci exonic loci harboring SVs. (A) 51 bp homozygous deletion in the last exon of the LONRF2 gene. (B) 114 bp homozygous insertion in a short tandem repeat region overlapping the first exon of the MED13L gene, a gene predicted to be loss of function intolerant. At bottom of each rendering, a horizontal black line represents the topologically sorted nodes of the graph. Black rectangles represent edges found in the graph. Above this rendering of the topology, the reference path from GRCh38 is shown (either in green or blue). Red and blue bars represent reads mapped to the graph. Thin lines in the reference path and read mappings highlight relative gaps (either insertions or deletions) against the full graph. The vg read mappings show consistent coverage even over these SVs." id="fig:hgsvc-ex" /><figcaption><span>Figure 3:</span> <strong>Exonic SVs in the HGSVC dataset correctly genotyped by vg</strong>. Visualizations of the HGSVC graph as augmented by reads aligned by vg at two loci exonic loci harboring SVs. (A) 51 bp homozygous deletion in the last exon of the LONRF2 gene. (B) 114 bp homozygous insertion in a short tandem repeat region overlapping the first exon of the MED13L gene, a gene predicted to be loss of function intolerant. At bottom of each rendering, a horizontal black line represents the topologically sorted nodes of the graph. Black rectangles represent edges found in the graph. Above this rendering of the topology, the reference path from GRCh38 is shown (either in green or blue). Red and blue bars represent reads mapped to the graph. Thin lines in the reference path and read mappings highlight relative gaps (either insertions or deletions) against the full graph. The vg read mappings show consistent coverage even over these SVs.</figcaption>
</figure>
<h3 id="other-long-read-datasets">Other long-read datasets</h3>
<h4 id="genome-in-a-bottle-consortium">Genome in a Bottle Consortium</h4>
<p>The Genome in a Bottle (GiaB) consortium is currently producing a high-quality SV catalog for an Ashkenazim individual (HG002)<span class="citation" data-cites="14neTdqfN 16GvGhO20">[<a href="#ref-14neTdqfN">21</a>,<a href="#ref-16GvGhO20">22</a>]</span>. Dozens of SV callers operating on datasets from short, long, and linked reads were used to produce this set of SVs. We evaluated the SV genotyping methods on this sample as well using the GIAB VCF, which also contains parental calls (HG003 and HG004), all totalling 30,224 SVs. vg performed similarly on this dataset as on the HGSVC dataset, with a F1 score of 0.75 for both insertions and deletions in non-repeat regions (Figures <a href="#fig:2">2</a>, <a href="#fig:giab-geno">S5</a> and <a href="#fig:giab">S6</a>, and Table <a href="#tbl:giab">S2</a>). As before, other methods produced lower F1 scores in most cases, although Delly and BayesTyper predicted better genotypes for deletions in non-repeat regions.</p>
<h4 id="smrt-sv-v2-catalog-and-training-data-3nnfs6u2">SMRT-SV v2 catalog and training data <span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span></h4>
<p>A recent study by Audano et al. generated a catalog of 97,368 SVs (referred as SVPOP below) using long-read sequencing across 15 individuals<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>. These variants were then genotyped from short reads across 440 individuals using the SMRT-SV v2 genotyper, a machine learning-based tool implemented for that study. The SMRT-SV v2 genotyper was trained on a pseudo-diploid genome constructed from high quality assemblies of two haploid cell lines (CHM1 and CHM13) and a single negative control (NA19240). We first used vg to genotype the SVs in this two-sample training dataset using 30X coverage reads, and compared the results with the SMRT-SV v2 genotyper. vg was systematically better at predicting the presence of an SV for both SV types, but SMRT-SV v2 produced better genotypes for deletions (see Figures <a href="#fig:chmpd-svpop">4</a>, <a href="#fig:chmpd-geno">S7</a> and <a href="#fig:chmpd">S8</a>, and Table <a href="#tbl:chmpd">S3</a>). To compare vg and SMRT-SV v2, we then genotyped SVs from the entire SVPOP catalog with both methods, using the read data from the three HGSVC samples described above. Given that the the SVPOP catalog contains these three samples, we once again evaluated accuracy by using the long-read genotypes as a baseline of comparison.</p>
<p>Compared to SMRT-SV v2, vg had a better precision-recall curve and a higher F1 for both insertions and deletions (SVPOP in Figures <a href="#fig:chmpd-svpop">4</a> and <a href="#fig:svpop">S9</a>, and Table <a href="#tbl:svpop">S4</a>). Of note, SMRT-SV v2 produces <em>no-calls</em> in regions where the read coverage is too low, and we observed that its recall increased when filtering these regions out the input set. Interestingly, vg performed well even in regions where SMRT-SV v2 produced <em>no-calls</em> (Figure <a href="#fig:svpop-regions">S10</a> and Table <a href="#tbl:svpop-regions">S5</a>). Audano et al. discovered 217 sequence-resolved inversions using long reads, which we attempted to genotype. vg correctly predicted the presence of around 14% of the inversions present in the three samples (Table <a href="#tbl:svpop">S4</a>). Inversions are often complex, harboring additional variation that makes their characterization and genotyping challenging.</p>
<figure>
<img src="images/chmpd-svpop-best-f1.png" alt="Figure 4: Structural variants from SMRT-SV v2 [5]. The pseudo-diploid genome built from two CHM cell lines and one negative control sample was originally used to train SMRT-SV v2 in Audano et al.[5]. It contains 16,180 SVs. The SVPOP panel shows the combined results for the HG00514, HG00733, and NA19240 individuals, three of the 15 individuals used to generate the high-quality SV catalog in Audano et al. [5]. Here, we report the maximum F1 score (y-axis) for each method (color), across the whole genome or focusing on non-repeat regions (x-axis). We evaluated the ability to predict the presence of an SV (transparent bars) and the exact genotype (solid bars). Genotype information is not available in the SVPOP catalog hence genotyping performance could not be evaluated." id="fig:chmpd-svpop" /><figcaption><span>Figure 4:</span> <strong>Structural variants from SMRT-SV v2 <span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span></strong>. The pseudo-diploid genome built from two CHM cell lines and one negative control sample was originally used to train SMRT-SV v2 in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>. It contains 16,180 SVs. The SVPOP panel shows the combined results for the HG00514, HG00733, and NA19240 individuals, three of the 15 individuals used to generate the high-quality SV catalog in Audano et al. <span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>. Here, we report the maximum F1 score (y-axis) for each method (color), across the whole genome or focusing on non-repeat regions (x-axis). We evaluated the ability to predict the presence of an SV (transparent bars) and the exact genotype (solid bars). Genotype information is not available in the SVPOP catalog hence genotyping performance could not be evaluated.</figcaption>
</figure>
<h3 id="graphs-from-alignment-of-de-novo-assemblies">Graphs from alignment of <em>de novo</em> assemblies</h3>
<p>We can construct variation graphs directly from whole genome alignments of multiple <em>de novo</em> assemblies<span class="citation" data-cites="10jxt15v0">[<a href="#ref-10jxt15v0">15</a>]</span>. This bypasses the need for generating an explicit variant catalog relative to a linear reference, which could be a source of error due to the reference bias inherent in read mapping and variant calling. Genome alignments from graph-based software such as Cactus <span class="citation" data-cites="1FgS53pXi">[<a href="#ref-1FgS53pXi">24</a>]</span> can contain complex structural variation that is extremely difficult to represent, let alone call, outside of a graph, but which is nevertheless representative of the actual genomic variation between the aligned assemblies. We sought to establish if graphs built in this fashion provide advantages for SV genotyping.</p>
<p>To do so, we analyzed public sequencing datasets for 12 yeast strains from two related clades (<em>S. cerevisiae</em> and <em>S. paradoxus</em>) <span class="citation" data-cites="7f5OKa5O">[<a href="#ref-7f5OKa5O">25</a>]</span>. We compared genotyping results using two different types of genome graphs. The graphs were constructed using 5 of the 12 strains. <em>S.c. S288C</em> was used as the reference strain, and we selected two other strains from each of the two clades (see <a href="#yeast-graph-analysis">Methods</a>). The first graph (called <em>VCF graph</em> below) was created from the linear reference genome of the <em>S.c. S288C</em> strain and a set of SVs relative to this reference strain in VCF format identified from the other four assemblies by three methods: Assemblytics <span class="citation" data-cites="krO7WgVi">[<a href="#ref-krO7WgVi">26</a>]</span>, AsmVar <span class="citation" data-cites="oVaXIwl5">[<a href="#ref-oVaXIwl5">27</a>]</span> and paftools <span class="citation" data-cites="172cJaw4Q">[<a href="#ref-172cJaw4Q">28</a>]</span>. The second graph (called <em>cactus graph</em> below) was derived from a multiple genome alignment of the five strains using Cactus <span class="citation" data-cites="1FgS53pXi">[<a href="#ref-1FgS53pXi">24</a>]</span>. The <em>VCF graph</em> is mostly linear and highly dependent on the reference genome. In contrast, the <em>cactus graph</em> is structurally complex and relatively free of reference bias.</p>
<p>First, we tested our hypothesis that the <em>cactus graph</em> has higher mappability due to its better representation of sequence diversity among the yeast strains (see <a href="#mappability-comparison-between-yeast-graphs">Supplementary Information</a>). Generally, more reads mapped to the <em>cactus graph</em> with high identity (Figure <a href="#fig:panel3">S12</a>a) and high mapping quality (Figure <a href="#fig:panel3">S12</a>b) than to the <em>VCF graph</em>.</p>
<p>Next, we compared the SV genotyping performance of both graphs. We mapped short reads from the 11 non-reference strains to both graphs and called variants for each strain using the vg toolkit’s variant calling module (see <a href="#toil-vg-call">Methods</a>). There is no gold standard call set for these samples, so we used an indirect measure of SV calling accuracy. We evaluated each call set based on the alignment of reads to a <em>sample graph</em> constructed from the call set (see <a href="#calling-and-genotyping-of-svs">Methods</a>). If a given call set is correct, we expect that reads from the same sample will be mapped with high identity and confidence to the corresponding sample graph. Therefore, we compared the average percent identity and mapping quality of the short reads on each sample graph (Figures <a href="#fig:4">5</a>a and b). Similar to the mappability results, the <em>cactus graph</em> clearly outperformed the <em>VCF graph</em> for strains in the <em>S. paradoxus</em> clade and performed slightly better for strains in the <em>S. cerevisiae</em> clade. While the higher percent identity shows that the <em>cactus graph</em> represents the reads better (Figures <a href="#fig:4">5</a>a), the higher mapping quality confirms that this did not come at the cost of added ambiguity or a more complex graph (Figures <a href="#fig:4">5</a>b). Our results did not show a substantial difference between strains included in the graph and those that were excluded. This suggests that two strains from each clade as well as the reference strain are sufficient to capture most of the genetic variation among all the strains. For a direct comparison, see Figure <a href="#fig:panel6">S14</a> which shows results of the same experiment on graphs generated from all 12 strains.</p>
<figure>
<img src="images/panel4.png" alt="Figure 5: SV genotyping comparison. Short reads from all 11 non-reference yeast strains were used to genotype SVs contained in both graphs. Subsequently, sample graphs were generated from the resulting SV callsets. The short reads were aligned to the sample graphs and the quality of the alignments was used to ascertain genotyping performance. More accurate genotypes should result in reference graphs that have mappings with high identity and confidence for a greater proportion of the reads. a) Average mapping identity of short reads aligned to the sample graphs derived from cactus graph (y-axis) and VCF graph (x-axis). b) Average mapping quality of short reads aligned to the sample graphs derived from cactus graph (y-axis) and VCF graph (x-axis). Colors and shapes represent the 11 non-reference strains and two clades, respectively. Transparency indicates whether the strain was used to construct the graphs." id="fig:4" /><figcaption><span>Figure 5:</span> <strong>SV genotyping comparison.</strong> Short reads from all 11 non-reference yeast strains were used to genotype SVs contained in both graphs. Subsequently, sample graphs were generated from the resulting SV callsets. The short reads were aligned to the sample graphs and the quality of the alignments was used to ascertain genotyping performance. More accurate genotypes should result in reference graphs that have mappings with high identity and confidence for a greater proportion of the reads. a) Average mapping identity of short reads aligned to the sample graphs derived from <em>cactus graph</em> (y-axis) and <em>VCF graph</em> (x-axis). b) Average mapping quality of short reads aligned to the sample graphs derived from <em>cactus graph</em> (y-axis) and <em>VCF graph</em> (x-axis). Colors and shapes represent the 11 non-reference strains and two clades, respectively. Transparency indicates whether the strain was used to construct the graphs.</figcaption>
</figure>
<h2 id="discussion">Discussion</h2>
<!-- Discuss why vg is doing better -->
<p>Overall, vg was the most accurate SV genotyper in our benchmarks. These results show that variant calling benefits from variant-aware read mapping and graph based genotyping, a finding consistent with previous studies<span class="citation" data-cites="10jxt15v0 DuODeStx 11Jy8B61m ohTIiqfV 14Uxmwbxm">[<a href="#ref-10jxt15v0">15</a>,<a href="#ref-DuODeStx">16</a>,<a href="#ref-11Jy8B61m">17</a>,<a href="#ref-ohTIiqfV">18</a>,<a href="#ref-14Uxmwbxm">19</a>]</span>. We took advantage of newly released datasets for our evaluation, which feature up to 3.7 times more variants than the more widely-used GIAB benchmark. More and more large-scale projects are using low cost short-read technologies to sequence the genomes of thousands to hundreds of thousands of individuals (e.g. the Pancancer Analysis of Whole Genomes<span class="citation" data-cites="10Jid8Wql">[<a href="#ref-10Jid8Wql">29</a>]</span>, the Genomics England initiative<span class="citation" data-cites="mWj2p7Xp">[<a href="#ref-mWj2p7Xp">30</a>]</span>, and the TOPMed consortium<span class="citation" data-cites="ir1O1h8n">[<a href="#ref-ir1O1h8n">31</a>]</span>). We believe pangenome graph-based approaches will improve both how efficiently SVs can be represented, and how accurately they can be genotyped with this type of data.</p>
<!-- Input data quality: "sequence-resolved", break-point fine-tuning. -->
<p>A particular advantage of our method is that it does not require exact breakpoint resolution in the variant library. Our simulations showed that vg’s SV genotyping algorithm is robust to errors of as much as 10 bp in breakpoint location. However, there is an upper limit to this flexibility, and we find that vg cannot accurately genotype variants with much higher uncertainty in the breakpoint location (like those discovered through read coverage analysis). vg is also capable of fine-tuning SV breakpoints by augmenting the graph with differences observed in read alignments. Simulations showed that this approach can usually correct small errors in SV breakpoints (Figure <a href="#fig:simerror-bkpt">S11</a> and Table <a href="#tbl:simerror-bkpt">S6</a>).</p>
<!-- Already superior but will only get better with new vg dev -->
<p>vg uses a unified framework to call and score different variant types simultaneously. In this work, we only considered graphs containing certain types of SVs, but the same methods can be extended to a broader range of graphs. For example, we are interested in evaluating how genotyping SVs together with SNPs and small indels using a combined graph effects the accuracy of studying either alone. The same methods used for genotyping known variants in this work can also be extended to call novel variants by first augmenting the graph with edits from the mapped reads. This approach, which was used only in the breakpoint fine-tuning portion of this work, could be further used to study small variants around and nested within SVs. Novel SVs could be called by augmenting the graph with long-read mappings. vg is entirely open source, and its ongoing development is supported by a growing community of researchers and users with common interest in scalable, unbiased pangenomic analyses and representation. We expect this collaboration to continue to foster increases in the speed, accuracy and applicability of methods based on pangenome graphs in the years ahead.</p>
<!-- Benefits of *de novo* assemblies -->
<p>Our results suggest that constructing a graph from <em>de novo</em> assembly alignment instead of a VCF leads to better SV genotyping. High quality <em>de novo</em> assemblies for human are becoming more and more common due to improvements in technologies like optimized mate-pair libraries<span class="citation" data-cites="pJAv1D8R">[<a href="#ref-pJAv1D8R">32</a>]</span> and long-read sequencing<span class="citation" data-cites="6KbgcueR">[<a href="#ref-6KbgcueR">12</a>]</span>. We expect future graphs to be built from the alignment of numerous <em>de novo</em> assemblies, and we are presently working on scaling our assembly-based pipeline to human-sized genome assemblies. Another challenge is creating genome graphs that integrate assemblies with variant-based data resources. One possible approach is to progressively align assembled contigs into variation graphs constructed from variant libraries, but methods for doing so are still experimental.</p>
<h2 id="conclusion">Conclusion</h2>
<p>In this study, the vg toolkit was compared to existing SV genotypers across several high-quality SV catalogs. We showed that its method of mapping reads to a variation graph leads to better SV genotyping compared to other state of the art methods. This work introduces a flexible strategy to integrate the growing number of SVs being discovered with higher resolution technologies into a unified framework for genome inference. Our work on whole genome alignment graphs shows the benefit of directly utilizing <em>de novo</em> assemblies rather than variant catalogs to integrate SVs in genome graphs. We expect this latter approach to increase in significance as the reduction in long read sequencing costs drives the creation of numerous new <em>de novo</em> assemblies. We envision a future in which the lines between variant calling, alignment, and assembly are blurred by rapid changes in sequencing technology. Fully graph based approaches, like the one we present here, will be of great utility in this new phase of genome inference.</p>
<h2 id="methods">Methods</h2>
<h3 id="the-vg-call-genotyping-algorithm">The vg call genotyping algorithm</h3>
<p>In <code>vg call</code>, we implemented a simple variant caller capable of operating on any kind of variation that can be represented in variation graphs. The algorithm uses a genome graph as a structuring prior, and consumes read mappings to the graph to drive the inference of the true genomic state at each locus. Here, we apply <code>vg call</code> to genotype structural variants already present in the graph, but the same algorithm can also be used for smaller variants such as SNPs, as well as making de-novo calls. The algorithm, illustrated in Figure <a href="#fig:1">1</a>a, proceeds through three main phases:</p>
<ol type="1">
<li>We compute the average read support for each node and edge, adjusted for mapping and base quality. The graph can optionally be augmented to include new variation from the reads using a minimum support cutoff.</li>
<li>We then decomposed the graph into snarls<span class="citation" data-cites="xJlNnKH2">[<a href="#ref-xJlNnKH2">23</a>]</span>. Briefly, a snarl is a subgraph defined by two end nodes, where cutting the graph at these nodes disconnects the subgraph from the rest of the graph. Snarls can be nested inside other snarls, and this nesting hierarchy forms a forest. As proposed in Paten et al.<span class="citation" data-cites="xJlNnKH2">[<a href="#ref-xJlNnKH2">23</a>]</span>, we use the snarl decomposition as a structure for identifying variants in a graph.</li>
<li>In parallel, we independently consider root-level snarls from the decomposition. Only snarls whose two ends lie on a reference (i.e. chromosome) path are considered as the VCF format used for output requires definite reference positions. For each root snarl, we:
<ol type="1">
<li>Compute a set of paths between the snarl boundary nodes using a heuristic search that enumerates paths until all nodes and edges in the snarl are contained in at least one path.</li>
<li>Rank these paths according to their average support from the reads.</li>
<li>Determine a genotype using the relative support of the best paths, as well as the background read depth. The same logic is used for all types of variation, each of which can be expressed simply as a path in the graph.</li>
<li>Project variants in the graph into VCF format.</li>
</ol></li>
</ol>
<h3 id="toil-vg">toil-vg</h3>
<p>toil-vg is a set of Python scripts for simplifying vg tasks such as graph construction, read mapping and SV genotyping. It uses the Toil workflow engine <span class="citation" data-cites="faeC2cx0">[<a href="#ref-faeC2cx0">33</a>]</span> to seamlessly run pipelines locally, on clusters, or on the cloud. All variation graph analyses in this report used toil-vg, with the exact commands available at <a href="https://github.com/vgteam/sv-genotyping-paper">github.com/vgteam/sv-genotyping-paper</a>. The principal toil-vg commands used are described below.</p>
<h4 id="toil-vg-construct">toil-vg construct</h4>
<p>toil-vg construct automates graph construction and indexing following the best practices put forth by the vg community. It parallelizes graph construction across different sequences from the reference FASTA, and creates different whole-genome indexes side by side when possible. When available, toil-vg construct can use phasing information from the input VCF to preserve haplotypes in the GCSA2 pruning step, as well as to extract haploid sequences to simulate from.</p>
<h4 id="toil-vg-map">toil-vg map</h4>
<p>toil-vg map splits the input reads into batches, maps each batch in parallel, and merges the result.</p>
<h4 id="toil-vg-call">toil-vg call</h4>
<p>Due to the high memory requirements of the current implementation of vg call, toil-vg call splits the input graph into 2.5Mb overlapping chunks along the reference path. Each chunk is called independently in parallel and the results are concatenated into the output VCF.</p>
<h4 id="toil-vg-sveval">toil-vg sveval</h4>
<p>toil-vg seval evaluates the SV calls relative to a truth set. The variants are first normalized with <code>bcftools norm</code> (1.9) to ensure consistent representation between called variants and baseline variants<span class="citation" data-cites="LAG2q9WK">[<a href="#ref-LAG2q9WK">34</a>]</span>. We then implemented an overlap-based strategy to compare SVs and compute evaluation metrics (sveval R package: <a href="https://github.com/jmonlong/sveval">https://github.com/jmonlong/sveval</a>). Figure <a href="#fig:sveval">S15</a> shows an overview of the SV evaluation approach which is described below.</p>
<p>For deletions and inversions, we begin by computing the overlaps between the SVs in the call set and the truth set. For each variant we then compute the proportion of its region that is covered by a variant in the other set, considering only variants overlapping with at least 10% reciprocal overlap. If this coverage proportion is higher than 50%, we consider the variant <em>covered</em>. True positives (TPs) are covered variants from the call set (when computing the precision) or the truth set (when computing the recall). Variants from the call set are considered false positives (FPs) if they are not covered by the truth set. Conversely, variants from the truth set are considered false negatives (FNs) if they are not covered by the call set.</p>
<p>For insertions, we select pairs of insertions that are located no farther than 20 bp from each other. We then align the inserted sequences using a Smith-Waterman alignment. For each insertion we compute the proportion of its inserted sequence that aligns a matched variant in the other set. If this proportion is at least 50% the insertions are considered covered. Covering relationships are used to define TPs, FPs, and FNs the same way as for deletions and inversions.</p>
<p>The coverage statistics are computed using any variant larger than 1 bp but a minimum size is required for a variant to be counted as TP, FP, or FN. In this work, we used the default minimum SV size of 50 bp.</p>
<p>sveval accepts VCF files with symbolic or explicit representation of the SVs. If the explicit representation is used, multi-allelic variants are split and their sequences right-trimmed. When using the explicit representation and when the REF and ALT sequences are longer than 10 bp, the reverse-complement of the ALT sequence is aligned to the REF sequence to identify potential inversions. If more than 80% of the sequence aligns, it is classified as an inversion.</p>
<p>We assess both the ability to predict the presence of an SV as well as the full genotype. For the <em>presence</em> evaluation, both heterozygous and homozygous alternate SVs are compared jointly using the approach described above. To compute genotype-level metrics, the heterozygous and homozygous SVs are compared separately. Before splitting the variants by genotype, consecutive heterozygous variants are first merged if located at less that 20 bp from each other. Pairs of heterozygous variants with reciprocal overlap of at least 80% are also merged into a homozygous ALT variant before splitting variants by genotype.</p>
<h3 id="other-sv-genotypers">Other SV genotypers</h3>
<h4 id="bayestyper-v1.5-beta-62888d6">BayesTyper (v1.5 beta 62888d6)</h4>
<p>Where not specified otherwise BayesTyper was run as follows. Raw reads were mapped to the reference genome using <code>bwa mem</code> (0.7.17). GATK<span class="citation" data-cites="NCr4QkOg">[<a href="#ref-NCr4QkOg">35</a>]</span> (3.8) and Platypus<span class="citation" data-cites="1DYmkalz4">[<a href="#ref-1DYmkalz4">36</a>]</span> (0.8.1.1) were run on the mapped reads to call SNVs and short indels (&lt;50bp) needed by BayesTyper for correct genotyping. The VCFs with these variants were then normalised using <code>bcftools norm</code> (1.9) and combined with the SVs across samples using <code>bayesTyperTools combine</code> to produce the input candidate set. k-mers in the raw reads were counted using <code>kmc</code> (3.1.1) with a k-mer size of 55. A Bloom filter was constructed from these k-mers using <code>bayesTyperTools makeBloom</code>. Finally, variants were clustered and genotyped using <code>bayestyper cluster</code> and <code>bayestyper genotype</code>, respectively, with default parameters except <code>--min-genotype-posterior 0</code>. Non-PASS variants were filtered prior to evaluation using <code>bcftools filter</code>.</p>
<h4 id="delly-v0.7.9">Delly (v0.7.9)</h4>
<p>The <code>delly call</code> command was run on the reads mapped by <code>bwa mem</code>, the reference genome FASTA file, and the VCF containing the SVs to genotype (converted to their explicit representations).</p>
<h4 id="svtyper-v0.7.0">SVTyper (v0.7.0)</h4>
<p>The VCF containing deletions was converted to symbolic representation and passed to <code>svtyper</code> with the reads mapped by <code>bwa mem</code>. The output VCF was converted back to explicit representation using <code>bayesTyperTools convertAllele</code> to facilitate variant normalization before evaluation.</p>
<h4 id="smrt-sv-v2-v2.0.0-feb-21-2019-commit-adb13f2">SMRT-SV v2 (v2.0.0 Feb 21 2019 commit adb13f2)</h4>
<p>SMRT-SV v2 was run with the “30x-4” model and min-call-depth 8 cutoff. It was run only on VCFs created by SMRT-SV, for which the required contig BAMs were available. The Illumina BAMs used where the same as the other methods described above. The output VCF was converted back to explicit representation to facilitate variant normalization later.</p>
<h3 id="simulation-experiment">Simulation experiment</h3>
<p>We simulated a synthetic genome with 1000 insertions, deletions and inversions. We separated each variant from the next by a buffer of at least 500 bp. The sizes of deletions and insertions followed the distribution of SV sizes from the HGSVC catalog. We used the same size distribution as deletions for inversions. A VCF file was produced for three simulated samples with genotypes chosen uniformly between homozygous reference, heterozygous, and homozygous alternate.</p>
<p>We created another VCF file containing errors in the SV breakpoint locations. We shifted one or both breakpoints of deletions and inversions by distances between 1 and 10 bp. The locations and sequences of insertions were also modified, either shifting the variants or shortening them at the flanks, again by up to 10 bp.</p>
<p>Paired-end reads were simulated using <code>vg sim</code> on the graph that contained the true SVs. Different read depths were tested: 1x, 3x, 7x, 10x, 13x, 20x. The base qualities and sequencing errors were trained to resemble real Illumina reads from NA12878 provided by the Genome in a Bottle Consortium.</p>
<p>The genotypes called in each experiment (genotyping method/VCF with or without errors/sequencing depth) were compared to the true SV genotypes to compute the precision, recall and F1 score (see <a href="#toil-vg-sveval">toil-vg sveval</a>).</p>
<h4 id="breakpoint-fine-tuning-using-graph-augmentation">Breakpoint fine-tuning using graph augmentation</h4>
<p>vg can call variants after augmenting the graph with the read alignments to discover new variants (see <a href="#toil-vg-call">toil-vg call</a>). We tested if this approach could fine-tune the breakpoint location of SVs in the graph. We started with the graph that contained approximate SVs (1-10 bp errors in breakpoint location) and 20x simulated reads from the simulation experiment (see <a href="#simulation-experiment">Simulation experiment</a>). The variants called after graph augmentation were compared with the true SVs. We considered fine-tuning correct if the breakpoints matched exactly.</p>
<h3 id="hgsvc-analysis">HGSVC Analysis</h3>
<p>We first obtained phased VCFs for the three Human Genome Structural Variation Consortium (HGSVC) samples from Chaisson et al.<span class="citation" data-cites="vQTymKCj">[<a href="#ref-vQTymKCj">20</a>]</span> and combined them with <code>bcftools merge</code>. A variation graph was created and indexed using the combined VCF and the HS38D1 reference with alt loci excluded. The phasing information was used to construct a GBWT index<span class="citation" data-cites="jj4JJlsg">[<a href="#ref-jj4JJlsg">37</a>]</span>, from which the two haploid sequences from HG00514 were extracted as a graph. Illumina read pairs with 30x coverage were simulated from these sequences using vg sim, with an error model learned from real reads from the same sample. These simulated reads reflect an idealized situation where the breakpoints of the SVs being genotyped are exactly known <em>a priori</em>. The reads were mapped to the graph, and the mappings used to genotype the SVs in the graph. Finally, the SV calls were compared back to the HG00514 genotypes from the HGSVC VCF. We repeated the process with the same reads on the linear reference, using <code>bwa mem</code> for mapping and Delly, SVTyper and BayesTyper for SV genotyping.</p>
<p>We downloaded Illumina HiSeq 2500 paired end reads from the EBI’s ENA FTP site for the three samples, using Run Accessions ERR903030, ERR895347 and ERR894724 for HG00514, HG00733 and NA19240, respectively. We ran the graph and linear mapping and genotyping pipelines exactly as for the simulation, and aggregated the comparison results across the three samples. We used BayesTyper to jointly genotype the 3 samples.</p>
<h3 id="giab-analysis">GIAB Analysis</h3>
<p>We obtained version 0.5 of the Genome in a Bottle (GIAB) SV VCF for the Ashkenazim son (HG002) and his parents from the NCBI FTP site. We obtained Illumina reads as described in Garrison et al.<span class="citation" data-cites="10jxt15v0">[<a href="#ref-10jxt15v0">15</a>]</span> and downsampled them to 50x coverage. We used these reads as input for <code>vg call</code> and the other SV genotyping pipelines described above (though with GRCh37 instead of GRCh38). For BayesTyper, we created the input variant set by combining the GIAB SVs with SNV and indels from the same study. Variants with reference allele or without a determined genotype for HG002 in the GIAB call set (10,569 out of 30,224) were considered “false positives” as a proxy measure for precision. These variants correspond to putative technical artifacts and parental calls not present in HG002.</p>
<h3 id="smrt-sv-v2-comparison-chmpd-and-svpop">SMRT-SV v2 Comparison (CHMPD and SVPOP)</h3>
<p>The SMRT-SV v2 genotyper can only be used to genotype sequence-resolved SVs present on contigs with known SV breakpoints, such as those created by SMRT-SV v2, and therefore could not be run on the simulated, HGSVC, or GIAB call sets. The authors shared their training and evaluation set: a pseudodiploid sample constructed from combining the haploid CHM1 and CHM13 samples (CHMPD), and a negative control (NA19240). The high quality of the CHM assemblies makes this set an attractive alternative to using simulated reads. We used this two-sample pseudodiploid VCF along with the 30X read set to construct, map and genotype with vg, and also ran SMRT-SV v2 genotyper with the “30x-4” model and min-call-depth 8 cutoff, and compared the two back to the original VCF.</p>
<p>In an effort to extend this comparison from the training data to a more realistic setting, we reran the three HGSVC samples against the SMRT-SV v2 discovery VCF (SVPOP, which contains 12 additional samples in addition to the three from HGSVC) published by Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span> using vg and SMRT-SV v2 Genotyper. The discovery VCF does not contain genotypes. In consequence, we were unable to distinguish between heterozygous and homozygous genotypes, and instead considered only the presence or absence of a non-reference allele for each variant.</p>
<p>SMRT-SV v2 produces explicit <em>no-call</em> predictions when the read coverage is too low to produce accurate genotypes. These no-calls are considered homozygous reference in the main accuracy evaluation. We also explored the performance of vg and SMRT-SV v2 in different sets of regions (Figure <a href="#fig:svpop-regions">S10</a> and Table <a href="#tbl:svpop-regions">S5</a>):</p>
<ol type="1">
<li>Non-repeat regions, i.e. excluding segmental duplications and tandem repeats (using the respective tracks from the UCSC Genome Browser).</li>
<li>Repeat regions defined as segmental duplications and tandem repeats.</li>
<li>Regions where SMRT-SV v2 could call variants.</li>
<li>Regions where SMRT-SV v2 produced no-calls.</li>
</ol>
<h3 id="yeast-graph-analysis">Yeast graph analysis</h3>
<p>For the analysis of graphs from <em>de novo</em> assemblies, we utilized publicly available PacBio-derived assemblies and Illumina short read sequencing datasets for 12 yeast strains from two related clades (Table <a href="#tbl:strains">1</a>) <span class="citation" data-cites="7f5OKa5O">[<a href="#ref-7f5OKa5O">25</a>]</span>. We selected five strains for graph contruction (two from different subclades of each clade plus the reference <em>S.c. S288C</em>): <em>S.c. SK1</em>, <em>S.c. YPS128</em>, <em>S.p. CBS432</em>, <em>S.p. UFRJ50816</em>, and <em>S.c. S288C</em>. We constructed two different genome graphs from the assemblies of the five selected strains. In this section, we describe the steps for the construction of both graphs and the calling of variants. More details and the precise commands used in our analyses can be found at <a href="https://github.com/vgteam/sv-genotyping-paper">github.com/vgteam/sv-genotyping-paper</a>.</p>
<a name="tbl:strains"></a>
<table>
<caption><span>Table 1:</span> 12 yeast strains from two related clades were used in our analysis. Five strains were selected to be included in the graphs while the remaining seven were used for variant calling only. </caption>
<thead>
<tr class="header">
<th>Strain</th>
<th>Clade</th>
<th>Included in graph</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>S288C</td>
<td>S. cerevisiae</td>
<td>✓</td>
</tr>
<tr class="even">
<td>SK1</td>
<td>S. cerevisiae</td>
<td>✓</td>
</tr>
<tr class="odd">
<td>YPS128</td>
<td>S. cerevisiae</td>
<td>✓</td>
</tr>
<tr class="even">
<td>UWOPS034614</td>
<td>S. cerevisiae</td>
<td></td>
</tr>
<tr class="odd">
<td>Y12</td>
<td>S. cerevisiae</td>
<td></td>
</tr>
<tr class="even">
<td>DBVPG6765</td>
<td>S. cerevisiae</td>
<td></td>
</tr>
<tr class="odd">
<td>DBVPG6044</td>
<td>S. cerevisiae</td>
<td></td>
</tr>
<tr class="even">
<td>CBS432</td>
<td>S. paradoxus</td>
<td>✓</td>
</tr>
<tr class="odd">
<td>UFRJ50816</td>
<td>S. paradoxus</td>
<td>✓</td>
</tr>
<tr class="even">
<td>N44</td>
<td>S. paradoxus</td>
<td></td>
</tr>
<tr class="odd">
<td>UWOPS919171</td>
<td>S. paradoxus</td>
<td></td>
</tr>
<tr class="even">
<td>YPS138</td>
<td>S. paradoxus</td>
<td></td>
</tr>
</tbody>
</table>
<h4 id="construction-of-the-vcf-graph">Construction of the <em>VCF graph</em></h4>
<p>We constructed the first graph (called the <em>VCF graph</em> throughout the paper) by adding variants onto a linear reference. This method requires one assembly to serve as a reference genome. The other assemblies must be converted to variant calls relative to this reference. The PacBio assembly of the S.c. S288C strain was chosen as the reference genome because this strain was used for the S. cerevisiae genome reference assembly. To obtain variants for the other assemblies, we combined three methods for SV detection from genome assemblies: Assemblytics <span class="citation" data-cites="krO7WgVi">[<a href="#ref-krO7WgVi">26</a>]</span> (commit df5361f), AsmVar (commit 5abd91a) <span class="citation" data-cites="oVaXIwl5">[<a href="#ref-oVaXIwl5">27</a>]</span> and paftools (version 2.14-r883) <span class="citation" data-cites="172cJaw4Q">[<a href="#ref-172cJaw4Q">28</a>]</span>. We constructed a union set of variants detected by the three methods (using bedtools <span class="citation" data-cites="1HWiAHnIw">[<a href="#ref-1HWiAHnIw">38</a>]</span>), and combined variants with a reciprocal overlap of at least 50% to avoid duplication in the union set. We merged these union sets of variants for each of the four selected (and non-reference) strains, and we then applied another deduplication step to combine variants with a reciprocal overlap of at least 90%. We then used <code>vg construct</code> to build the <em>VCF graph</em> with the total set of variants and the linear reference genome.</p>
<h4 id="construction-of-the-cactus-graph">Construction of the <em>cactus graph</em></h4>
<p>The second graph (called the <em>cactus graph</em> throughout the paper) was constructed from a whole genome alignment between the assemblies. First, the repeat-masked PacBio-assemblies of the five selected strains were aligned with our Cactus tool <span class="citation" data-cites="1FgS53pXi">[<a href="#ref-1FgS53pXi">24</a>]</span>. Cactus requires a phylogenetic tree of the strains which was estimated using Mash (version 2.1) <span class="citation" data-cites="mH9pzoIn">[<a href="#ref-mH9pzoIn">39</a>]</span> and PHYLIP (version 3.695) <span class="citation" data-cites="tIvRXd6o">[<a href="#ref-tIvRXd6o">40</a>]</span>. Subsequently, we converted the HAL format output file to a variation graph with hal2vg (<a href="https://github.com/ComparativeGenomicsToolkit/hal2vg">https://github.com/ComparativeGenomicsToolkit/hal2vg</a>).</p>
<h4 id="calling-and-genotyping-of-svs">Calling and genotyping of SVs</h4>
<p>Prior to variant calling, we mapped the Illumina short reads of all 12 yeast strains to both graphs using <code>vg map</code>. We measured the fractions of reads mapped with specific properties using <code>vg view</code> and the JSON processor <code>jq</code>. Then, we applied <code>toil-vg call</code> (commit be8b6da) to call variants, obtaining a separate variant call set for each of the strains on both graphs. To evaluate the callsets, we generated a sample graph (i.e. a graph representation of the callset) for each callset using <code>vg construct</code> and <code>vg mod</code> on the reference assembly <em>S.c. S288C</em> and the callset. Subsequently, we mapped short reads from the respective strains to each sample graph using <code>vg map</code>. We analyzed the resulting alignments with <code>vg view</code> and <code>jq</code>.</p>
<h2 id="declarations">Declarations</h2>
<h3 id="availability-of-data-and-material">Availability of data and material</h3>
<p>The commands used to run the analyses presented in this study are available at <a href="https://github.com/vgteam/sv-genotyping-paper">github.com/vgteam/sv-genotyping-paper</a> (ZENODO DOI WHEN READY). The datasets generated and/or analysed during the current study are also listed in this repository.</p>
<p>The scripts to generate the manuscript, including figures and tables, are available at <a href="https://github.com/jmonlong/manu-vgsv">github.com/jmonlong/manu-vgsv</a>.</p>
<h3 id="competing-interests">Competing interests</h3>
<p>The authors declare that they have no competing interests.</p>
<h3 id="funding">Funding</h3>
<p>Research reported in this publication was supported by the National Human Genome Research Institute of the National Institutes of Health under Award Number U54HG007990 and U01HL137183. This publication was supported by a Subagreement from European Molecular Biology Laboratory with funds provided by Agreement No. 2U41HG007234 from National Institute of Health, NHGRI. Its contents are solely the responsibility of the authors and do not necessarily represent the official views of National Institute of Health, NHGRI or European Molecular Biology Laboratory. The research was made possible by the generous financial support of the W.M. Keck Foundation (DT06172015).</p>
<p>DH was supported by the International Max Planck Research School for Computational Biology and Scientific Computing doctoral program. JE was supported by the Jack Baskin and Peggy Downes-Baskin Fellowship. AMN was supported by the National Institutes of Health (5U41HG007234), the W.M. Keck Foundation (DT06172015) and the Simons Foundation (SFLIFE# 35190). JAS was supported by the Carlsberg Foundation (CF17-0881).</p>
<h3 id="authors-contributions">Authors’ contributions</h3>
<p>EG, AN, GH, JS, JE and ED implemented the read mapping and variant calling in the vg toolkit. GH, DH, JM, JAS and EG performed analysis on the different datasets. GH, DH, JM and BP designed the study. GH, DH and JM drafted the manuscript. All authors read, reviewed, and approved the final manuscript.</p>
<h3 id="acknowledgements">Acknowledgements</h3>
<p>We thank Peter Audano for sharing the CHMPD dataset and for his assistance with SMRT-SV v2.</p>
<h3 id="authors-information">Authors’ information</h3>
<p>These authors contributed equally: Glenn Hickey, David Heller, Jean Monlong.</p>
<h2 id="supplementary-material" class="page_break_before">Supplementary Material</h2>
<h3 id="supplementary-tables">Supplementary Tables</h3>
<a name="tbl:hgsvc"></a>
<table>
<caption><span>Table S1:</span> Genotyping evaluation on the HGSVC dataset. Precision, recall and F1 score for the call set with the best F1 score. The numbers in parentheses corresponds to the results in non-repeat regions. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Experiment</th>
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Type</th>
<th style="text-align: left;">Precision</th>
<th style="text-align: left;">Recall</th>
<th style="text-align: left;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Simulated reads</td>
<td style="text-align: left;">vg</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.795 (0.885)</td>
<td style="text-align: left;">0.796 (0.883)</td>
<td style="text-align: left;">0.795 (0.884)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.869 (0.971)</td>
<td style="text-align: left;">0.771 (0.92)</td>
<td style="text-align: left;">0.817 (0.945)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">BayesTyper</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.91 (0.935)</td>
<td style="text-align: left;">0.835 (0.9)</td>
<td style="text-align: left;">0.871 (0.917)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.898 (0.981)</td>
<td style="text-align: left;">0.806 (0.929)</td>
<td style="text-align: left;">0.849 (0.954)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">SVTyper</td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.809 (0.876)</td>
<td style="text-align: left;">0.328 (0.754)</td>
<td style="text-align: left;">0.467 (0.81)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">Delly</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.767 (0.866)</td>
<td style="text-align: left;">0.093 (0.225)</td>
<td style="text-align: left;">0.166 (0.358)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.696 (0.903)</td>
<td style="text-align: left;">0.707 (0.846)</td>
<td style="text-align: left;">0.701 (0.874)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Real reads</td>
<td style="text-align: left;">vg</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.431 (0.683)</td>
<td style="text-align: left;">0.541 (0.726)</td>
<td style="text-align: left;">0.48 (0.704)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.65 (0.886)</td>
<td style="text-align: left;">0.519 (0.708)</td>
<td style="text-align: left;">0.577 (0.787)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">BayesTyper</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.601 (0.747)</td>
<td style="text-align: left;">0.254 (0.433)</td>
<td style="text-align: left;">0.357 (0.549)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.627 (0.91)</td>
<td style="text-align: left;">0.325 (0.381)</td>
<td style="text-align: left;">0.428 (0.537)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">SVTyper</td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.661 (0.733)</td>
<td style="text-align: left;">0.236 (0.551)</td>
<td style="text-align: left;">0.348 (0.629)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">Delly</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.516 (0.621)</td>
<td style="text-align: left;">0.068 (0.176)</td>
<td style="text-align: left;">0.12 (0.275)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.55 (0.838)</td>
<td style="text-align: left;">0.445 (0.547)</td>
<td style="text-align: left;">0.492 (0.662)</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:giab"></a>
<table>
<caption><span>Table S2:</span> Genotyping evaluation on the Genome in a Bottle dataset. Precision, recall and F1 score for the call set with the best F1 score. The numbers in parentheses corresponds to the results in non-repeat regions. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Type</th>
<th style="text-align: left;">Precision</th>
<th style="text-align: left;">Recall</th>
<th style="text-align: left;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">vg</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.658 (0.774)</td>
<td style="text-align: left;">0.646 (0.735)</td>
<td style="text-align: left;">0.652 (0.754)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.68 (0.768)</td>
<td style="text-align: left;">0.643 (0.735)</td>
<td style="text-align: left;">0.661 (0.751)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BayesTyper</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.776 (0.879)</td>
<td style="text-align: left;">0.286 (0.379)</td>
<td style="text-align: left;">0.418 (0.53)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.808 (0.886)</td>
<td style="text-align: left;">0.512 (0.696)</td>
<td style="text-align: left;">0.627 (0.779)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SVTyper</td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.742 (0.818)</td>
<td style="text-align: left;">0.342 (0.496)</td>
<td style="text-align: left;">0.468 (0.618)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Delly</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.822 (0.894)</td>
<td style="text-align: left;">0.177 (0.268)</td>
<td style="text-align: left;">0.291 (0.412)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.722 (0.822)</td>
<td style="text-align: left;">0.645 (0.768)</td>
<td style="text-align: left;">0.681 (0.794)</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:chmpd"></a>
<table>
<caption><span>Table S3:</span> Genotyping evaluation on the pseudo-diploid genome built from CHM cell lines in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Region</th>
<th style="text-align: left;">Type</th>
<th style="text-align: right;">Precision</th>
<th style="text-align: right;">Recall</th>
<th style="text-align: right;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">vg</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">0.665</td>
<td style="text-align: right;">0.661</td>
<td style="text-align: right;">0.663</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">0.688</td>
<td style="text-align: right;">0.500</td>
<td style="text-align: right;">0.579</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">0.806</td>
<td style="text-align: right;">0.784</td>
<td style="text-align: right;">0.795</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">0.869</td>
<td style="text-align: right;">0.762</td>
<td style="text-align: right;">0.812</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SMRT-SV</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">0.757</td>
<td style="text-align: right;">0.536</td>
<td style="text-align: right;">0.628</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">0.848</td>
<td style="text-align: right;">0.630</td>
<td style="text-align: right;">0.723</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">0.880</td>
<td style="text-align: right;">0.680</td>
<td style="text-align: right;">0.767</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">0.971</td>
<td style="text-align: right;">0.824</td>
<td style="text-align: right;">0.891</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:svpop"></a>
<table>
<caption><span>Table S4:</span> Calling evaluation on the SVPOP dataset. Combined results for the HG00514, HG00733 and NA19240 individuals, 3 of the 15 individuals used to generate the high-quality SV catalog in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span>. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Region</th>
<th style="text-align: left;">Type</th>
<th style="text-align: right;">TP</th>
<th style="text-align: right;">FP</th>
<th style="text-align: right;">FN</th>
<th style="text-align: right;">Precision</th>
<th style="text-align: right;">Recall</th>
<th style="text-align: right;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">vg</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">25838</td>
<td style="text-align: right;">22042</td>
<td style="text-align: right;">15772</td>
<td style="text-align: right;">0.540</td>
<td style="text-align: right;">0.621</td>
<td style="text-align: right;">0.577</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">14545</td>
<td style="text-align: right;">6824</td>
<td style="text-align: right;">15425</td>
<td style="text-align: right;">0.681</td>
<td style="text-align: right;">0.485</td>
<td style="text-align: right;">0.567</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">173</td>
<td style="text-align: right;">0.509</td>
<td style="text-align: right;">0.135</td>
<td style="text-align: right;">0.213</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">8051</td>
<td style="text-align: right;">3258</td>
<td style="text-align: right;">1817</td>
<td style="text-align: right;">0.712</td>
<td style="text-align: right;">0.816</td>
<td style="text-align: right;">0.760</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">3769</td>
<td style="text-align: right;">623</td>
<td style="text-align: right;">818</td>
<td style="text-align: right;">0.858</td>
<td style="text-align: right;">0.822</td>
<td style="text-align: right;">0.840</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">75</td>
<td style="text-align: right;">0.613</td>
<td style="text-align: right;">0.202</td>
<td style="text-align: right;">0.304</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SMRT-SV</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">16270</td>
<td style="text-align: right;">26031</td>
<td style="text-align: right;">25340</td>
<td style="text-align: right;">0.385</td>
<td style="text-align: right;">0.391</td>
<td style="text-align: right;">0.388</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">11793</td>
<td style="text-align: right;">10106</td>
<td style="text-align: right;">18177</td>
<td style="text-align: right;">0.539</td>
<td style="text-align: right;">0.393</td>
<td style="text-align: right;">0.455</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">4483</td>
<td style="text-align: right;">4659</td>
<td style="text-align: right;">5385</td>
<td style="text-align: right;">0.490</td>
<td style="text-align: right;">0.454</td>
<td style="text-align: right;">0.472</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">2928</td>
<td style="text-align: right;">930</td>
<td style="text-align: right;">1659</td>
<td style="text-align: right;">0.759</td>
<td style="text-align: right;">0.638</td>
<td style="text-align: right;">0.693</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:svpop-regions"></a>
<table>
<caption><span>Table S5:</span> Calling evaluation on the SVPOP dataset in different sets of regions for the HG5014 individual. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Region</th>
<th style="text-align: left;">Type</th>
<th style="text-align: right;">TP</th>
<th style="text-align: right;">FP</th>
<th style="text-align: right;">FN</th>
<th style="text-align: right;">Precision</th>
<th style="text-align: right;">Recall</th>
<th style="text-align: right;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">vg</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">8618</td>
<td style="text-align: right;">7237</td>
<td style="text-align: right;">5416</td>
<td style="text-align: right;">0.546</td>
<td style="text-align: right;">0.614</td>
<td style="text-align: right;">0.578</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">4762</td>
<td style="text-align: right;">2048</td>
<td style="text-align: right;">5145</td>
<td style="text-align: right;">0.696</td>
<td style="text-align: right;">0.481</td>
<td style="text-align: right;">0.569</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">54</td>
<td style="text-align: right;">0.579</td>
<td style="text-align: right;">0.169</td>
<td style="text-align: right;">0.262</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">6176</td>
<td style="text-align: right;">6923</td>
<td style="text-align: right;">4678</td>
<td style="text-align: right;">0.475</td>
<td style="text-align: right;">0.569</td>
<td style="text-align: right;">0.518</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">2428</td>
<td style="text-align: right;">1701</td>
<td style="text-align: right;">4542</td>
<td style="text-align: right;">0.584</td>
<td style="text-align: right;">0.348</td>
<td style="text-align: right;">0.436</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0.500</td>
<td style="text-align: right;">0.143</td>
<td style="text-align: right;">0.222</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">2677</td>
<td style="text-align: right;">987</td>
<td style="text-align: right;">514</td>
<td style="text-align: right;">0.731</td>
<td style="text-align: right;">0.839</td>
<td style="text-align: right;">0.781</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">1180</td>
<td style="text-align: right;">176</td>
<td style="text-align: right;">321</td>
<td style="text-align: right;">0.869</td>
<td style="text-align: right;">0.786</td>
<td style="text-align: right;">0.825</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">0.636</td>
<td style="text-align: right;">0.259</td>
<td style="text-align: right;">0.368</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">called in SMRT-SV</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">3410</td>
<td style="text-align: right;">3789</td>
<td style="text-align: right;">2108</td>
<td style="text-align: right;">0.478</td>
<td style="text-align: right;">0.618</td>
<td style="text-align: right;">0.539</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">2544</td>
<td style="text-align: right;">1092</td>
<td style="text-align: right;">1518</td>
<td style="text-align: right;">0.699</td>
<td style="text-align: right;">0.626</td>
<td style="text-align: right;">0.661</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">52</td>
<td style="text-align: right;">0.500</td>
<td style="text-align: right;">0.133</td>
<td style="text-align: right;">0.210</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">not called in SMRT-SV</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">4838</td>
<td style="text-align: right;">542</td>
<td style="text-align: right;">3678</td>
<td style="text-align: right;">0.899</td>
<td style="text-align: right;">0.568</td>
<td style="text-align: right;">0.696</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">2034</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">3723</td>
<td style="text-align: right;">0.987</td>
<td style="text-align: right;">0.353</td>
<td style="text-align: right;">0.520</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SMRT-SV</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">5245</td>
<td style="text-align: right;">8563</td>
<td style="text-align: right;">8789</td>
<td style="text-align: right;">0.394</td>
<td style="text-align: right;">0.374</td>
<td style="text-align: right;">0.384</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">3741</td>
<td style="text-align: right;">3382</td>
<td style="text-align: right;">6166</td>
<td style="text-align: right;">0.533</td>
<td style="text-align: right;">0.378</td>
<td style="text-align: right;">0.442</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">3848</td>
<td style="text-align: right;">7125</td>
<td style="text-align: right;">7006</td>
<td style="text-align: right;">0.368</td>
<td style="text-align: right;">0.354</td>
<td style="text-align: right;">0.361</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">1990</td>
<td style="text-align: right;">2832</td>
<td style="text-align: right;">4980</td>
<td style="text-align: right;">0.426</td>
<td style="text-align: right;">0.286</td>
<td style="text-align: right;">0.342</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">1396</td>
<td style="text-align: right;">1468</td>
<td style="text-align: right;">1795</td>
<td style="text-align: right;">0.493</td>
<td style="text-align: right;">0.438</td>
<td style="text-align: right;">0.464</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">901</td>
<td style="text-align: right;">308</td>
<td style="text-align: right;">600</td>
<td style="text-align: right;">0.745</td>
<td style="text-align: right;">0.600</td>
<td style="text-align: right;">0.665</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">called in SMRT-SV</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">4343</td>
<td style="text-align: right;">5595</td>
<td style="text-align: right;">1175</td>
<td style="text-align: right;">0.445</td>
<td style="text-align: right;">0.787</td>
<td style="text-align: right;">0.569</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">3227</td>
<td style="text-align: right;">2451</td>
<td style="text-align: right;">835</td>
<td style="text-align: right;">0.573</td>
<td style="text-align: right;">0.794</td>
<td style="text-align: right;">0.666</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">not called in SMRT-SV</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">116</td>
<td style="text-align: right;">109</td>
<td style="text-align: right;">8400</td>
<td style="text-align: right;">0.551</td>
<td style="text-align: right;">0.014</td>
<td style="text-align: right;">0.026</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">206</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">5551</td>
<td style="text-align: right;">0.911</td>
<td style="text-align: right;">0.036</td>
<td style="text-align: right;">0.069</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:simerror-bkpt"></a>
<table>
<caption><span>Table S6:</span> Breakpoint fine-tuning using graph augmentation from the read alignment. For deletions and inversions, either one or both breakpoints were shifted to introduce errors in the input VCF. For insertions, the insertion location and sequence contained errors. In all cases, the errors affected 1-10 bp. </caption>
<colgroup>
<col style="width: 10%" />
<col style="width: 15%" />
<col style="width: 13%" />
<col style="width: 10%" />
<col style="width: 13%" />
<col style="width: 17%" />
<col style="width: 19%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">SV type</th>
<th style="text-align: left;">Error type</th>
<th style="text-align: left;">Breakpoint</th>
<th style="text-align: right;">Variant</th>
<th style="text-align: right;">Proportion</th>
<th style="text-align: right;">Mean size (bp)</th>
<th style="text-align: right;">Mean error (bp)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">one end</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">220</td>
<td style="text-align: right;">0.219</td>
<td style="text-align: right;">422.655</td>
<td style="text-align: right;">6.095</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">784</td>
<td style="text-align: right;">0.781</td>
<td style="text-align: right;">670.518</td>
<td style="text-align: right;">5.430</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">both ends</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">811</td>
<td style="text-align: right;">0.814</td>
<td style="text-align: right;">826.070</td>
<td style="text-align: right;">6.275</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">185</td>
<td style="text-align: right;">0.186</td>
<td style="text-align: right;">586.676</td>
<td style="text-align: right;">2.232</td>
</tr>
<tr class="odd">
<td style="text-align: left;">INS</td>
<td style="text-align: left;">location/seq</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">123</td>
<td style="text-align: right;">0.062</td>
<td style="text-align: right;">428.724</td>
<td style="text-align: right;">6.667</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">1877</td>
<td style="text-align: right;">0.938</td>
<td style="text-align: right;">440.043</td>
<td style="text-align: right;">6.439</td>
</tr>
<tr class="odd">
<td style="text-align: left;">INV</td>
<td style="text-align: left;">one end</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">868</td>
<td style="text-align: right;">0.835</td>
<td style="text-align: right;">762.673</td>
<td style="text-align: right;">5.161</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">172</td>
<td style="text-align: right;">0.165</td>
<td style="text-align: right;">130.244</td>
<td style="text-align: right;">5.884</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">both ends</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">950</td>
<td style="text-align: right;">0.992</td>
<td style="text-align: right;">556.274</td>
<td style="text-align: right;">5.624</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">0.008</td>
<td style="text-align: right;">200.000</td>
<td style="text-align: right;">1.375</td>
</tr>
</tbody>
</table>
<h3 id="supplementary-figures">Supplementary Figures</h3>
<figure>
<img src="images/hgsvc-sim-geno.png" alt="Figure S1: Genotyping evaluation on the HGSVC dataset using simulated reads." id="fig:hgsvc-sim-geno" data-tag="S1" /><figcaption><span>Figure S1:</span> <strong>Genotyping evaluation on the HGSVC dataset using simulated reads.</strong></figcaption>
</figure>
<figure>
<img src="images/hgsvc-sim.png" alt="Figure S2: Calling evaluation on the HGSVC dataset using simulated reads." id="fig:hgsvc-sim" data-tag="S2" /><figcaption><span>Figure S2:</span> <strong>Calling evaluation on the HGSVC dataset using simulated reads.</strong></figcaption>
</figure>
<figure>
<img src="images/hgsvc-real-geno.png" alt="Figure S3: Genotyping evaluation on the HGSVC dataset using real reads. Combined results across the HG00514, HG00733 and NA19240." id="fig:hgsvc-real-geno" data-tag="S3" /><figcaption><span>Figure S3:</span> <strong>Genotyping evaluation on the HGSVC dataset using real reads.</strong> Combined results across the HG00514, HG00733 and NA19240.</figcaption>
</figure>
<figure>
<img src="images/hgsvc-real.png" alt="Figure S4: Calling evaluation on the HGSVC dataset using real reads. Combined results across the HG00514, HG00733 and NA19240." id="fig:hgsvc-real" data-tag="S4" /><figcaption><span>Figure S4:</span> <strong>Calling evaluation on the HGSVC dataset using real reads.</strong> Combined results across the HG00514, HG00733 and NA19240.</figcaption>
</figure>
<figure>
<img src="images/giab5-geno.png" alt="Figure S5: Genotyping evaluation on the Genome in a Bottle dataset. Predicted genotypes on HG002 were compared to the high-quality SVs from this same individual." id="fig:giab-geno" data-tag="S5" /><figcaption><span>Figure S5:</span> <strong>Genotyping evaluation on the Genome in a Bottle dataset.</strong> Predicted genotypes on HG002 were compared to the high-quality SVs from this same individual.</figcaption>
</figure>
<figure>
<img src="images/giab5.png" alt="Figure S6: Calling evaluation on the Genome in a Bottle dataset. Calls on HG002 were compared to the high-quality SVs from this same individual." id="fig:giab" data-tag="S6" /><figcaption><span>Figure S6:</span> <strong>Calling evaluation on the Genome in a Bottle dataset.</strong> Calls on HG002 were compared to the high-quality SVs from this same individual.</figcaption>
</figure>
<figure>
<img src="images/chmpd-geno.png" alt="Figure S7: Genotyping evaluation on the CHM pseudo-diploid dataset. The pseudo-diploid genome was built from CHM cell lines and used to train SMRT-SV v2 in Audano et al.[5]" id="fig:chmpd-geno" data-tag="S7" /><figcaption><span>Figure S7:</span> <strong>Genotyping evaluation on the CHM pseudo-diploid dataset.</strong> The pseudo-diploid genome was built from CHM cell lines and used to train SMRT-SV v2 in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span></figcaption>
</figure>
<figure>
<img src="images/chmpd.png" alt="Figure S8: Calling evaluation on the CHM pseudo-diploid dataset. The pseudo-diploid genome was built from CHM cell lines and used to train SMRT-SV v2 in Audano et al.[5]" id="fig:chmpd" data-tag="S8" /><figcaption><span>Figure S8:</span> <strong>Calling evaluation on the CHM pseudo-diploid dataset.</strong> The pseudo-diploid genome was built from CHM cell lines and used to train SMRT-SV v2 in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">5</a>]</span></figcaption>
</figure>
<figure>
<img src="images/svpop.png" alt="Figure S9: Calling evaluation on the SVPOP dataset. Combined results across the HG00514, HG00733 and NA19240." id="fig:svpop" data-tag="S9" /><figcaption><span>Figure S9:</span> <strong>Calling evaluation on the SVPOP dataset.</strong> Combined results across the HG00514, HG00733 and NA19240.</figcaption>
</figure>
<figure>
<img src="images/svpop-regions.png" alt="Figure S10: Evaluation across different sets of regions in HG00514 (SVPOP dataset). Calling evaluation." id="fig:svpop-regions" data-tag="S10" /><figcaption><span>Figure S10:</span> <strong>Evaluation across different sets of regions in HG00514 (SVPOP dataset)</strong>. Calling evaluation.</figcaption>
</figure>
<figure>
<img src="images/simerror-bkpt-finetuning-vgcall.png" alt="Figure S11: Breakpoint fine-tuning using augmentation through “vg call”. For deletions and inversions, either one or both breakpoints were shifted to introduce errors in the input VCF. For insertions, the insertion location and sequence contained errors. a) Proportion of variant for which breakpoints could be fine-tuned. b) Distribution of the amount of errors that could be corrected or not. c) Distribution of the size of the variants whose breakpoints could be fine-tuned or not." id="fig:simerror-bkpt" data-tag="S11" /><figcaption><span>Figure S11:</span> <strong>Breakpoint fine-tuning using augmentation through “vg call”.</strong> For deletions and inversions, either one or both breakpoints were shifted to introduce errors in the input VCF. For insertions, the insertion location and sequence contained errors. a) Proportion of variant for which breakpoints could be fine-tuned. b) Distribution of the amount of errors that could be corrected or not. c) Distribution of the size of the variants whose breakpoints could be fine-tuned or not.</figcaption>
</figure>
<figure>
<img src="images/panel3.png" alt="Figure S12: Mapping comparison. Short reads from all 12 yeast strains were aligned to both graphs. The fraction of reads mapped to the cactus graph (y-axis) and the VCF graph (x-axis) are compared. a) Stratified by percent identity threshold. b) Stratified by mapping quality threshold. Colors and shapes represent the 12 strains and two clades, respectively. Transparency indicates whether the strain was included or excluded in the graphs." id="fig:panel3" data-tag="S12" /><figcaption><span>Figure S12:</span> <strong>Mapping comparison.</strong> Short reads from all 12 yeast strains were aligned to both graphs. The fraction of reads mapped to the cactus graph (y-axis) and the VCF graph (x-axis) are compared. a) Stratified by percent identity threshold. b) Stratified by mapping quality threshold. Colors and shapes represent the 12 strains and two clades, respectively. Transparency indicates whether the strain was included or excluded in the graphs.</figcaption>
</figure>
<figure>
<img src="images/panel5.png" alt="Figure S13: Mapping comparison on graphs of all 12 strains. Short reads from all 12 yeast strains were aligned to both graphs. The fraction of reads mapped to the cactus graph (y-axis) and the VCF graph (x-axis) are compared. a) Stratified by percent identity threshold. b) Stratified by mapping quality threshold. Colors and shapes represent the 12 strains and two clades, respectively." id="fig:panel5" data-tag="S13" /><figcaption><span>Figure S13:</span> <strong>Mapping comparison on graphs of all 12 strains.</strong> Short reads from all 12 yeast strains were aligned to both graphs. The fraction of reads mapped to the <em>cactus graph</em> (y-axis) and the <em>VCF graph</em> (x-axis) are compared. a) Stratified by percent identity threshold. b) Stratified by mapping quality threshold. Colors and shapes represent the 12 strains and two clades, respectively.</figcaption>
</figure>
<figure>
<img src="images/panel6.png" alt="Figure S14: SV genotyping comparison on graphs of all 12 strains. Short reads from all 11 non-reference yeast strains were used to genotype SVs contained in both graphs. Subsequently, sample graphs were generated from the resulting SV callsets. The short reads were again aligned to the sample graphs and the quality of the alignments was used to ascertain genotyping performance. a) Average mapping quality of short reads aligned to the sample graphs derived from cactus graph (y-axis) and VCF graph (x-axis). b) Average mapping identity of short reads aligned to the sample graphs derived from cactus graph (y-axis) and VCF graph (x-axis). Colors and shapes represent the 11 non-reference strains and two clades, respectively." id="fig:panel6" data-tag="S14" /><figcaption><span>Figure S14:</span> <strong>SV genotyping comparison on graphs of all 12 strains.</strong> Short reads from all 11 non-reference yeast strains were used to genotype SVs contained in both graphs. Subsequently, sample graphs were generated from the resulting SV callsets. The short reads were again aligned to the sample graphs and the quality of the alignments was used to ascertain genotyping performance. a) Average mapping quality of short reads aligned to the sample graphs derived from <em>cactus graph</em> (y-axis) and <em>VCF graph</em> (x-axis). b) Average mapping identity of short reads aligned to the sample graphs derived from <em>cactus graph</em> (y-axis) and <em>VCF graph</em> (x-axis). Colors and shapes represent the 11 non-reference strains and two clades, respectively.</figcaption>
</figure>
<figure>
<img src="images/sveval-cartoon.png" alt="Figure S15: Overview of the SV evaluation by the sveval package. For deletions and inversions, we compute the proportion of a variant that is covered by variants in the other set, considering only variants overlapping with at least 10% reciprocal overlap. A variant is considered true positive if this coverage proportion is higher than 50% and false-positive or false-negative otherwise. A similar approach is used for insertions, although they are first clustered into pairs located less than 20 bp from each other. Then their inserted sequences are aligned to derive the coverage statistics. The SV evaluation approach is described in more detail in the Methods." id="fig:sveval" data-tag="S15" /><figcaption><span>Figure S15:</span> <strong>Overview of the SV evaluation by the <em>sveval</em> package</strong>. For deletions and inversions, we compute the proportion of a variant that is covered by variants in the other set, considering only variants overlapping with at least 10% reciprocal overlap. A variant is considered true positive if this coverage proportion is higher than 50% and false-positive or false-negative otherwise. A similar approach is used for insertions, although they are first clustered into pairs located less than 20 bp from each other. Then their inserted sequences are aligned to derive the coverage statistics. The SV evaluation approach is described in more detail in the <a href="#toil-vg-sveval">Methods</a>.</figcaption>
</figure>
<h3 id="supplementary-information">Supplementary Information</h3>
<h4 id="variation-graph-and-structural-variation">Variation graph and structural variation</h4>
<p>A variation graph encodes DNA sequence in its nodes. Such graphs are bidirected, in that we distinguish between edges incident on the starts of nodes from those incident on their ends. A path in such a graph is an ordered list of nodes where each is associated with an orientation. If a path walks from, for example, node A in the forward orientation to node B in the reverse orientation, then an edge must exist from the end of node A to the end of node B. Concatenating the sequences on each node in the path, taking the reverse complement when the node is visited in reverse orientation, produces a DNA sequence. Accordingly, variation graphs are constructed so as to encode haplotype sequences as walks through the graph. Variation between sequences shows up as bubbles in the graph <span class="citation" data-cites="xJlNnKH2">[<a href="#ref-xJlNnKH2">23</a>]</span>.</p>
<h4 id="breakpoint-fine-tuning">Breakpoint fine-tuning</h4>
<p>In addition to genotyping, vg can use an augmentation step to modify the graph based on the read alignment and discover novel variants. On the simulated SVs from Figure <a href="#fig:1">1</a>b, this approach was able to correct many of the 1-10 bp breakpoint errors that were added to the input VCF. The breakpoints were accurately fine-tuned for 93.8% of the insertions (Figure <a href="#fig:simerror-bkpt">S11</a>a and Table <a href="#tbl:simerror-bkpt">S6</a>). For deletions, 78.1% of the variants were corrected when only one breakpoint had an error. In situations where both breakpoints of the deletions were incorrect, only 18.6% were corrected through graph augmentation, and only when the amount of error was small (Figure <a href="#fig:simerror-bkpt">S11</a>b). The breakpoints of less than 20% of the inversions could be corrected. Across all SV types, the size of the variant didn’t affect the ability to fine-tune the breakpoints through graph augmentation (Figure <a href="#fig:simerror-bkpt">S11</a>c).</p>
<h4 id="mappability-comparison-between-yeast-graphs">Mappability comparison between yeast graphs</h4>
<p>In order to elucidate whether the <em>cactus graph</em> represents the sequence diversity among the yeast strains better than the <em>VCF graph</em>, we mapped Illumina short reads to both graphs using <code>vg map</code>. Generally, more reads mapped to the <em>cactus graph</em> with high identity (Figure <a href="#fig:panel3">S12</a>a) and high mapping quality (Figure <a href="#fig:panel3">S12</a>b) than to the <em>VCF graph</em>. The <em>VCF graph</em> exhibited higher mappability only on the reference strain <em>S.c. S288C</em> with a marginal difference. The benefit of using the <em>cactus graph</em> is largest for strains in the <em>S. paradoxus</em> clade and smaller for strains in the <em>S. cerevisiae</em> clade. We found that the genetic distance to the reference strain (as estimated using Mash v2.1 <span class="citation" data-cites="mH9pzoIn">[<a href="#ref-mH9pzoIn">39</a>]</span>) correlated with the increase in confidently mapped reads (mapping quality &gt;= 60) between the <em>cactus graph</em> and the <em>VCF graph</em> (Spearman’s rank correlation, p-value=3.993e-06). These results suggest that the improvement in mappability is not driven by the higher sequence content in the <em>cactus graph</em> alone (15.4 Mb compared to 12.4 Mb in the <em>VCF graph</em>). Instead, an explanation could be the construction of the <em>VCF graph</em> from a comprehensive but still limited list of variants and the lack of SNPs and small Indels in this list. Consequently, substantially fewer reads mapped to the <em>VCF graph</em> with perfect identity (Figure <a href="#fig:panel3">S12</a>a, percent identity threshold = 100%) than to the <em>cactus graph</em>. The latter has the advantage of implicitly incorporating variants of all types and sizes from the <em>de novo</em> assemblies. As a consequence, the <em>cactus graph</em> captures the genetic makeup of each strain more comprehensively and enables more reads to be mapped.</p>
<p>Interestingly, our measurements did not show a substantial difference between strains that were used to construct the graph and the other strains. Only the number of alignments with perfect identity is substantially lower for the strains that were not included in the creation of the graphs (Figure <a href="#fig:panel3">S12</a>a). For a direct comparison, see Figure <a href="#fig:panel5">S13</a> which shows results of the same experiment on graphs generated from all 12 strains.</p>
<h2 id="references" class="page_break_before">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs">
<div id="ref-2gpKwL67">
<p>1. <strong>The impact of structural variation on human gene expression</strong><br />
Colby ChiangAlexandra J Scott, Joe R Davis, Emily K Tsang, Xin Li, Yungil Kim, Tarik Hadzic, Farhan N Damani, Liron Ganel, … Ira M Hall<br />
<em>Nature Genetics</em> (2017-04-03) <a href="https://doi.org/f9xvr6">https://doi.org/f9xvr6</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3834">10.1038/ng.3834</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28369037">28369037</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5406250">PMC5406250</a></p>
</div>
<div id="ref-ebc66eBr">
<p>2. <strong>Phenotypic impact of genomic structural variation: insights from and for human disease</strong><br />
Joachim Weischenfeldt, Orsolya Symmons, François Spitz, Jan O. Korbel<br />
<em>Nature Reviews Genetics</em> (2013-02) <a href="https://doi.org/f4nhxh">https://doi.org/f4nhxh</a><br />
DOI: <a href="https://doi.org/10.1038/nrg3373">10.1038/nrg3373</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23329113">23329113</a></p>
</div>
<div id="ref-AltPnocw">
<p>3. <strong>SpeedSeq: ultra-fast personal genome analysis and interpretation</strong><br />
Colby Chiang, Ryan M Layer, Gregory G Faust, Michael R Lindberg, David B Rose, Erik P Garrison, Gabor T Marth, Aaron R Quinlan, Ira M Hall<br />
<em>Nature Methods</em> (2015-08-10) <a href="https://doi.org/gcpgfh">https://doi.org/gcpgfh</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.3505">10.1038/nmeth.3505</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26258291">26258291</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4589466">PMC4589466</a></p>
</div>
<div id="ref-nLvQCjXU">
<p>4. <strong>DELLY: structural variant discovery by integrated paired-end and split-read analysis</strong><br />
T. Rausch, T. Zichner, A. Schlattl, A. M. Stutz, V. Benes, J. O. Korbel<br />
<em>Bioinformatics</em> (2012-09-07) <a href="https://doi.org/f38r2c">https://doi.org/f38r2c</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/bts378">10.1093/bioinformatics/bts378</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22962449">22962449</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3436805">PMC3436805</a></p>
</div>
<div id="ref-3NNFS6U2">
<p>5. <strong>Characterizing the Major Structural Variant Alleles of the Human Genome</strong><br />
Peter A. Audano, Arvis Sulovari, Tina A. Graves-Lindsay, Stuart Cantsilieris, Melanie Sorensen, AnneMarie E. Welch, Max L. Dougherty, Bradley J. Nelson, Ankeeta Shah, Susan K. Dutcher, … Evan E. Eichler<br />
<em>Cell</em> (2019-01) <a href="https://doi.org/gfthvz">https://doi.org/gfthvz</a><br />
DOI: <a href="https://doi.org/10.1016/j.cell.2018.12.019">10.1016/j.cell.2018.12.019</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30661756">30661756</a></p>
</div>
<div id="ref-qA6dWFP">
<p>6. <strong>An integrated map of structural variation in 2,504 human genomes</strong><br />
Peter H. SudmantTobias Rausch, Eugene J. Gardner, Robert E. Handsaker, Alexej Abyzov, John Huddleston, Yan Zhang, Kai Ye, Goo Jun, … Jan O. Korbel<br />
<em>Nature</em> (2015-10) <a href="https://doi.org/73c">https://doi.org/73c</a><br />
DOI: <a href="https://doi.org/10.1038/nature15394">10.1038/nature15394</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26432246">26432246</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4617611">PMC4617611</a></p>
</div>
<div id="ref-py6BC5kj">
<p>7. <strong>Whole-genome sequence variation, population structure and demographic history of the Dutch population</strong><br />
Laurent C FrancioliAndroniki Menelaou, Sara L Pulit, Freerk van Dijk, Pier Francesco Palamara, Clara C Elbers, Pieter BT Neerincx, Kai Ye, Victor Guryev, … Cisca Wijmenga<br />
<em>Nature Genetics</em> (2014-06-29) <a href="https://doi.org/f6bxm8">https://doi.org/f6bxm8</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3021">10.1038/ng.3021</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24974849">24974849</a></p>
</div>
<div id="ref-z91V6jjU">
<p>8. <strong>Resolving the complexity of the human genome using single-molecule sequencing</strong><br />
Mark J. P. Chaisson, John Huddleston, Megan Y. Dennis, Peter H. Sudmant, Maika Malig, Fereydoun Hormozdiari, Francesca Antonacci, Urvashi Surti, Richard Sandstrom, Matthew Boitano, … Evan E. Eichler<br />
<em>Nature</em> (2014-11-10) <a href="https://doi.org/w69">https://doi.org/w69</a><br />
DOI: <a href="https://doi.org/10.1038/nature13907">10.1038/nature13907</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25383537">25383537</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4317254">PMC4317254</a></p>
</div>
<div id="ref-rs7e40wC">
<p>9. <strong>Discovery and genotyping of structural variation from long-read haploid genome sequence data</strong><br />
John Huddleston, Mark J.P. Chaisson, Karyn Meltz Steinberg, Wes Warren, Kendra Hoekzema, David Gordon, Tina A. Graves-Lindsay, Katherine M. Munson, Zev N. Kronenberg, Laura Vives, … Evan E. Eichler<br />
<em>Genome Research</em> (2016-11-28) <a href="https://doi.org/f9x79h">https://doi.org/f9x79h</a><br />
DOI: <a href="https://doi.org/10.1101/gr.214007.116">10.1101/gr.214007.116</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27895111">27895111</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411763">PMC5411763</a></p>
</div>
<div id="ref-PRx3qEIm">
<p>10. <strong>Mapping and phasing of structural variation in patient genomes using nanopore sequencing</strong><br />
Mircea Cretu Stancu, Markus J. van Roosmalen, Ivo Renkens, Marleen M. Nieboer, Sjors Middelkamp, Joep de Ligt, Giulia Pregno, Daniela Giachino, Giorgia Mandrile, Jose Espejo Valle-Inclan, … Wigard P. Kloosterman<br />
<em>Nature Communications</em> (2017-11-06) <a href="https://doi.org/gftpt9">https://doi.org/gftpt9</a><br />
DOI: <a href="https://doi.org/10.1038/s41467-017-01343-4">10.1038/s41467-017-01343-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29109544">29109544</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5673902">PMC5673902</a></p>
</div>
<div id="ref-121OWcTA4">
<p>11. <strong>Genome-wide reconstruction of complex structural variants using read clouds</strong><br />
Noah Spies, Ziming Weng, Alex Bishara, Jennifer McDaniel, David Catoe, Justin M Zook, Marc Salit, Robert B West, Serafim Batzoglou, Arend Sidow<br />
<em>Nature Methods</em> (2017-07-17) <a href="https://doi.org/gbnhwk">https://doi.org/gbnhwk</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.4366">10.1038/nmeth.4366</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28714986">28714986</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5578891">PMC5578891</a></p>
</div>
<div id="ref-6KbgcueR">
<p>12. <strong>Nanopore sequencing and assembly of a human genome with ultra-long reads</strong><br />
Miten Jain, Sergey Koren, Karen H Miga, Josh Quick, Arthur C Rand, Thomas A Sasani, John R Tyson, Andrew D Beggs, Alexander T Dilthey, Ian T Fiddes, … Matthew Loose<br />
<em>Nature Biotechnology</em> (2018-01-29) <a href="https://doi.org/gczffw">https://doi.org/gczffw</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.4060">10.1038/nbt.4060</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29431738">29431738</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5889714">PMC5889714</a></p>
</div>
<div id="ref-Pu6SY37C">
<p>13. <strong>Phased diploid genome assembly with single-molecule real-time sequencing</strong><br />
Chen-Shan Chin, Paul Peluso, Fritz J Sedlazeck, Maria Nattestad, Gregory T Concepcion, Alicia Clum, Christopher Dunn, Ronan O’Malley, Rosa Figueroa-Balderas, Abraham Morales-Cruz, … Michael C Schatz<br />
<em>Nature Methods</em> (2016-10-17) <a href="https://doi.org/f9fv4w">https://doi.org/f9fv4w</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.4035">10.1038/nmeth.4035</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27749838">27749838</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5503144">PMC5503144</a></p>
</div>
<div id="ref-Qa8mx6Ll">
<p>14. <strong>Genome graphs and the evolution of genome inference</strong><br />
Benedict Paten, Adam M. Novak, Jordan M. Eizenga, Erik Garrison<br />
<em>Genome Research</em> (2017-03-30) <a href="https://doi.org/f95nhd">https://doi.org/f95nhd</a><br />
DOI: <a href="https://doi.org/10.1101/gr.214155.116">10.1101/gr.214155.116</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28360232">28360232</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411762">PMC5411762</a></p>
</div>
<div id="ref-10jxt15v0">
<p>15. <strong>Variation graph toolkit improves read mapping by representing genetic variation in the reference</strong><br />
Erik Garrison, Jouni Sirén, Adam M Novak, Glenn Hickey, Jordan M Eizenga, Eric T Dawson, William Jones, Shilpa Garg, Charles Markello, Michael F Lin, … Richard Durbin<br />
<em>Nature Biotechnology</em> (2018-08-20) <a href="https://doi.org/gd2zqs">https://doi.org/gd2zqs</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.4227">10.1038/nbt.4227</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30125266">30125266</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6126949">PMC6126949</a></p>
</div>
<div id="ref-DuODeStx">
<p>16. <strong>Genome Graphs</strong><br />
Adam M Novak, Glenn Hickey, Erik Garrison, Sean Blum, Abram Connelly, Alexander Dilthey, Jordan Eizenga, M. A. Saleh Elmohamed, Sally Guthrie, André Kahles, … Benedict Paten<br />
<em>Cold Spring Harbor Laboratory</em> (2017-01-18) <a href="https://doi.org/gdcc74">https://doi.org/gdcc74</a><br />
DOI: <a href="https://doi.org/10.1101/101378">10.1101/101378</a></p>
</div>
<div id="ref-11Jy8B61m">
<p>17. <strong>Fast and accurate genomic analyses using genome graphs</strong><br />
Goran Rakocevic, Vladimir Semenyuk, Wan-Ping Lee, James Spencer, John Browning, Ivan J. Johnson, Vladan Arsenijevic, Jelena Nadj, Kaushik Ghose, Maria C. Suciu, … Deniz Kural<br />
<em>Nature Genetics</em> (2019-01-14) <a href="https://doi.org/gftd46">https://doi.org/gftd46</a><br />
DOI: <a href="https://doi.org/10.1038/s41588-018-0316-4">10.1038/s41588-018-0316-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30643257">30643257</a></p>
</div>
<div id="ref-ohTIiqfV">
<p>18. <strong>Graphtyper enables population-scale genotyping using pangenome graphs</strong><br />
Hannes P Eggertsson, Hakon Jonsson, Snaedis Kristmundsdottir, Eirikur Hjartarson, Birte Kehr, Gisli Masson, Florian Zink, Kristjan E Hjorleifsson, Aslaug Jonasdottir, Adalbjorg Jonasdottir, … Bjarni V Halldorsson<br />
<em>Nature Genetics</em> (2017-09-25) <a href="https://doi.org/gbx7v6">https://doi.org/gbx7v6</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3964">10.1038/ng.3964</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28945251">28945251</a></p>
</div>
<div id="ref-14Uxmwbxm">
<p>19. <strong>Accurate genotyping across variant classes and lengths using variant graphs</strong><br />
Jonas Andreas SibbesenLasse Maretty, Anders Krogh<br />
<em>Nature Genetics</em> (2018-06-18) <a href="https://doi.org/gdndnz">https://doi.org/gdndnz</a><br />
DOI: <a href="https://doi.org/10.1038/s41588-018-0145-5">10.1038/s41588-018-0145-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29915429">29915429</a></p>
</div>
<div id="ref-vQTymKCj">
<p>20. <strong>Multi-platform discovery of haplotype-resolved structural variation in human genomes</strong><br />
Mark J.P. Chaisson, Ashley D. Sanders, Xuefang Zhao, Ankit Malhotra, David Porubsky, Tobias Rausch, Eugene J. Gardner, Oscar Rodriguez, Li Guo, Ryan L. Collins, … Charles Lee<br />
<em>Cold Spring Harbor Laboratory</em> (2017-09-23) <a href="https://doi.org/gftxhc">https://doi.org/gftxhc</a><br />
DOI: <a href="https://doi.org/10.1101/193144">10.1101/193144</a></p>
</div>
<div id="ref-14neTdqfN">
<p>21. <strong>Extensive sequencing of seven human genomes to characterize benchmark reference materials</strong><br />
Justin M. Zook, David Catoe, Jennifer McDaniel, Lindsay Vang, Noah Spies, Arend Sidow, Ziming Weng, Yuling Liu, Christopher E. Mason, Noah Alexander, … Marc Salit<br />
<em>Scientific Data</em> (2016-06-07) <a href="https://doi.org/f84nqc">https://doi.org/f84nqc</a><br />
DOI: <a href="https://doi.org/10.1038/sdata.2016.25">10.1038/sdata.2016.25</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27271295">27271295</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4896128">PMC4896128</a></p>
</div>
<div id="ref-16GvGhO20">
<p>22. <strong>Reproducible integration of multiple sequencing datasets to form high-confidence SNP, indel, and reference calls for five human genome reference materials</strong><br />
Justin Zook, Jennifer McDaniel, Hemang Parikh, Haynes Heaton, Sean A Irvine, Len Trigg, Rebecca Truty, Cory Y McLean, Francisco M De La Vega, Chunlin Xiao, … <br />
<em>Cold Spring Harbor Laboratory</em> (2018-03-13) <a href="https://doi.org/gfwsmj">https://doi.org/gfwsmj</a><br />
DOI: <a href="https://doi.org/10.1101/281006">10.1101/281006</a></p>
</div>
<div id="ref-xJlNnKH2">
<p>23. <strong>Superbubbles, Ultrabubbles, and Cacti</strong><br />
Benedict Paten, Jordan M. Eizenga, Yohei M. Rosen, Adam M. Novak, Erik Garrison, Glenn Hickey<br />
<em>Journal of Computational Biology</em> (2018-07) <a href="https://doi.org/gdw582">https://doi.org/gdw582</a><br />
DOI: <a href="https://doi.org/10.1089/cmb.2017.0251">10.1089/cmb.2017.0251</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29461862">29461862</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6067107">PMC6067107</a></p>
</div>
<div id="ref-1FgS53pXi">
<p>24. <strong>Cactus: Algorithms for genome multiple sequence alignment</strong><br />
B. Paten, D. Earl, N. Nguyen, M. Diekhans, D. Zerbino, D. Haussler<br />
<em>Genome Research</em> (2011-06-10) <a href="https://doi.org/bk4697">https://doi.org/bk4697</a><br />
DOI: <a href="https://doi.org/10.1101/gr.123356.111">10.1101/gr.123356.111</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21665927">21665927</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3166836">PMC3166836</a></p>
</div>
<div id="ref-7f5OKa5O">
<p>25. <strong>Contrasting evolutionary genome dynamics between domesticated and wild yeasts</strong><br />
Jia-Xing Yue, Jing Li, Louise Aigrain, Johan Hallin, Karl Persson, Karen Oliver, Anders Bergström, Paul Coupland, Jonas Warringer, Marco Cosentino Lagomarsino, … Gianni Liti<br />
<em>Nature Genetics</em> (2017-04-17) <a href="https://doi.org/f93kpp">https://doi.org/f93kpp</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3847">10.1038/ng.3847</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28416820">28416820</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5446901">PMC5446901</a></p>
</div>
<div id="ref-krO7WgVi">
<p>26. <strong>Assemblytics: a web analytics tool for the detection of variants from an assembly</strong><br />
Maria Nattestad, Michael C. Schatz<br />
<em>Bioinformatics</em> (2016-06-17) <a href="https://doi.org/f9c485">https://doi.org/f9c485</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/btw369">10.1093/bioinformatics/btw369</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27318204">27318204</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6191160">PMC6191160</a></p>
</div>
<div id="ref-oVaXIwl5">
<p>27. <strong>Discovery, genotyping and characterization of structural variation and novel sequence at single nucleotide resolution from de novo genome assemblies on a population scale</strong><br />
Siyang LiuShujia Huang, Junhua Rao, Weijian Ye, Anders Krogh, Jun Wang<br />
<em>GigaScience</em> (2015-12) <a href="https://doi.org/f75r4n">https://doi.org/f75r4n</a><br />
DOI: <a href="https://doi.org/10.1186/s13742-015-0103-4">10.1186/s13742-015-0103-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26705468">26705468</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4690232">PMC4690232</a></p>
</div>
<div id="ref-172cJaw4Q">
<p>28. <strong>Minimap2: pairwise alignment for nucleotide sequences</strong><br />
Heng Li<br />
<em>Bioinformatics</em> (2018-05-10) <a href="https://doi.org/gdhbqt">https://doi.org/gdhbqt</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/bty191">10.1093/bioinformatics/bty191</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29750242">29750242</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6137996">PMC6137996</a></p>
</div>
<div id="ref-10Jid8Wql">
<p>29. <strong>The Pancancer Analysis of Whole Genomes (PCAWG).</strong><a href="https://dcc.icgc.org/pcawg">https://dcc.icgc.org/pcawg</a></p>
</div>
<div id="ref-mWj2p7Xp">
<p>30. <strong>Genomics England 100,000 Genomes Project.</strong><a href="https://www.genomicsengland.co.uk">https://www.genomicsengland.co.uk</a></p>
</div>
<div id="ref-ir1O1h8n">
<p>31. <strong>Whole Genome Sequencing in the NHLBI Trans-Omics for Precision Medicine (TOPMed).</strong><a href="https://www.nhlbiwgs.org/">https://www.nhlbiwgs.org/</a></p>
</div>
<div id="ref-pJAv1D8R">
<p>32. <strong>Sequencing and de novo assembly of 150 genomes from Denmark as a population reference</strong><br />
Lasse Maretty, Jacob Malte Jensen, Bent Petersen, Jonas Andreas Sibbesen, Siyang Liu, Palle Villesen, Laurits Skov, Kirstine Belling, Christian Theil Have, Jose M. G. Izarzugaza, … Mikkel Heide Schierup<br />
<em>Nature</em> (2017-07-26) <a href="https://doi.org/gbpnnx">https://doi.org/gbpnnx</a><br />
DOI: <a href="https://doi.org/10.1038/nature23264">10.1038/nature23264</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28746312">28746312</a></p>
</div>
<div id="ref-faeC2cx0">
<p>33. <strong>Toil enables reproducible, open source, big biomedical data analyses</strong><br />
John Vivian, Arjun Arkal Rao, Frank Austin Nothaft, Christopher Ketchum, Joel Armstrong, Adam Novak, Jacob Pfeil, Jake Narkizian, Alden D Deran, Audrey Musselman-Brown, … Benedict Paten<br />
<em>Nature Biotechnology</em> (2017-04) <a href="https://doi.org/gfxbhs">https://doi.org/gfxbhs</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.3772">10.1038/nbt.3772</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28398314">28398314</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5546205">PMC5546205</a></p>
</div>
<div id="ref-LAG2q9WK">
<p>34. <strong>Bcftools 1.9</strong><a href="https://samtools.github.io/bcftools/">https://samtools.github.io/bcftools/</a></p>
</div>
<div id="ref-NCr4QkOg">
<p>35. <strong>A framework for variation discovery and genotyping using next-generation DNA sequencing data</strong><br />
Mark A DePristo, Eric Banks, Ryan Poplin, Kiran V Garimella, Jared R Maguire, Christopher Hartl, Anthony A Philippakis, Guillermo del Angel, Manuel A Rivas, Matt Hanna, … Mark J Daly<br />
<em>Nature Genetics</em> (2011-04-10) <a href="https://doi.org/d9k453">https://doi.org/d9k453</a><br />
DOI: <a href="https://doi.org/10.1038/ng.806">10.1038/ng.806</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21478889">21478889</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3083463">PMC3083463</a></p>
</div>
<div id="ref-1DYmkalz4">
<p>36. <strong>Integrating mapping-, assembly- and haplotype-based approaches for calling variants in clinical sequencing applications</strong><br />
Andy RimmerHang Phan, Iain Mathieson, Zamin Iqbal, Stephen RF Twigg, Andrew OM Wilkie, Gil McVean, Gerton Lunter<br />
<em>Nature Genetics</em> (2014-07-13) <a href="https://doi.org/f6b6dk">https://doi.org/f6b6dk</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3036">10.1038/ng.3036</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25017105">25017105</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4753679">PMC4753679</a></p>
</div>
<div id="ref-jj4JJlsg">
<p>37. <strong>Haplotype-aware graph indexes</strong><br />
Jouni Sirén, Erik Garrison, Adam M. Novak, Benedict Paten, Richard Durbin<br />
<em>Schloss Dagstuhl - Leibniz-Zentrum fuer Informatik GmbH, Wadern/Saarbruecken, Germany</em> (2018) <a href="https://doi.org/gf2jss">https://doi.org/gf2jss</a><br />
DOI: <a href="https://doi.org/10.4230/lipics.wabi.2018.4">10.4230/lipics.wabi.2018.4</a></p>
</div>
<div id="ref-1HWiAHnIw">
<p>38. <strong>BEDTools: a flexible suite of utilities for comparing genomic features</strong><br />
Aaron R. Quinlan, Ira M. Hall<br />
<em>Bioinformatics</em> (2010-01-28) <a href="https://doi.org/cmrms3">https://doi.org/cmrms3</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/btq033">10.1093/bioinformatics/btq033</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20110278">20110278</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2832824">PMC2832824</a></p>
</div>
<div id="ref-mH9pzoIn">
<p>39. <strong>Mash: fast genome and metagenome distance estimation using MinHash</strong><br />
Brian D. Ondov, Todd J. Treangen, Páll Melsted, Adam B. Mallonee, Nicholas H. Bergman, Sergey Koren, Adam M. Phillippy<br />
<em>Genome Biology</em> (2016-06-20) <a href="https://doi.org/gfx74q">https://doi.org/gfx74q</a><br />
DOI: <a href="https://doi.org/10.1186/s13059-016-0997-x">10.1186/s13059-016-0997-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27323842">27323842</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4915045">PMC4915045</a></p>
</div>
<div id="ref-tIvRXd6o">
<p>40. <strong>PHYLIP - Phylogeny Inference Package (Version 3.2).</strong><br />
Joel Felsenstein<br />
<em>Cladistics</em> (1989)</p>
</div>
</div>
<script>
// AnchorJS minified version below.
// Source https://github.com/bryanbraun/anchorjs/blob/064abdd0987f305933ec4982af6d0c1cf2fd0814/anchor.js

/**
 * AnchorJS - v4.0.0 - 2017-06-02
 * https://github.com/bryanbraun/anchorjs
 * Copyright (c) 2017 Bryan Braun; Licensed MIT
 */
!function(A,e){"use strict";"function"==typeof define&&define.amd?define([],e):"object"==typeof module&&module.exports?module.exports=e():(A.AnchorJS=e(),A.anchors=new A.AnchorJS)}(this,function(){"use strict";function A(A){function e(A){A.icon=A.hasOwnProperty("icon")?A.icon:"",A.visible=A.hasOwnProperty("visible")?A.visible:"hover",A.placement=A.hasOwnProperty("placement")?A.placement:"right",A.class=A.hasOwnProperty("class")?A.class:"",A.truncate=A.hasOwnProperty("truncate")?Math.floor(A.truncate):64}function t(A){var e;if("string"==typeof A||A instanceof String)e=[].slice.call(document.querySelectorAll(A));else{if(!(Array.isArray(A)||A instanceof NodeList))throw new Error("The selector provided to AnchorJS was invalid.");e=[].slice.call(A)}return e}function n(){if(null===document.head.querySelector("style.anchorjs")){var A,e=document.createElement("style");e.className="anchorjs",e.appendChild(document.createTextNode("")),void 0===(A=document.head.querySelector('[rel="stylesheet"], style'))?document.head.appendChild(e):document.head.insertBefore(e,A),e.sheet.insertRule(" .anchorjs-link {   opacity: 0;   text-decoration: none;   -webkit-font-smoothing: antialiased;   -moz-osx-font-smoothing: grayscale; }",e.sheet.cssRules.length),e.sheet.insertRule(" *:hover > .anchorjs-link, .anchorjs-link:focus  {   opacity: 1; }",e.sheet.cssRules.length),e.sheet.insertRule(" [data-anchorjs-icon]::after {   content: attr(data-anchorjs-icon); }",e.sheet.cssRules.length),e.sheet.insertRule(' @font-face {   font-family: "anchorjs-icons";   src: url(data:n/a;base64,AAEAAAALAIAAAwAwT1MvMg8yG2cAAAE4AAAAYGNtYXDp3gC3AAABpAAAAExnYXNwAAAAEAAAA9wAAAAIZ2x5ZlQCcfwAAAH4AAABCGhlYWQHFvHyAAAAvAAAADZoaGVhBnACFwAAAPQAAAAkaG10eASAADEAAAGYAAAADGxvY2EACACEAAAB8AAAAAhtYXhwAAYAVwAAARgAAAAgbmFtZQGOH9cAAAMAAAAAunBvc3QAAwAAAAADvAAAACAAAQAAAAEAAHzE2p9fDzz1AAkEAAAAAADRecUWAAAAANQA6R8AAAAAAoACwAAAAAgAAgAAAAAAAAABAAADwP/AAAACgAAA/9MCrQABAAAAAAAAAAAAAAAAAAAAAwABAAAAAwBVAAIAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAMCQAGQAAUAAAKZAswAAACPApkCzAAAAesAMwEJAAAAAAAAAAAAAAAAAAAAARAAAAAAAAAAAAAAAAAAAAAAQAAg//0DwP/AAEADwABAAAAAAQAAAAAAAAAAAAAAIAAAAAAAAAIAAAACgAAxAAAAAwAAAAMAAAAcAAEAAwAAABwAAwABAAAAHAAEADAAAAAIAAgAAgAAACDpy//9//8AAAAg6cv//f///+EWNwADAAEAAAAAAAAAAAAAAAAACACEAAEAAAAAAAAAAAAAAAAxAAACAAQARAKAAsAAKwBUAAABIiYnJjQ3NzY2MzIWFxYUBwcGIicmNDc3NjQnJiYjIgYHBwYUFxYUBwYGIwciJicmNDc3NjIXFhQHBwYUFxYWMzI2Nzc2NCcmNDc2MhcWFAcHBgYjARQGDAUtLXoWOR8fORYtLTgKGwoKCjgaGg0gEhIgDXoaGgkJBQwHdR85Fi0tOAobCgoKOBoaDSASEiANehoaCQkKGwotLXoWOR8BMwUFLYEuehYXFxYugC44CQkKGwo4GkoaDQ0NDXoaShoKGwoFBe8XFi6ALjgJCQobCjgaShoNDQ0NehpKGgobCgoKLYEuehYXAAAADACWAAEAAAAAAAEACAAAAAEAAAAAAAIAAwAIAAEAAAAAAAMACAAAAAEAAAAAAAQACAAAAAEAAAAAAAUAAQALAAEAAAAAAAYACAAAAAMAAQQJAAEAEAAMAAMAAQQJAAIABgAcAAMAAQQJAAMAEAAMAAMAAQQJAAQAEAAMAAMAAQQJAAUAAgAiAAMAAQQJAAYAEAAMYW5jaG9yanM0MDBAAGEAbgBjAGgAbwByAGoAcwA0ADAAMABAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAH//wAP) format("truetype"); }',e.sheet.cssRules.length)}}this.options=A||{},this.elements=[],e(this.options),this.isTouchDevice=function(){return!!("ontouchstart"in window||window.DocumentTouch&&document instanceof DocumentTouch)},this.add=function(A){var i,o,s,c,r,a,h,l,u,d,f,g,p=[];if(e(this.options),"touch"===(g=this.options.visible)&&(g=this.isTouchDevice()?"always":"hover"),A||(A="h2, h3, h4, h5, h6"),0===(i=t(A)).length)return this;for(n(),o=document.querySelectorAll("[id]"),s=[].map.call(o,function(A){return A.id}),r=0;r<i.length;r++)if(this.hasAnchorJSLink(i[r]))p.push(r);else{if(i[r].hasAttribute("id"))c=i[r].getAttribute("id");else if(i[r].hasAttribute("data-anchor-id"))c=i[r].getAttribute("data-anchor-id");else{u=l=this.urlify(i[r].textContent),h=0;do{void 0!==a&&(u=l+"-"+h),a=s.indexOf(u),h+=1}while(-1!==a);a=void 0,s.push(u),i[r].setAttribute("id",u),c=u}d=c.replace(/-/g," "),(f=document.createElement("a")).className="anchorjs-link "+this.options.class,f.href="#"+c,f.setAttribute("aria-label","Anchor link for: "+d),f.setAttribute("data-anchorjs-icon",this.options.icon),"always"===g&&(f.style.opacity="1"),""===this.options.icon&&(f.style.font="1em/1 anchorjs-icons","left"===this.options.placement&&(f.style.lineHeight="inherit")),"left"===this.options.placement?(f.style.position="absolute",f.style.marginLeft="-1em",f.style.paddingRight="0.5em",i[r].insertBefore(f,i[r].firstChild)):(f.style.paddingLeft="0.375em",i[r].appendChild(f))}for(r=0;r<p.length;r++)i.splice(p[r]-r,1);return this.elements=this.elements.concat(i),this},this.remove=function(A){for(var e,n,i=t(A),o=0;o<i.length;o++)(n=i[o].querySelector(".anchorjs-link"))&&(-1!==(e=this.elements.indexOf(i[o]))&&this.elements.splice(e,1),i[o].removeChild(n));return this},this.removeAll=function(){this.remove(this.elements)},this.urlify=function(A){var t=/[& +$,:;=?@"#{}|^~[`%!'<>\]\.\/\(\)\*\\]/g;return this.options.truncate||e(this.options),A.trim().replace(/\'/gi,"").replace(t,"-").replace(/-{2,}/g,"-").substring(0,this.options.truncate).replace(/^-+|-+$/gm,"").toLowerCase()},this.hasAnchorJSLink=function(A){var e=A.firstChild&&(" "+A.firstChild.className+" ").indexOf(" anchorjs-link ")>-1,t=A.lastChild&&(" "+A.lastChild.className+" ").indexOf(" anchorjs-link ")>-1;return e||t||!1}}return A});

// Enable links for selected headers
var anchors = new AnchorJS();
anchors.add("h2, h3, h4")
</script>
<!-- Enable Hypothesis annotation. https://web.hypothes.is/ -->
<script src="https://hypothes.is/embed.js" async></script>
</body>
</html>
