<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Glenn Hickey" />
  <meta name="author" content="David Heller" />
  <meta name="author" content="Jean Monlong" />
  <meta name="author" content="Benedict Paten" />
  <meta name="dcterms.date" content="2019-01-22" />
  <meta name="keywords" content="structural variation, pangenome, variant genotyping" />
  <title>Genotyping structural variation in variation graphs with the vg toolkit</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="github-pandoc.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!-- Insert Analytics Script Below -->
  <script>
  </script>
  <!-- End Analytics Script -->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Genotyping structural variation in variation graphs with the vg toolkit</h1>
</header>
<p><small><em> This manuscript (<a href="https://jmonlong.github.io/manu-vgsv/v/2e0caef1fe69af1cc8232e14eb8a669880bc5fcd/">permalink</a>) was automatically generated from <a href="https://github.com/jmonlong/manu-vgsv/tree/2e0caef1fe69af1cc8232e14eb8a669880bc5fcd">jmonlong/manu-vgsv@2e0caef</a> on January 22, 2019. </em></small></p>
<h2 id="authors">Authors</h2>
<p><a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Glenn Hickey<sup>1,☯</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> David Heller<sup>1,☯</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Jean Monlong<sup>1,☯</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Benedict Paten<sup>1,†</sup></p>
<p><sup>☯</sup> — These authors contributed equally to this work</p>
<p><sup>†</sup> — To whom correspondence should be addressed: bpaten@ucsc.edu <small></p>
<ol type="1">
<li>UC Santa Cruz Genomics Institute, University of California, Santa Cruz, California, USA</li>
</ol>
<p></small></p>
<h2 id="abstract" class="page_break_before">Abstract</h2>
<h2 id="introduction" class="page_break_before">Introduction</h2>
<p>Structural variation (SV) represents genomic mutation involving 50 bp or more and can take several forms, such as for example deletions, insertions, inversions, or translocations. Although whole-genome sequencing (WGS) made it possible to assess virtually any type of structural variation, many challenges remain. In particular, SV-supporting reads are difficult to map to reference genomes. Multi-mapping, caused by widespread repeated sequences in the genome, is another issue because it often resembles SV-supporting signal. As a result, many SV detection algorithms have been developed and multiple methods must usually be combined to minimize false positives. Several large-scale projects used this ensemble approach, cataloging tens of thousands of SV in humans<span class="citation" data-cites="qA6dWFP py6BC5kj">[<a href="#ref-qA6dWFP">1</a>,<a href="#ref-py6BC5kj">2</a>]</span>. SV detection from short-read sequencing remains laborious and of lower accuracy, explaining why these variants and their impact have been under-studied as compared to single-nucleotide variants (SNVs) and small insertions/deletions (indels).</p>
<p>Over the last few years, exciting developments in sequencing technologies and library preparation made it possible to produce long reads or retrieve long-range information over kilobases of sequence. These approaches are maturing to the point were it is feasible to analyze the human genome. This multi-kbp information is particularly useful for SV detection and de novo assembly. In the last few years, several studies using long-read or linked-read sequencing have produced large catalogs of structural variation, the majority of which were novel and sequence-resolved<span class="citation" data-cites="z91V6jjU rs7e40wC PRx3qEIm 121OWcTA4">[<a href="#ref-z91V6jjU">3</a>,<a href="#ref-rs7e40wC">4</a>,<a href="#ref-PRx3qEIm">5</a>,<a href="#ref-121OWcTA4">6</a>]</span> (<em>REF_PETER_SOON</em>). These technologies are also enabling high-quality de novo genome assemblies to be produced<span class="citation" data-cites="z91V6jjU 6KbgcueR">[<a href="#ref-z91V6jjU">3</a>,<a href="#ref-6KbgcueR">7</a>]</span>, as well as large blocks of haplotype-resolved sequences<span class="citation" data-cites="Pu6SY37C">[<a href="#ref-Pu6SY37C">8</a>]</span>. These technological advances promise to expand the amount of known genomic variation in humans in the near future.</p>
<p>In parallel, the reference genome is evolving from a linear reference to a graph-based reference that contains known genomic variation<span class="citation" data-cites="Qa8mx6Ll 10jxt15v0 11Jy8B61m">[<a href="#ref-10jxt15v0">10</a>,<a href="#ref-11Jy8B61m">11</a>,<a href="#ref-Qa8mx6Ll">9</a>]</span>. By having variants in the graph, mapping rates are increased and variants are more uniformly covered, including indels and variants in complex regions<span class="citation" data-cites="10jxt15v0">[<a href="#ref-10jxt15v0">10</a>]</span>. Both the mapping and variant calling become variant-aware and benefit in term of accuracy and sensitivity. In addition, different variant types are called simultaneously by a unified framework. Graphs have also been used locally, i.e. to call variants at the region level. GraphTyper<span class="citation" data-cites="ohTIiqfV">[<a href="#ref-ohTIiqfV">12</a>]</span> and BayesTyper<span class="citation" data-cites="14Uxmwbxm">[<a href="#ref-14Uxmwbxm">13</a>]</span> both construct variation graphs of small regions and use them for variant genotyping. Here again, the graph-approach showed clear advantages over standard approaches that use the linear reference. Other SV genotyping approaches compare read mapping in the reference genome and a sequence modified with the SV. For example SMRT-SV was designed to genotype SVs identified on PacBio reads<span class="citation" data-cites="rs7e40wC">[<a href="#ref-rs7e40wC">4</a>]</span>, SVTyper uses paired-end mapping and split-read mapping information<span class="citation" data-cites="AltPnocw">[<a href="#ref-AltPnocw">14</a>]</span>, and Delly provides a genotyping feature in addition to its discovery mode<span class="citation" data-cites="nLvQCjXU">[<a href="#ref-nLvQCjXU">15</a>]</span>.</p>
<h2 id="results">Results</h2>
<h2 id="methods">Methods</h2>
<h2 id="discussion">Discussion</h2>
<h2 id="references" class="page_break_before">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs">
<div id="ref-qA6dWFP">
<p>1. <strong>An integrated map of structural variation in 2,504 human genomes</strong><br />
Peter H. SudmantTobias Rausch, Eugene J. Gardner, Robert E. Handsaker, Alexej Abyzov, John Huddleston, Yan Zhang, Kai Ye, Goo Jun, … Jan O. Korbel<br />
<em>Nature</em> (2015-10) <a href="https://doi.org/73c">https://doi.org/73c</a><br />
DOI: <a href="https://doi.org/10.1038/nature15394">10.1038/nature15394</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26432246">26432246</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4617611">PMC4617611</a></p>
</div>
<div id="ref-py6BC5kj">
<p>2. <strong>Whole-genome sequence variation, population structure and demographic history of the Dutch population</strong><br />
Laurent C FrancioliAndroniki Menelaou, Sara L Pulit, Freerk van Dijk, Pier Francesco Palamara, Clara C Elbers, Pieter BT Neerincx, Kai Ye, Victor Guryev, … Cisca Wijmenga<br />
<em>Nature Genetics</em> (2014-06-29) <a href="https://doi.org/f6bxm8">https://doi.org/f6bxm8</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3021">10.1038/ng.3021</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24974849">24974849</a></p>
</div>
<div id="ref-z91V6jjU">
<p>3. <strong>Resolving the complexity of the human genome using single-molecule sequencing</strong><br />
Mark J. P. Chaisson, John Huddleston, Megan Y. Dennis, Peter H. Sudmant, Maika Malig, Fereydoun Hormozdiari, Francesca Antonacci, Urvashi Surti, Richard Sandstrom, Matthew Boitano, … Evan E. Eichler<br />
<em>Nature</em> (2014-11-10) <a href="https://doi.org/w69">https://doi.org/w69</a><br />
DOI: <a href="https://doi.org/10.1038/nature13907">10.1038/nature13907</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25383537">25383537</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4317254">PMC4317254</a></p>
</div>
<div id="ref-rs7e40wC">
<p>4. <strong>Discovery and genotyping of structural variation from long-read haploid genome sequence data</strong><br />
John Huddleston, Mark J.P. Chaisson, Karyn Meltz Steinberg, Wes Warren, Kendra Hoekzema, David Gordon, Tina A. Graves-Lindsay, Katherine M. Munson, Zev N. Kronenberg, Laura Vives, … Evan E. Eichler<br />
<em>Genome Research</em> (2016-11-28) <a href="https://doi.org/f9x79h">https://doi.org/f9x79h</a><br />
DOI: <a href="https://doi.org/10.1101/gr.214007.116">10.1101/gr.214007.116</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27895111">27895111</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411763">PMC5411763</a></p>
</div>
<div id="ref-PRx3qEIm">
<p>5. <strong>Mapping and phasing of structural variation in patient genomes using nanopore sequencing</strong><br />
Mircea Cretu Stancu, Markus J. van Roosmalen, Ivo Renkens, Marleen M. Nieboer, Sjors Middelkamp, Joep de Ligt, Giulia Pregno, Daniela Giachino, Giorgia Mandrile, Jose Espejo Valle-Inclan, … Wigard P. Kloosterman<br />
<em>Nature Communications</em> (2017-11-06) <a href="https://doi.org/gftpt9">https://doi.org/gftpt9</a><br />
DOI: <a href="https://doi.org/10.1038/s41467-017-01343-4">10.1038/s41467-017-01343-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29109544">29109544</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5673902">PMC5673902</a></p>
</div>
<div id="ref-121OWcTA4">
<p>6. <strong>Genome-wide reconstruction of complex structural variants using read clouds</strong><br />
Noah Spies, Ziming Weng, Alex Bishara, Jennifer McDaniel, David Catoe, Justin M Zook, Marc Salit, Robert B West, Serafim Batzoglou, Arend Sidow<br />
<em>Nature Methods</em> (2017-07-17) <a href="https://doi.org/gbnhwk">https://doi.org/gbnhwk</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.4366">10.1038/nmeth.4366</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28714986">28714986</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5578891">PMC5578891</a></p>
</div>
<div id="ref-6KbgcueR">
<p>7. <strong>Nanopore sequencing and assembly of a human genome with ultra-long reads</strong><br />
Miten Jain, Sergey Koren, Karen H Miga, Josh Quick, Arthur C Rand, Thomas A Sasani, John R Tyson, Andrew D Beggs, Alexander T Dilthey, Ian T Fiddes, … Matthew Loose<br />
<em>Nature Biotechnology</em> (2018-01-29) <a href="https://doi.org/gczffw">https://doi.org/gczffw</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.4060">10.1038/nbt.4060</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29431738">29431738</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5889714">PMC5889714</a></p>
</div>
<div id="ref-Pu6SY37C">
<p>8. <strong>Phased diploid genome assembly with single-molecule real-time sequencing</strong><br />
Chen-Shan Chin, Paul Peluso, Fritz J Sedlazeck, Maria Nattestad, Gregory T Concepcion, Alicia Clum, Christopher Dunn, Ronan O’Malley, Rosa Figueroa-Balderas, Abraham Morales-Cruz, … Michael C Schatz<br />
<em>Nature Methods</em> (2016-10-17) <a href="https://doi.org/f9fv4w">https://doi.org/f9fv4w</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.4035">10.1038/nmeth.4035</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27749838">27749838</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5503144">PMC5503144</a></p>
</div>
<div id="ref-Qa8mx6Ll">
<p>9. <strong>Genome graphs and the evolution of genome inference</strong><br />
Benedict Paten, Adam M. Novak, Jordan M. Eizenga, Erik Garrison<br />
<em>Genome Research</em> (2017-03-30) <a href="https://doi.org/f95nhd">https://doi.org/f95nhd</a><br />
DOI: <a href="https://doi.org/10.1101/gr.214155.116">10.1101/gr.214155.116</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28360232">28360232</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411762">PMC5411762</a></p>
</div>
<div id="ref-10jxt15v0">
<p>10. <strong>Variation graph toolkit improves read mapping by representing genetic variation in the reference</strong><br />
Erik Garrison, Jouni Sirén, Adam M Novak, Glenn Hickey, Jordan M Eizenga, Eric T Dawson, William Jones, Shilpa Garg, Charles Markello, Michael F Lin, … Richard Durbin<br />
<em>Nature Biotechnology</em> (2018-08-20) <a href="https://doi.org/gd2zqs">https://doi.org/gd2zqs</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.4227">10.1038/nbt.4227</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30125266">30125266</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6126949">PMC6126949</a></p>
</div>
<div id="ref-11Jy8B61m">
<p>11. <strong>Fast and accurate genomic analyses using genome graphs</strong><br />
Goran Rakocevic, Vladimir Semenyuk, Wan-Ping Lee, James Spencer, John Browning, Ivan J. Johnson, Vladan Arsenijevic, Jelena Nadj, Kaushik Ghose, Maria C. Suciu, … Deniz Kural<br />
<em>Nature Genetics</em> (2019-01-14) <a href="https://doi.org/gftd46">https://doi.org/gftd46</a><br />
DOI: <a href="https://doi.org/10.1038/s41588-018-0316-4">10.1038/s41588-018-0316-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30643257">30643257</a></p>
</div>
<div id="ref-ohTIiqfV">
<p>12. <strong>Graphtyper enables population-scale genotyping using pangenome graphs</strong><br />
Hannes P Eggertsson, Hakon Jonsson, Snaedis Kristmundsdottir, Eirikur Hjartarson, Birte Kehr, Gisli Masson, Florian Zink, Kristjan E Hjorleifsson, Aslaug Jonasdottir, Adalbjorg Jonasdottir, … Bjarni V Halldorsson<br />
<em>Nature Genetics</em> (2017-09-25) <a href="https://doi.org/gbx7v6">https://doi.org/gbx7v6</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3964">10.1038/ng.3964</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28945251">28945251</a></p>
</div>
<div id="ref-14Uxmwbxm">
<p>13. <strong>Accurate genotyping across variant classes and lengths using variant graphs</strong><br />
Jonas Andreas SibbesenLasse Maretty, Anders Krogh<br />
<em>Nature Genetics</em> (2018-06-18) <a href="https://doi.org/gdndnz">https://doi.org/gdndnz</a><br />
DOI: <a href="https://doi.org/10.1038/s41588-018-0145-5">10.1038/s41588-018-0145-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29915429">29915429</a></p>
</div>
<div id="ref-AltPnocw">
<p>14. <strong>SpeedSeq: ultra-fast personal genome analysis and interpretation</strong><br />
Colby Chiang, Ryan M Layer, Gregory G Faust, Michael R Lindberg, David B Rose, Erik P Garrison, Gabor T Marth, Aaron R Quinlan, Ira M Hall<br />
<em>Nature Methods</em> (2015-08-10) <a href="https://doi.org/gcpgfh">https://doi.org/gcpgfh</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.3505">10.1038/nmeth.3505</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26258291">26258291</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4589466">PMC4589466</a></p>
</div>
<div id="ref-nLvQCjXU">
<p>15. <strong>DELLY: structural variant discovery by integrated paired-end and split-read analysis</strong><br />
T. Rausch, T. Zichner, A. Schlattl, A. M. Stutz, V. Benes, J. O. Korbel<br />
<em>Bioinformatics</em> (2012-09-07) <a href="https://doi.org/f38r2c">https://doi.org/f38r2c</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/bts378">10.1093/bioinformatics/bts378</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22962449">22962449</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3436805">PMC3436805</a></p>
</div>
</div>
<script>
// AnchorJS minified version below.
// Source https://github.com/bryanbraun/anchorjs/blob/064abdd0987f305933ec4982af6d0c1cf2fd0814/anchor.js

/**
 * AnchorJS - v4.0.0 - 2017-06-02
 * https://github.com/bryanbraun/anchorjs
 * Copyright (c) 2017 Bryan Braun; Licensed MIT
 */
!function(A,e){"use strict";"function"==typeof define&&define.amd?define([],e):"object"==typeof module&&module.exports?module.exports=e():(A.AnchorJS=e(),A.anchors=new A.AnchorJS)}(this,function(){"use strict";function A(A){function e(A){A.icon=A.hasOwnProperty("icon")?A.icon:"",A.visible=A.hasOwnProperty("visible")?A.visible:"hover",A.placement=A.hasOwnProperty("placement")?A.placement:"right",A.class=A.hasOwnProperty("class")?A.class:"",A.truncate=A.hasOwnProperty("truncate")?Math.floor(A.truncate):64}function t(A){var e;if("string"==typeof A||A instanceof String)e=[].slice.call(document.querySelectorAll(A));else{if(!(Array.isArray(A)||A instanceof NodeList))throw new Error("The selector provided to AnchorJS was invalid.");e=[].slice.call(A)}return e}function n(){if(null===document.head.querySelector("style.anchorjs")){var A,e=document.createElement("style");e.className="anchorjs",e.appendChild(document.createTextNode("")),void 0===(A=document.head.querySelector('[rel="stylesheet"], style'))?document.head.appendChild(e):document.head.insertBefore(e,A),e.sheet.insertRule(" .anchorjs-link {   opacity: 0;   text-decoration: none;   -webkit-font-smoothing: antialiased;   -moz-osx-font-smoothing: grayscale; }",e.sheet.cssRules.length),e.sheet.insertRule(" *:hover > .anchorjs-link, .anchorjs-link:focus  {   opacity: 1; }",e.sheet.cssRules.length),e.sheet.insertRule(" [data-anchorjs-icon]::after {   content: attr(data-anchorjs-icon); }",e.sheet.cssRules.length),e.sheet.insertRule(' @font-face {   font-family: "anchorjs-icons";   src: url(data:n/a;base64,AAEAAAALAIAAAwAwT1MvMg8yG2cAAAE4AAAAYGNtYXDp3gC3AAABpAAAAExnYXNwAAAAEAAAA9wAAAAIZ2x5ZlQCcfwAAAH4AAABCGhlYWQHFvHyAAAAvAAAADZoaGVhBnACFwAAAPQAAAAkaG10eASAADEAAAGYAAAADGxvY2EACACEAAAB8AAAAAhtYXhwAAYAVwAAARgAAAAgbmFtZQGOH9cAAAMAAAAAunBvc3QAAwAAAAADvAAAACAAAQAAAAEAAHzE2p9fDzz1AAkEAAAAAADRecUWAAAAANQA6R8AAAAAAoACwAAAAAgAAgAAAAAAAAABAAADwP/AAAACgAAA/9MCrQABAAAAAAAAAAAAAAAAAAAAAwABAAAAAwBVAAIAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAMCQAGQAAUAAAKZAswAAACPApkCzAAAAesAMwEJAAAAAAAAAAAAAAAAAAAAARAAAAAAAAAAAAAAAAAAAAAAQAAg//0DwP/AAEADwABAAAAAAQAAAAAAAAAAAAAAIAAAAAAAAAIAAAACgAAxAAAAAwAAAAMAAAAcAAEAAwAAABwAAwABAAAAHAAEADAAAAAIAAgAAgAAACDpy//9//8AAAAg6cv//f///+EWNwADAAEAAAAAAAAAAAAAAAAACACEAAEAAAAAAAAAAAAAAAAxAAACAAQARAKAAsAAKwBUAAABIiYnJjQ3NzY2MzIWFxYUBwcGIicmNDc3NjQnJiYjIgYHBwYUFxYUBwYGIwciJicmNDc3NjIXFhQHBwYUFxYWMzI2Nzc2NCcmNDc2MhcWFAcHBgYjARQGDAUtLXoWOR8fORYtLTgKGwoKCjgaGg0gEhIgDXoaGgkJBQwHdR85Fi0tOAobCgoKOBoaDSASEiANehoaCQkKGwotLXoWOR8BMwUFLYEuehYXFxYugC44CQkKGwo4GkoaDQ0NDXoaShoKGwoFBe8XFi6ALjgJCQobCjgaShoNDQ0NehpKGgobCgoKLYEuehYXAAAADACWAAEAAAAAAAEACAAAAAEAAAAAAAIAAwAIAAEAAAAAAAMACAAAAAEAAAAAAAQACAAAAAEAAAAAAAUAAQALAAEAAAAAAAYACAAAAAMAAQQJAAEAEAAMAAMAAQQJAAIABgAcAAMAAQQJAAMAEAAMAAMAAQQJAAQAEAAMAAMAAQQJAAUAAgAiAAMAAQQJAAYAEAAMYW5jaG9yanM0MDBAAGEAbgBjAGgAbwByAGoAcwA0ADAAMABAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAH//wAP) format("truetype"); }',e.sheet.cssRules.length)}}this.options=A||{},this.elements=[],e(this.options),this.isTouchDevice=function(){return!!("ontouchstart"in window||window.DocumentTouch&&document instanceof DocumentTouch)},this.add=function(A){var i,o,s,c,r,a,h,l,u,d,f,g,p=[];if(e(this.options),"touch"===(g=this.options.visible)&&(g=this.isTouchDevice()?"always":"hover"),A||(A="h2, h3, h4, h5, h6"),0===(i=t(A)).length)return this;for(n(),o=document.querySelectorAll("[id]"),s=[].map.call(o,function(A){return A.id}),r=0;r<i.length;r++)if(this.hasAnchorJSLink(i[r]))p.push(r);else{if(i[r].hasAttribute("id"))c=i[r].getAttribute("id");else if(i[r].hasAttribute("data-anchor-id"))c=i[r].getAttribute("data-anchor-id");else{u=l=this.urlify(i[r].textContent),h=0;do{void 0!==a&&(u=l+"-"+h),a=s.indexOf(u),h+=1}while(-1!==a);a=void 0,s.push(u),i[r].setAttribute("id",u),c=u}d=c.replace(/-/g," "),(f=document.createElement("a")).className="anchorjs-link "+this.options.class,f.href="#"+c,f.setAttribute("aria-label","Anchor link for: "+d),f.setAttribute("data-anchorjs-icon",this.options.icon),"always"===g&&(f.style.opacity="1"),""===this.options.icon&&(f.style.font="1em/1 anchorjs-icons","left"===this.options.placement&&(f.style.lineHeight="inherit")),"left"===this.options.placement?(f.style.position="absolute",f.style.marginLeft="-1em",f.style.paddingRight="0.5em",i[r].insertBefore(f,i[r].firstChild)):(f.style.paddingLeft="0.375em",i[r].appendChild(f))}for(r=0;r<p.length;r++)i.splice(p[r]-r,1);return this.elements=this.elements.concat(i),this},this.remove=function(A){for(var e,n,i=t(A),o=0;o<i.length;o++)(n=i[o].querySelector(".anchorjs-link"))&&(-1!==(e=this.elements.indexOf(i[o]))&&this.elements.splice(e,1),i[o].removeChild(n));return this},this.removeAll=function(){this.remove(this.elements)},this.urlify=function(A){var t=/[& +$,:;=?@"#{}|^~[`%!'<>\]\.\/\(\)\*\\]/g;return this.options.truncate||e(this.options),A.trim().replace(/\'/gi,"").replace(t,"-").replace(/-{2,}/g,"-").substring(0,this.options.truncate).replace(/^-+|-+$/gm,"").toLowerCase()},this.hasAnchorJSLink=function(A){var e=A.firstChild&&(" "+A.firstChild.className+" ").indexOf(" anchorjs-link ")>-1,t=A.lastChild&&(" "+A.lastChild.className+" ").indexOf(" anchorjs-link ")>-1;return e||t||!1}}return A});

// Enable links for selected headers
var anchors = new AnchorJS();
anchors.add("h2, h3, h4")
</script>
<!-- Enable Hypothesis annotation. https://web.hypothes.is/ -->
<script src="https://hypothes.is/embed.js" async></script>
</body>
</html>
