<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Glenn Hickey" />
  <meta name="author" content="David Heller" />
  <meta name="author" content="Jean Monlong" />
  <meta name="author" content="Jonas Andreas Sibbesen" />
  <meta name="author" content="Adam Novak" />
  <meta name="author" content="Benedict Paten" />
  <meta name="dcterms.date" content="2019-04-10" />
  <meta name="keywords" content="structural variation, pangenome, variant genotyping" />
  <title>Genotyping structural variation in variation graphs with the vg toolkit</title>
  <style type="text/css">
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      span.underline{text-decoration: underline;}
      div.column{display: inline-block; vertical-align: top; width: 50%;}
  </style>
  <link rel="stylesheet" href="github-pandoc.css" />
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!-- Insert Analytics Script Below -->
  <script>
  </script>
  <!-- End Analytics Script -->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Genotyping structural variation in variation graphs with the vg toolkit</h1>
</header>
<p><small><em> This manuscript (<a href="https://jmonlong.github.io/manu-vgsv/v/c53a6ec62c0721f0f3381dcd2daf8c7890a6adba/">permalink</a>) was automatically generated from <a href="https://github.com/jmonlong/manu-vgsv/tree/c53a6ec62c0721f0f3381dcd2daf8c7890a6adba">jmonlong/manu-vgsv@c53a6ec</a> on April 10, 2019. </em></small></p>
<h2 id="authors">Authors</h2>
<p><a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Glenn Hickey<sup>1,☯</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> David Heller<sup>1,2,☯</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Jean Monlong<sup>1,☯</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Jonas Andreas Sibbesen<sup>1</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Adam Novak<sup>1</sup>, <a href="https://orcid.org/XXXX-XXXX-XXXX-XXXX"><img src="images/orcid.svg" alt="ORCID icon" width="11" height="11" /></a> Benedict Paten<sup>1,†</sup></p>
<p><sup>☯</sup> — These authors contributed equally to this work</p>
<p><sup>†</sup> — To whom correspondence should be addressed: bpaten@ucsc.edu <small></p>
<ol type="1">
<li>UC Santa Cruz Genomics Institute, University of California, Santa Cruz, California, USA</li>
<li>Max Planck Institute for Molecular Genetics, Berlin, Germany</li>
</ol>
<p></small></p>
<h2 id="abstract" class="page_break_before">Abstract</h2>
<p>Variation graphs have been proposed to represent human pan-genomes, offering reduced reference bias and better mapping accuracy than linear reference genomes. Structural variants (SVs) are significant components of genetic diversity and have been associated with diseases, but the technological challenges surrounding their representation and identification make their study difficult relative to point mutations. Still, thousands of SVs have been characterized and better catalogs continue to be generated thanks to new technologies. We contend that variation graphs provide the most elegant representation of such catalogs and an effective means for leveraging them for short-read SV genotyping experiments. In this work, we extended the vg toolkit for working with variation graphs to support SV genotyping, and showed that it is capable of genotyping insertions, deletions and inversions, even in the presence of small errors in the variant definition. We then benchmarked vg across three high-quality sequence-resolved SV catalogs generated by recent studies. vg was compared to state-of-the-art SV genotypers using simulated and real Illumina short reads. On real data, vg produced the best genotype predictions systematically in all datasets. Finally, we found that graphs constructed directly from aligned de novo assemblies can improve genotyping compared to graphs built from intermediate SV catalogs in the VCF format. We experimented with assemblies from 12 yeast strains and showed that the incorporation of only a few assemblies in the graph sufficiently represents genetic variation across all strains. Our results demonstrate the power of variation graphs for SV genotyping. Beyond single nucleotide variants and short insertions/deletions, the vg toolkit now includes SV in its unified variant calling framework and provides a natural solution to integrate high-quality SV catalogs and assemblies.</p>
<h2 id="introduction" class="page_break_before">Introduction</h2>
<p>A structural variant (SV) is a genomic mutation involving 50 or more base pairs. SVs can take several forms such as deletions, insertions, inversions, or translocations. It is intuitive that their greater size relative to smaller events such as single nucleotide variants (SNVs) and small insertions and deletions (indels) would imply that SVs can have a larger impact on phenotype. Indeed, SVs have long been associated with developmental disorders, cancer and other complex diseases and phenotypes<span class="citation" data-cites="ebc66eBr">[<a href="#ref-ebc66eBr">1</a>]</span>, but remain much more poorly studied than their smaller mutational counterparts. This discrepancy stems technological limitation: short read sequencing has provided the basis of most modern genome sequencing studies due to its high base-level accuracy and relatively low cost. However, it is poorly suited for discovering SVs, which often lie in repeat-rich regions and whose lengths can far exceed read sizes. The central obstacle is in mapping short reads to the human reference genome. It is generally difficult or impossible to unambiguously map a short read if the sample whose genome is being analyzed differs substantially from the reference at the read’s location. If, for example, the read derives from the middle of what would be termed a large insertion relative to the reference, the best result a read mapper could hope to produce would be to leave it unmapped. In some cases, SVs can be inferred indirectly from short read mappings via split reads, discordant read pairs, or read depth, but the accuracy of such approaches remains limited<span class="citation" data-cites="vQTymKCj">[<a href="#ref-vQTymKCj">2</a>]</span>. Assembling deeply sequenced short reads and aligning the contigs or scaffolds back to the reference provides an effective means of SV discovery at increased computational and sequencing cost <span class="citation" data-cites="pJAv1D8R">[<a href="#ref-pJAv1D8R">3</a>]</span>.</p>
<p>In practice, large-scale projects must often combine short read mapping to several other methods to achieve better accuracy. This methodology has been used to compile catalogs with tens of thousands of SVs in humans<span class="citation" data-cites="qA6dWFP py6BC5kj">[<a href="#ref-qA6dWFP">4</a>,<a href="#ref-py6BC5kj">5</a>]</span>. Recent developments in sequencing technologies and library preparation continue to pave the way for better SV detection. Several recent studies using long-read or linked-read sequencing have produced large catalogs of structural variation, the majority of which was novel and sequence-resolved<span class="citation" data-cites="z91V6jjU rs7e40wC PRx3qEIm 121OWcTA4 3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>,<a href="#ref-z91V6jjU">6</a>,<a href="#ref-rs7e40wC">7</a>,<a href="#ref-PRx3qEIm">8</a>,<a href="#ref-121OWcTA4">9</a>]</span>. These technologies are also enabling the production of high-quality de novo genome assemblies<span class="citation" data-cites="z91V6jjU 6KbgcueR">[<a href="#ref-6KbgcueR">11</a>,<a href="#ref-z91V6jjU">6</a>]</span>, and large blocks of haplotype-resolved sequences<span class="citation" data-cites="Pu6SY37C">[<a href="#ref-Pu6SY37C">12</a>]</span>. Such technical advances promise to expand the amount of known genomic variation in humans in the near future. However, their cost prohibits their use in large-scale studies that require hundreds or thousands of samples, such as disease association studies.</p>
<p>Representing structural variation poses an additional challenge. Standard formats such as VCF are unweildly when used for SVs due to ambiguity in their specification and limited support across tools. Incorporating SVs into a linear pan-genome reference via alt contigs involves custom pipelines and non-standard metadata formats linking the contigs back to the reference. They often contain redundant sequence. These issues make them difficult to use, maintain and scale as SV catalogs continue to expand. [todo: cite? even just grch38 is a pretty good case study of how awful this is] Development of pan-genome graph reference representations offers a solution to such problems <span class="citation" data-cites="Qa8mx6Ll">[<a href="#ref-Qa8mx6Ll">13</a>]</span>.<br />
The graph structure enables SVs to be as succinctly and directly represented as point mutations.<br />
By including variants in the graph, both read mapping and variant calling become variant-aware and benefit in terms of accuracy and sensitivity<span class="citation" data-cites="10jxt15v0 DuODeStx 11Jy8B61m">[<a href="#ref-10jxt15v0">14</a>,<a href="#ref-DuODeStx">15</a>,<a href="#ref-11Jy8B61m">16</a>]</span>. In addition, different variant types are called simultaneously by a unified framework. vg is the first openly available variation graph tool to scale to multi-gigabase genomes and provides read mapping, variant calling and haplotype modeling<span class="citation" data-cites="10jxt15v0">[<a href="#ref-10jxt15v0">14</a>]</span>. In vg, graphs can be built from both variant catalogs in the VCF format or from assembly alignments. Other genome graph implementations have also been used specifically to genotype variants. Using a sliding-window approach, GraphTyper realigns reads to a graph built from known SNVs and short indels<span class="citation" data-cites="ohTIiqfV">[<a href="#ref-ohTIiqfV">17</a>]</span>. BayesTyper builds graphs with both short variants and SVs, and genotypes variants based on the k-mer distribution of sequencing reads<span class="citation" data-cites="14Uxmwbxm">[<a href="#ref-14Uxmwbxm">18</a>]</span>. Here again, the graph-based approaches showed clear advantages over standard methods that use the linear reference.</p>
<p>Other SV genotyping approaches typically compare read mapping to the reference genome and to a sequence modified with the SV. For example SMRT-SV was designed to genotype SVs identified on PacBio reads<span class="citation" data-cites="rs7e40wC">[<a href="#ref-rs7e40wC">7</a>]</span>. SVTyper uses paired-end mapping and split-read mapping information and can genotype deletions, duplications, inversions and translocations<span class="citation" data-cites="AltPnocw">[<a href="#ref-AltPnocw">19</a>]</span>. Delly provides a genotyping feature in addition to its discovery mode and can genotype all types of SVs although the VCF needs special formatting for some<span class="citation" data-cites="nLvQCjXU">[<a href="#ref-nLvQCjXU">20</a>]</span>. SMRT-SV2 is a machine learning tool that was trained to genotype SVs from the alignment of reads to the reference genome augmented with SV-containing sequences as alternate contigs<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span>. [todo: I think we need to expand on the description of these methods, especially smrt-sv. I will take a stab in a future pass]</p>
<p>We show that the unified variant calling framework implemented in vg is capable of genotyping deletions, insertions and inversions. We compare vg with state-of-the-art SV genotypers: SVTyper<span class="citation" data-cites="AltPnocw">[<a href="#ref-AltPnocw">19</a>]</span>, Delly<span class="citation" data-cites="nLvQCjXU">[<a href="#ref-nLvQCjXU">20</a>]</span>, BayesTyper<span class="citation" data-cites="14Uxmwbxm">[<a href="#ref-14Uxmwbxm">18</a>]</span> and SMRT-SV2<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span>. On simulated variants, vg is robust to small errors in the breakpoint location and outperforms most other methods on shallow sequencing experiments. Starting from SVs discovered in recent long-read sequencing studies<span class="citation" data-cites="3NNFS6U2 vQTymKCj 14neTdqfN 16GvGhO20">[<a href="#ref-3NNFS6U2">10</a>,<a href="#ref-vQTymKCj">2</a>,<a href="#ref-14neTdqfN">21</a>,<a href="#ref-16GvGhO20">22</a>]</span>, we evaluated the genotyping accuracy when using simulated or real Illumina reads. Across all three datasets that we tested, vg is the best performing SV genotyper on real short-read data for all SV types. Going further, we show that building graphs from the alignment of de novo assemblies leads to better genotyping performance.</p>
<h2 id="results">Results</h2>
<h3 id="structural-variation-in-vg">Structural variation in vg</h3>
<p>In addition to SNVs and short indels, vg can handle large deletions, insertions and inversions (Figure <a href="#fig:1">1</a>a). As a proof-of-concept we simulated genomes and different types of SVs with a size distribution matching real SVs<span class="citation" data-cites="vQTymKCj">[<a href="#ref-vQTymKCj">2</a>]</span>. We compared vg against SVTyper, Delly and BayesTyper across different levels of sequencing depth. Some errors were also added at the breakpoints to investigate their effect on genotyping accuracy (see <a href="#simulation-experiment">Methods</a>). The results are shown in Figure <a href="#fig:1">1</a>b. When using the correct breakpoints, vg tied with Delly as the best genotyper for deletions, and with BayesTyper as the best genotyper for insertions. For inversions, vg was the second best genotyper after BayesTyper. The differences between the methods were the most visible at lower sequencing depth. In the presence of 1-10 bp errors in the breakpoint definition, the performance of Delly and BayesTyper dropped significantly. The dramatic drop for BayesTyper can be explained by its khmer-based approach that requires exact SV definition. In contrast, vg was only slightly affected by the presence of errors in the input VCF (Figure <a href="#fig:1">1</a>b). For vg, the F1 scores for all SV types decreased no more than 0.07 point. Overall, these results show that vg is capable of genotyping SVs and is robust to errors in the input VCF.</p>
<figure>
<img src="images/panel1.png" alt="Figure 1: Structural variation in vg. a) Adding large insertions, deletions and inversions in a variation graph. b) Simulation experiment. For each experiment (method, depth and input VCF with/without errors), the y-axis shows the maximum F1 across different minimum quality thresholds." id="fig:1" /><figcaption><span>Figure 1:</span> <strong>Structural variation in vg.</strong> a) Adding large insertions, deletions and inversions in a variation graph. b) Simulation experiment. For each experiment (method, depth and input VCF with/without errors), the y-axis shows the maximum F1 across different minimum quality thresholds.</figcaption>
</figure>
<h3 id="hgsvc-dataset">HGSVC dataset</h3>
<p>The Human Genome Structural Variation Consortium (HGSVC) generated a high-quality SV catalog from three samples, obtained using a consensus from different sequencing, phasing and variant calling technologies<span class="citation" data-cites="vQTymKCj">[<a href="#ref-vQTymKCj">2</a>]</span>. The three samples come from different human populations: a Han Chinese individual (HG00514), a Puerto-Rican individual (HG00733), and a Yoruban Nigerian individual (NA19240). These SVs were used to construct a graph with vg and as input for the other genotypers. SVs were genotyped from short reads and compared with the original catalog (see <a href="#hgsvc-analysis">Methods</a>).</p>
<p>First, by simulating reads for HG00514, we compared the different methods in the ideal situation where the SV catalog is correct and matches exactly the SVs supported by the reads. While vg outperformed Delly and SVTyper, BayesTyper showed the best F1 score and precision-recall trade-off (Figures <a href="#fig:2">2</a> and <a href="#fig:hgsvc-sim-geno">S1</a>, Table <a href="#tbl:hgsvc">S1</a>). When restricting the comparisons to regions not identified as tandem repeats or segmental duplications, the genotyping predictions were significantly better for all methods, with vg almost as good as BayesTyper on deletions (F1 of 0.944 vs 0.955). We observed similar results when evaluating the absence/presence of a SV instead of the exact genotype (Figures <a href="#fig:2">2</a> and <a href="#fig:hgsvc-sim">S2</a>). Overall, both graph-based methods, vg and BayesTyper, outperformed the other two methods tested.</p>
<p>We then repeated the analysis using real Illumina reads from HG00514, to benchmark the methods on a more realistic experiment. Here vg clearly outperformed other approaches, most likely because of its graph-based strategy and robustness to errors in the SV catalog (Figures <a href="#fig:2">2</a> and <a href="#fig:hgsvc-real-geno">S3</a>). In non-repeat regions and across the whole genome, the F1 scores and precision-recall curves were higher for vg compared to other methods. For example, for deletions in non-repeat regions, the F1 score for vg was 0.801 while the second best method, Delly, had a F1 score of 0.692. We observed similar results when evaluating the absence/presence of a SV instead of the exact genotype (Figures <a href="#fig:2">2</a> and <a href="#fig:hgsvc-real">S4</a>).</p>
<figure>
<img src="images/hgsvc-giab-best-f1.png" alt="Figure 2: Structural variants from the HGSVC and Genome in a Bottle datasets. HGSVC: Simulated and real reads were used to genotype SVs and compared with the high-quality calls from Chaisson et al.[2]. Reads were simulated from the HG00514 individual. Using real reads, the three HG00514, HG00733, and NA19240 individuals were tested. GiaB: Real reads from the HG002 individual were used to genotype SVs and compared with the high-quality calls from the Genome in a Bottle consortium[21,22]. Maximum F1 score for each method (color), across the whole genome or focusing on non-repeat regions (x-axis). The calling and genotyping evaluation are shown with different shades." id="fig:2" /><figcaption><span>Figure 2:</span> <strong>Structural variants from the HGSVC and Genome in a Bottle datasets</strong>. HGSVC: Simulated and real reads were used to genotype SVs and compared with the high-quality calls from Chaisson et al.<span class="citation" data-cites="vQTymKCj">[<a href="#ref-vQTymKCj">2</a>]</span>. Reads were simulated from the HG00514 individual. Using real reads, the three HG00514, HG00733, and NA19240 individuals were tested. GiaB: Real reads from the HG002 individual were used to genotype SVs and compared with the high-quality calls from the Genome in a Bottle consortium<span class="citation" data-cites="14neTdqfN 16GvGhO20">[<a href="#ref-14neTdqfN">21</a>,<a href="#ref-16GvGhO20">22</a>]</span>. Maximum F1 score for each method (color), across the whole genome or focusing on non-repeat regions (x-axis). The calling and genotyping evaluation are shown with different shades.</figcaption>
</figure>
<h3 id="other-long-read-datasets">Other long-read datasets</h3>
<p>The Genome in a Bottle (GiaB) consortium is currently producing a high-quality SV catalog for a Ashkenazim individual (HG002)<span class="citation" data-cites="14neTdqfN 16GvGhO20">[<a href="#ref-14neTdqfN">21</a>,<a href="#ref-16GvGhO20">22</a>]</span>. Dozens of SV callers and datasets from short, long and linked reads were used to produce this set of SVs. vg performed similarly on this dataset as on the HGSVC dataset, with a F1 score of 0.75 for both insertions and deletions in non-repeat regions (Figures <a href="#fig:2">2</a>, <a href="#fig:giab-geno">S5</a> and <a href="#fig:giab">S6</a>, and Table <a href="#tbl:giab">S2</a>). As before, other methods produced lower F1 scores in most cases, although Delly and BayesTyper predicted better genotype for deletions in non-repeat regions. Because this catalog contained SVs for only one individual, this benchmark is mostly relevant to compare the recall which is what contributes most to the F1 score differences (Figure <a href="#fig:giab">S6</a>).</p>
<p>A recent study by Audano et al. generated a SV catalog using long-read sequencing across 15 individuals <span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span>. These variants were then genotyped from short reads across 440 individuals using SMRT-SV2, a machine-learning genotyper implemented for this study. We first called SVs from the pseudo-diploid genome and reads used to train SMRT-SV2 and constructed by merging datasets from two haploid cell lines<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span>. The absence/presence predictions from vg were systematically better than SMRT-SV2 for both SV types but SMRT-SV2 produced better genotypes for deletions (see Figures <a href="#fig:chmpd-svpop">3</a>, <a href="#fig:chmpd-geno">S7</a> and <a href="#fig:chmpd">S8</a>, and Table <a href="#tbl:chmpd">S3</a>). Using publicly available Illumina reads, we then genotyped SVs in 3 of the 15 individuals that were used for discovery in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span>. Compared to SMRT-SV2, vg had a better precision-recall curve and a higher F1 for both insertions and deletions (Figures <a href="#fig:chmpd-svpop">3</a> and <a href="#fig:svpop">S9</a>, and Table <a href="#tbl:svpop">S4</a>). Of note, SMRT-SV2 produces <em>no-calls</em> when the read coverage is too low and we observed that its recall increased when filtering these regions out. Interestingly, vg performed well even in regions where SMRT-SV2 produced <em>no-calls</em> (Figure <a href="#fig:svpop-regions">S10</a> and Table <a href="#tbl:svpop-regions">S5</a>). Finally, Audano et al. had identified 217 sequence-resolved inversions. vg correctly predicted the presence of around 14% of the inversions present in the three samples (Table <a href="#tbl:svpop">S4</a>). Inversions are often complex, harboring additional variation that makes their characterization and genotyping challenging.</p>
<figure>
<img src="images/chmpd-svpop-best-f1.png" alt="Figure 3: Structural variants from Audano et al.[10]. The pseudo-diploid genome built from CHM cell lines was originally used to train SMRT-SV2 in Audano et al.[10]. The SVPOP panel shows the combined results for the HG5014, HG00733 and NA19240 individuals, 3 of the 15 individuals used to generate the high-quality SV catalog in Audano et al.[10]. Maximum F1 score for each method (color), across the whole genome or focusing on non-repeat regions (x-axis). The calling and genotyping evaluation are shown with different shades." id="fig:chmpd-svpop" /><figcaption><span>Figure 3:</span> <strong>Structural variants from Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span></strong>. The pseudo-diploid genome built from CHM cell lines was originally used to train SMRT-SV2 in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span>. The SVPOP panel shows the combined results for the HG5014, HG00733 and NA19240 individuals, 3 of the 15 individuals used to generate the high-quality SV catalog in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span>. Maximum F1 score for each method (color), across the whole genome or focusing on non-repeat regions (x-axis). The calling and genotyping evaluation are shown with different shades.</figcaption>
</figure>
<h3 id="breakpoint-fine-tuning">Breakpoint fine-tuning</h3>
<p><em>(Maybe better in discussion.)</em></p>
<p>In addition to genotyping, vg can use an augmentation step to modify the graph based on the read alignment and discover novel variants. On the simulated SVs from Figure <a href="#fig:1">1</a>b, this approach was able to correct many of the 1-10 bp breakpoint errors injected in the input VCF. The breakpoints were accurately fine-tuned for 93.8% of the insertions (Figure <a href="#fig:simerror-bkpt">S11</a>a and Table <a href="#tbl:simerror-bkpt">S6</a>). For deletions, 78.1% of the variants were corrected when only one breakpoint had an error. In situations where both breakpoints of the deletions were incorrect, only 18.6% were corrected through graph augmentation, and only when the amount of error was small (Figure <a href="#fig:simerror-bkpt">S11</a>b). The breakpoints of less than 20% of the inversions could be corrected. Across all SV types, the size of the variant didn’t affect the ability to fine-tune the breakpoints through graph augmentation (Figure <a href="#fig:simerror-bkpt">S11</a>c).</p>
<h3 id="graphs-from-alignment-of-de-novo-assemblies">Graphs from alignment of de novo assemblies</h3>
<p>We investigated whether genome graphs derived from de-novo assembly alignments yield advantages for SV genotyping. To this end, we analyzed public sequencing datasets for 12 yeast strains from two related clades (<em>S. cerevisiae</em> and <em>S. paradoxus</em>) <span class="citation" data-cites="7f5OKa5O">[<a href="#ref-7f5OKa5O">23</a>]</span>. By generating genome graphs from only five of the strains we could measure how well variation from a small subset of strains represents the variation present in the full set of 12 strains. We generated and compared two different types of genome graphs. The first graph type (in the following called <em>construct graph</em>) was created from a linear reference genome of the <em>S.c. S288C</em> strain and a set of SVs relative to this reference strain in VCF format. We compiled the SV set using the output of three methods for SV detection from genome assemblies: Assemblytics <span class="citation" data-cites="krO7WgVi">[<a href="#ref-krO7WgVi">24</a>]</span>, AsmVar <span class="citation" data-cites="oVaXIwl5">[<a href="#ref-oVaXIwl5">25</a>]</span> and paftools <span class="citation" data-cites="172cJaw4Q">[<a href="#ref-172cJaw4Q">26</a>]</span>. All three methods were run to detect SVs between the reference strain and each of the other strains. Merging the results from the three methods and four of the eleven strains provided us with a representative set of SVs occurring in the two yeast clades that we could use to construct the <em>construct graph</em>. The second graph (in the following called <em>cactus graph</em>) was derived from a multiple genome alignment of the five strains using our Cactus tool <span class="citation" data-cites="1FgS53pXi">[<a href="#ref-1FgS53pXi">27</a>]</span>. While the <em>construct graph</em> is still mainly linear and highly dependent on the reference genome, the <em>cactus graph</em> is completely unbiased in that regard.</p>
<figure>
<img src="images/panel3.png" alt="Figure 4: Mapping comparison. Short reads from all 12 yeast strains were aligned to both graphs. The fraction of reads mapped to the cactus graph (y-axis) and the construct graph (x-axis) are compared. a) Stratified by mapping quality threshold. b) Stratified by percent identity threshold. Colors and shapes represent the 12 strains and two clades, respectively. Transparency indicates whether the strain was included or excluded in the graphs." id="fig:3" /><figcaption><span>Figure 4:</span> <strong>Mapping comparison.</strong> Short reads from all 12 yeast strains were aligned to both graphs. The fraction of reads mapped to the cactus graph (y-axis) and the construct graph (x-axis) are compared. a) Stratified by mapping quality threshold. b) Stratified by percent identity threshold. Colors and shapes represent the 12 strains and two clades, respectively. Transparency indicates whether the strain was included or excluded in the graphs.</figcaption>
</figure>
<p>First, we tested our hypothesis that the <em>cactus graph</em> has higher mappability due to its better representation of sequence diversity among the yeast strains. Figure <a href="#fig:3">4</a>a shows the fraction of Illumina reads from the 12 strains that was mapped with a mapping quality above a certain threshold to the <em>cactus graph</em> and to the <em>construct graph</em>. Generally, more reads were mapped to the <em>cactus graph</em> than to the <em>construct graph</em> regardless of the chosen mapping quality threshold. Only for the reference strain <em>S.c. S288C</em>, the <em>construct graph</em> exhibited slightly better mappability. This suggests that the improvement in mappability is not driven by the higher sequence content in the <em>cactus graph</em> alone (15.4 Mb compared to 12.4 Mb in the <em>construct graph</em>). Instead, our measurements suggest that the genetic distance to the reference strain correlates with the better mapping on the <em>cactus graph</em> over the <em>construct graph</em>. Consequently, the benefit of using the <em>cactus graph</em> is largest for strains in the <em>S. paradoxus</em> clade and smaller for reads from strains in the <em>S. cerevisiae</em> clade.</p>
<p>We observed a similar trend when exploring the mapping identity of the short reads on the graphs (see Figure <a href="#fig:3">4</a>b). For strains in the <em>S. paradoxus</em> clade, the <em>cactus graph</em> resulted in substantially more mappings with high percent identity than the <em>construct graph</em>. With strains in the <em>S. cerevisiae</em> clade, the difference was smaller, at least for a percent identity threshold up to 90%. When comparing read fractions with perfect identity (i.e. percent identity threshold = 100%), the <em>cactus graph</em> clearly outperforms the <em>construct graph</em> on 11 out of 12 samples, the exception again being the reference strain <em>S.c. S288C</em>.</p>
<p>Interestingly, our measurements did not show a substantial difference between strains included in the graph and excluded strains. The results suggest that two strains from each clade as well as the reference strain are sufficient to capture most of the genetic variation among all the strains. Only the number of alignments with perfect identity is substantially lower for the strains that were not included in the creation of the graphs (see Figure <a href="#fig:3">4</a>b). For a direct comparison, see Figure <a href="#fig:panel5">S12</a> which shows results of the same experiment on graphs generated from all 12 strains.</p>
<p>Next, we compared the SV genotype performance of both graphs. We mapped short reads from the 11 non-reference strains to both graphs and called variants for each strain using the vg toolkit’s variant calling module (see <a href="#toil-vg-call">Methods</a>). In the absence of a gold standard, we evaluated each callset based on the alignment of reads to a <em>sample graph</em> constructed from the callset (see <a href="#calling-and-genotyping-of-svs">Methods</a>). If a given callset is correct, we expect that reads from the same sample will be mapped confidently and with high identity to the corresponding sample graph. Therefore, we compared the average mapping quality and percent identity of the short reads on each sample graph (see Figures <a href="#fig:4">5</a>a and b). Similar to the results of our mapping analysis above, the <em>cactus graph</em> clearly outperformed the <em>construct graph</em> for strains in the <em>S. paradoxus</em> clade and performed slightly better for strains in the <em>S. cerevisiae</em> clade. Again, our measurements did not show a large difference between strains included in the graph and those that were excluded. For a direct comparison, see Figure <a href="#fig:panel6">S13</a> which shows results of the same experiment on graphs generated from all 12 strains.</p>
<figure>
<img src="images/panel4.png" alt="Figure 5: SV genotyping comparison. Short reads from all 11 non-reference yeast strains were used to genotype SVs contained in both graphs. Subsequently, sample graphs were generated from the resulting SV callsets. The short reads were again aligned to the sample graphs and the quality of the alignments was used to ascertain genotyping performance. a) Average mapping quality of short reads aligned to the sample graphs derived from cactus graph (y-axis) and construct graph (x-axis). b) Average mapping identity of short reads aligned to the sample graphs derived from cactus graph (y-axis) and construct graph (x-axis). Colors and shapes represent the 11 non-reference strains and two clades, respectively. Transparency indicates whether the strain was included or excluded in the graphs." id="fig:4" /><figcaption><span>Figure 5:</span> <strong>SV genotyping comparison.</strong> Short reads from all 11 non-reference yeast strains were used to genotype SVs contained in both graphs. Subsequently, sample graphs were generated from the resulting SV callsets. The short reads were again aligned to the sample graphs and the quality of the alignments was used to ascertain genotyping performance. a) Average mapping quality of short reads aligned to the sample graphs derived from <em>cactus graph</em> (y-axis) and <em>construct graph</em> (x-axis). b) Average mapping identity of short reads aligned to the sample graphs derived from <em>cactus graph</em> (y-axis) and <em>construct graph</em> (x-axis). Colors and shapes represent the 11 non-reference strains and two clades, respectively. Transparency indicates whether the strain was included or excluded in the graphs.</figcaption>
</figure>
<h2 id="methods">Methods</h2>
<h3 id="toil-vg">Toil-vg</h3>
<p>Toil-vg is a set of Python scripts for simplifying vg tasks such as graph construction, read mapping and SV genotyping. It uses the Toil workflow engine <span class="citation" data-cites="faeC2cx0">[<a href="#ref-faeC2cx0">28</a>]</span> to seamlessly run pipelines locally, on clusters or on the cloud. All variation graph analysis in this report was done using toil-vg, with the exact commands available <a href="https://github.com/glennhickey/hgsvc">here</a> [todo: get everything in one place with reasonable docs]. The principal toil-vg commands used are described below.</p>
<h4 id="toil-vg-construct">Toil-vg construct</h4>
<p>Toil-vg construct automates graph construction and indexing following the best practices put forth by the vg community. Graph construction is parallelized across different sequences from the reference fasta, and different whole-genome indexes are created side by side when possible. Phasing information from the input VCF can be used when available to preserve haplotypes in the GCSA2 pruning step, as well as to extract haploid sequences to simulate from.</p>
<h4 id="toil-vg-map">Toil-vg map</h4>
<p>Toil-vg map splits the input reads into batches, maps each batch in parallel, then merges the result.</p>
<h4 id="toil-vg-call">Toil-vg call</h4>
<p>A simple though very general variant caller has been implemented as <code>vg call</code>. Here it is used to genotype structural variants already present in the graph, but the same algorithm can also be used for smaller variants such as SNPs, as well as making de-novo calls. The algorithm is as follows:</p>
<ol type="1">
<li>The average read support for each node and edge, adjusted for mapping and base quality, is computed. The graph can optionally be augmented to include new variation from the reads.</li>
<li>The graph is then decomposed into snarls. Briefly, a snarl is a subgraph defined by two end nodes, where cutting the graph at these nodes disconnects the snarl from the rest of the graph. (todo: work on way to define this without getting into bidirected graphs, or is that a lost cause?). Snarls can be nested inside other snarls, and this nesting hierarchy forms a forest (todo: I don’t think chains get used anywhere in vg call so we can ignore here). The snarl decomposition is a fundamental structure for identifying variants in a graph and were formally defined by Paten et al.<span class="citation" data-cites="xJlNnKH2">[<a href="#ref-xJlNnKH2">29</a>]</span>, along with an algorithm to identify them.</li>
<li>Root-level snarls from the decomposition are considered independently and in parallel. Only snarls whose two ends lie on a reference (i.e. chromosome) path are considered as the VCF format used for output requires reference positions. The following steps are performed on each root snarl.
<ol type="1">
<li>A set of paths between the snarls end nodes are computed. (todo, consult with Adam about writing up the RepresentativeTraversalFinder)</li>
<li>The paths are ranked according to their average support.</li>
<li>A genotype is determined using the relative support of the best traversals, as well as the background read depth.</li>
<li>The VCF variants are derived from the paths.</li>
</ol></li>
</ol>
<p>(todo: expand and clarify these last steps. could leave them fairly brief here and go into detail in the supplement).</p>
<p>Due to the high memory requirements of the current implementation of vg call, toil-vg call splits the input graph into 2.5Mb overlapping chunks along the reference path. Each chunk is called independently in parallel and the results are concatenated into the output VCF.</p>
<h4 id="toil-vg-sveval">Toil-vg sveval</h4>
<p>The variants are first normalized with <code>bcftools norm</code> to ensure consistent representation between called variants and baseline variants. We then implemented an overlap-based strategy to compare SVs and compute evaluation metrics (sveval R package: <a href="https://github.com/jmonlong/sveval">https://github.com/jmonlong/sveval</a>).</p>
<p>For deletions and inversions, the affected regions in the reference genome are overlapped and matched between the two sets of SVs. First, we select pairs of SVs with at least 10% reciprocal overlap. Then for each variant we compute the proportion of its region that is covered by an overlapping variant in the other set. If this coverage proportion is higher than 50%, the variant is considered <em>covered</em>. True positives are <em>covered</em> variants from the call set or the truth set. False positives are variants from the call set that are not <em>covered</em>. False negative are variants from the truth set that are not <em>covered</em>.</p>
<p>For insertions, we select pairs of insertions that are located no farther than 20 bp from each other. We then align the inserted sequences using a Smith-Waterman alignment. For each insertion we compute the proportion of its inserted sequence that aligns a matched variant in the other set. As for deletions/inversions, this coverage proportion is used to annotate variants as true positives, false positives and false negatives.</p>
<p>sveval accepts VCF files with symbolic or explicit representation of the SVs. If the explicit representation is used, multi-allelic variants are split and their sequences right-trimmed. When inversions are considered, the reverse-complement of the ALT sequence of variants larger than 10 bp is aligned to the REF sequence and classified as an inversion if more than 80% of the sequence aligns.</p>
<p>We assess either the <em>calling</em> performance (absence/presence of a SV) or the <em>genotyping</em> performance. For the <em>calling</em> evaluation, both heterozygous and homozygous alternate SVs are compared jointly using the approach described above. To compute genotype-level metrics, the heterozygous and homozygous SVs are compared separately. Before splitting the variants by genotype, consecutive heterozygous variants are first stitched together if located at less that 20 bp from each other. Pairs of heterozygous variants with reciprocal overlap of at least 80% are also merged into homozygous variant before splitting variants by genotype.</p>
<h3 id="other-sv-genotypers">Other SV genotypers</h3>
<h4 id="bayestyper">BayesTyper</h4>
<p>If not specified otherwise BayesTyper was run as follows. Raw reads were mapped to the reference genome using <code>bwa mem</code>. GATK and Platypus were run on the mapped reads to call SNVs and short indels (&lt;50bp). The VCFs with these variants and the SVs to genotype were combined and used as input for BayesTyper. The BayesTyper genotyping pipeline started by counting the kmers in the raw reads using <code>kmc -k55 -ci1 -m8</code>. These kmers are then passed through a Bloom filter using <code>bayesTyperTools makeBloom</code>. Finally, variants are clustered and genotyped using <code>bayestyper cluster</code> and <code>bayestyper genotype --min-genotype-posterior 0</code></p>
<h4 id="delly">Delly</h4>
<p>The <code>delly call</code> command was run on the reads mapped by <code>bwa mem</code>, the reference genome fasta file and the VCF containing the SVs to genotype in their explicit representation.</p>
<h4 id="svtyper">SVTyper</h4>
<p>The VCF containing deletions was converted to symbolic representation and passed to <code>svtyper</code> with the reads mapped by <code>bwa mem</code>. The output VCF was converted back to explicit representation using <code>bayesTyperTools convertAllele</code> to facilitate variant normalization before evaluation.</p>
<h4 id="smrt-sv2">SMRT-SV2</h4>
<p>SMRT-SV2 was run on VCFs generated for SMRT-SV2 with the “30x-4” model and min-call-depth 8 cutoff. The output VCF was converted back to explicit representation, to facilitate variant normalization later.</p>
<p>(todo: double-check and maybe more details from Glenn)</p>
<h3 id="simulation-experiment">Simulation experiment</h3>
<p>We simulated a synthetic genome with 1000 insertions, deletions and inversions. Each variant was separated from the next by a buffer region of 500 bp. The size of deletions and insertions followed the distribution of real SVs from the HGSVC catalog. We used the same size distribution as deletions for inversions. A VCF file was produced for three simulated samples with random genotypes (homozygous reference, heterozygous, homozygous alternate).</p>
<p>We created another VCF file containing errors in the SV definition. One or both breakpoints of deletions and inversions were shifted between 1 and 10 bp. The location and sequence of insertions were also modified, either shifted or deleted at the flanks, again by up to 10 bp.</p>
<p>Paired-end reads were simulated using <code>vg sim</code> on the graph that contained the true SVs. Different read depths were tested: 1x, 3x, 7x, 10x, 13x, 20x. We used real Illumina reads from NA12878 provided by the Genome in a Bottle consortium to model base qualities and sequencing errors.</p>
<p>The different methods were tested using either the true VCF or the VCF that contained errors. For vg, a graph was constructed from the VCF file, indexed, then used to map simulated reads and call variants using toil-vg (see <a href="#toil-vg">Toil-vg</a>). A beta version 1.5 of BayesTyper was run directly on the simulated reads and using an input VCF with SV only. In order to run the other methods, reads were mapped to the linear reference sequence using <code>bwa mem</code> and sorted using <code>samtools</code>. For Delly, insertions and deletions were first genotyped together using these mapped reads and the <code>delly call</code> command. Inversions were genotyped separately using a VCF that was formatted according to Delly’s preference. SVTyper was run on the mapped reads and a VCF that was converted to symbolic variant representation. All commands used for this analysis are available at <a href="XXX">ANALYSIS_REPO</a>.</p>
<p>The genotypes called in each experiment (method/VCF with or without errors/sequencing depth) were compared to the true SV genotypes to compute the precision, recall and F1 score (see <a href="#toil-vg-sveval">Toil-vg sveval</a>).</p>
<h4 id="breakpoint-fine-tuning-using-graph-augmentation">Breakpoint fine-tuning using graph augmentation</h4>
<p>vg can call variants after augmenting the graph with the read alignments to discover new variants (see <a href="#toil-vg-call">Toil-vg call</a>). We tested if this approach could fine-tune the breakpoint location of SVs in the graph. We started with the graph that contained approximate SVs (1-10 bp errors in breakpoint location) and 20x simulated reads from the simulation experiment (see <a href="#simulation-experiment">Simulation experiment</a>). The variants called after graph augmentation were compared with the true SVs and considered fine-tuned if the breakpoints matched exactly.</p>
<h3 id="hgsvc-analysis">HGSVC Analysis</h3>
<p>Phased VCFs were obtained for the three HGSVC samples from Chaisson et al.<span class="citation" data-cites="vQTymKCj">[<a href="#ref-vQTymKCj">2</a>]</span> and combined with <code>bcftools merge</code>. A variation graph was created and indexed using the combined VCF and the HS38D1 reference with alt loci excluded. The phasing information was used to construct a GBWT index, from which the two haploid sequences from HG00514 were extracted. Illumina read pairs with 30x coverage were simulated from these sequences using vg, with an error model learned from real reads from the same sample. Still, these reads reflect the idealized situation where the breakpoints of the SVs being genotyped are exactly known a priori. The reads were mapped to the graph and the mappings used to genotype the SVs in the graph, which were finally compared back to the HG00514 genotypes from the HGSVC VCF. The process was repeated with the same reads on the linear reference, using bwa-mem for mapping and Delly, SVTyper and BayesTyper for SV genotyping.</p>
<p>Illumina HiSeq 2500 paired end reads were downloaded from the EBI’s ENA FTP site for the three samples, using Run Accessions ERR903030, ERR895347 and ERR894724 for HG00514, HG00733 and NA19240, respectively. The graph and linear mapping and genotyping pipelines were run exactly as for the simulation, and the comparison results were aggregated across the three samples.</p>
<h3 id="giab-analysis">GIAB Analysis</h3>
<p>Version 0.6 of the GIAB SV VCF for the Ashkenazim son (HG002) was obtained from the NCBI FTP site. Illumina reads downsampled to 50x coverage obtained as described in Garrison et al.<span class="citation" data-cites="10jxt15v0">[<a href="#ref-10jxt15v0">14</a>]</span>, were used to run the vg and linear SV genotyping pipelines described above though with GRCh37 instead of 38. Since this dataset contains only one sample, variants without a determined genotype (14649 out of 74012) were considered “false positives” as a proxy measure for precision.</p>
<h3 id="smrt-sv2-comparison-chmpd-and-svpop">SMRT-SV2 Comparison (CHMPD and SVPOP)</h3>
<p>The SMRT-SV2 genotyper can only be used to genotype VCFs that were created by SMRT-SV2, and therefore could not be run on our simulated, HGSVC or GIAB data. The authors shared their training and evaluation set, a pseudodiploid sample constructed from combining the haploid CHM1 and CHM13 samples, along with a negative control (NA19240). The high quality of the CHM assemblies makes this set an attractive alternative to using simulated reads. We used this two-sample pseudodiploid VCF along with the 30X read set to construct, map and genotype with vg, and also ran SMRT-SV2 genotyper with the “30x-4” model and min-call-depth 8 cutoff, and compared the two back to the original VCF.</p>
<p>In an effort to extend this comparison to a more realistic setting, we reran the three HGSVC samples against the SMRT-SV2 discovery VCF (which contains them in addition to 12 other samples) published by Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span> using vg and SMRT-SV2 Genotyper. The discovery VCF does not contain genotypes so we did not distinguish between heterozygous and homozygous genotypes, looking at only the presence or absence of an alt allele for each variant.</p>
<p>SMRT-SV2 produces some explicit <em>no-calls</em> predictions when the read coverage is too low to produce accurate genotypes. These no-calls are considered homozygous reference in the main evaluation. We also explored the performance of vg and SMRT-SV2 in different sets of regions:</p>
<ol type="1">
<li>Non-repeat regions, i.e. excluding segmental duplications and tandem repeats.</li>
<li>Repeat regions defined as segmental duplications and tandem repeats.</li>
<li>Regions where SMRT-SV2 could call variants.</li>
<li>Regions where SMRT-SV2 produced no-calls.</li>
</ol>
<h3 id="yeast-graph-analysis">Yeast graph analysis</h3>
<p>For the analysis of graphs from de novo assemblies, we utilized publicly available PacBio-derived assemblies and Illumina short read sequencing datasets for 12 yeast strains from two related clades (Table <a href="#tbl:strains">1</a>) <span class="citation" data-cites="7f5OKa5O">[<a href="#ref-7f5OKa5O">23</a>]</span>. Two different genome graphs were constructed from the assemblies of five selected strains (<em>S.c. S288C</em>, <em>S.c. SK1</em>, <em>S.c. YPS128</em>, <em>S.p. CBS432</em>, and <em>S.p. UFRJ50816</em>). In the following, we describe the steps for the construction of both graphs and the calling of variants. For more details and the precise commands used in our analyses, see the following GitHub repository: https://github.com/eldariont/yeast_sv.</p>
<p>(todo: merge all analysis code to one documented repo)</p>
<a name="tbl:strains"></a>
<table>
<caption><span>Table 1:</span> 12 yeast strains from two related clades were used in our analysis. Five strains were selected to be included in the graphs while the remaining seven were used for variant calling only. </caption>
<thead>
<tr class="header">
<th>Strain</th>
<th>Clade</th>
<th>Included in graph</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>S288C</td>
<td>S. cerevisiae</td>
<td>✓</td>
</tr>
<tr class="even">
<td>SK1</td>
<td>S. cerevisiae</td>
<td>✓</td>
</tr>
<tr class="odd">
<td>YPS128</td>
<td>S. cerevisiae</td>
<td>✓</td>
</tr>
<tr class="even">
<td>UWOPS034614</td>
<td>S. cerevisiae</td>
<td></td>
</tr>
<tr class="odd">
<td>Y12</td>
<td>S. cerevisiae</td>
<td></td>
</tr>
<tr class="even">
<td>DBVPG6765</td>
<td>S. cerevisiae</td>
<td></td>
</tr>
<tr class="odd">
<td>DBVPG6044</td>
<td>S. cerevisiae</td>
<td></td>
</tr>
<tr class="even">
<td>CBS432</td>
<td>S. paradoxus</td>
<td>✓</td>
</tr>
<tr class="odd">
<td>UFRJ50816</td>
<td>S. paradoxus</td>
<td>✓</td>
</tr>
<tr class="even">
<td>N44</td>
<td>S. paradoxus</td>
<td></td>
</tr>
<tr class="odd">
<td>UWOPS919171</td>
<td>S. paradoxus</td>
<td></td>
</tr>
<tr class="even">
<td>YPS138</td>
<td>S. paradoxus</td>
<td></td>
</tr>
</tbody>
</table>
<h4 id="construction-of-the-construct-graph">Construction of the <em>construct graph</em></h4>
<p>For the first graph (called <em>construct graph</em> throughout the paper), the most common graph construction method was applied. It requires a linear reference genome and a VCF file of variants on that reference to build the graph. As reference genome, the PacBio assembly of the S.c. S288C strain was chosen because this strain was used for the S. cerevisiae genome reference assembly. To obtain variants three methods for SV detection from genome assemblies were combined: Assemblytics <span class="citation" data-cites="krO7WgVi">[<a href="#ref-krO7WgVi">24</a>]</span> (commit df5361f), AsmVar (commit 5abd91a) <span class="citation" data-cites="oVaXIwl5">[<a href="#ref-oVaXIwl5">25</a>]</span> and paftools (version 2.14-r883) <span class="citation" data-cites="172cJaw4Q">[<a href="#ref-172cJaw4Q">26</a>]</span>. All three methods were run to detect SVs between the PacBio assembly of reference strain S.c. S288C and the PacBio assemblies of each of the four other selected yeast strains. The union of variants detected by the three methods was produced (using bedtools [̍@1HWiAHnIw]) and variants with a reciprocal overlap of at least 50% were combined to avoid duplication in the union set. These union sets of variants for each of the four selected (and non-reference) strains were merged and another deduplication step was applied to combine variants with a reciprocal overlap of at least 90%. The resulting total set of variants in VCF format and the linear reference genome were used to build the <em>construct graph</em> with <code>vg construct</code>.</p>
<h4 id="construction-of-the-cactus-graph">Construction of the <em>cactus graph</em></h4>
<p>For the second graph (called <em>cactus graph</em> throughout the paper), an alternative graph construction methods directly from de novo genome assemblies was applied. First, the repeat-masked PacBio-assemblies of the five selected strains were aligned with our Cactus tool <span class="citation" data-cites="1FgS53pXi">[<a href="#ref-1FgS53pXi">27</a>]</span>. Cactus requires a phylogenetic tree of the strains which was estimated using Mash <span class="citation" data-cites="mH9pzoIn">[<a href="#ref-mH9pzoIn">30</a>]</span> and PHYLIP. Subsequently, the output file in HAL format was converted to a variant graph with hal2vg (<a href="https://github.com/ComparativeGenomicsToolkit/hal2vg">https://github.com/ComparativeGenomicsToolkit/hal2vg</a>).</p>
<h4 id="calling-and-genotyping-of-svs">Calling and genotyping of SVs</h4>
<p>Prior to variant calling, the Illumina short reads of all 12 yeast strains were mapped to both graphs using <code>vg map</code>. The fractions of reads mapped with specific properties were measured using <code>vg view</code> and the JSON processor <code>jq</code>. Then, <code>toil-vg call</code> (commit be8b6da) was used to analyze the mapped reads of each of the 11 non-reference strains and to call variants. Thus, a separate variant callset was obtained for each of the strains and both graphs. To evaluate the callsets, a sample graph (i.e. a graph representation of the callset) was generated for each callset using <code>vg construct</code> and <code>vg mod</code> on the reference assembly <em>S.c. S288C</em> and the callset. Subsequently, short reads from the respective strains were mapped to each sample graph using <code>vg map</code>. The resulting alignments were analyzed with <code>vg view</code> and <code>jq</code>.</p>
<h2 id="discussion">Discussion</h2>
<p><em>Potential topics for the discussion.</em></p>
<h4 id="performance-across-datasets">Performance across datasets</h4>
<p>vg was overall the best genotyper in our benchmarks. The other methods were superior in few datasets and situations. This might be partly explained by the quality of the input SV catalog. The GiaB catalog is more curated and, specifically for deletions in non-repeat regions, Delly and BayesTyper were better at predicting genotypes compared to vg. This might be because the breakpoint resolution for this type of SV in these regions is better in this dataset compared to the HGSVC dataset which was derived mostly from long-read sequencing. Similarly, SMRT-SV2 performs better for deletions in the pseudo-diploid genome constructed from two high quality genome assemblies of CHM cell lines.</p>
<h4 id="providing-a-resource-to-be-used-by-large-scale-sequencing-project">Providing a resource to be used by large-scale sequencing project</h4>
<p>As a result of this study we provide a variation graph containing XX millions of SNVs and indels from the 1000 Genomes Project as well as XX thousands of SVs derived from long-read sequencing. This variation graph could serve as a richer reference for large scale projects that use short-read sequencing. For instance, more and more large-scale projects are sequencing the genomes of thousands or hundreds of thousands of individuals, e.g. the Pancancer Analysis of Whole Genomes<span class="citation" data-cites="10Jid8Wql">[<a href="#ref-10Jid8Wql">31</a>]</span>, the Genomics England initiative<span class="citation" data-cites="mWj2p7Xp">[<a href="#ref-mWj2p7Xp">32</a>]</span>, and the TOPMed consortium<span class="citation" data-cites="ir1O1h8n">[<a href="#ref-ir1O1h8n">33</a>]</span>. These large WGS studies will provide a deeper look into the mechanism of common diseases and, in some cases, will be used directly in a clinical setting. Clinicians and researchers are eager to use these growing WGS resources to interrogate the importance of SVs in disease at a scale never achieved before, either to get a more complete picture of the genetic factors of a disease or to produce a more comprehensive clinical report. As sequencing reaches the clinic, whole-genome sequencing will become routine for many patients. Clinicians will rely on variant calling and interpretation for diagnosis and treatment. For variant interpretation in particular, a comprehensive and unified characterization of the genomic variation will be extremely valuable.</p>
<h4 id="simplified-usage">Simplified usage</h4>
<p>Some methods require additional information or special VCF formatting <span class="citation" data-cites="16glGyppX">[<a href="#ref-16glGyppX">34</a>]</span>. SVTyper was designed to use VCFs created by Lumpy. The genotyping module from Delly was implemented for variants found by its discovery module. SMRT-SV2 required the VCF created specifically for it by Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span>. Nebula, a new k-mer-based genotyper, requires reads from a individual with the SV during k-mer selection<span class="citation" data-cites="zRHJDqQ9">[<a href="#ref-zRHJDqQ9">35</a>]</span>. In contrast, vg can take as input either explicit or symbolic VCFs, as well as assembly alignment.</p>
<h4 id="benefits-of-de-novo-assemblies">Benefits of de novo assemblies</h4>
<p>Our results suggest that constructing a graph from de novo assembly alignment is more representative of the sequencing reads and leads to better SV genotyping. De novo assemblies for human are becoming more and more common, for example from optimized mate-pair libraries<span class="citation" data-cites="pJAv1D8R">[<a href="#ref-pJAv1D8R">3</a>]</span> or long-read sequencing<span class="citation" data-cites="pJAv1D8R">[<a href="#ref-pJAv1D8R">3</a>]</span>. For an optimal representation of the genomic variation, we expect the future graphs to include information from the alignment of numerous de novo assemblies. Aligning assembled contigs to existing variation graphs, like to ones created from SVs catalogs, is still experimental but could generate a genome graph augmented with both existing variant databases and new high-quality assemblies.</p>
<h4 id="future-improvements-in-vg">Future improvements in vg</h4>
<p>The vg toolkit is in active development. Read mapping is an area of constant improvement, both in terms of computational efficiency and accuracy. For example, haplotype information can be modeled in variation graphs and, in the future, assist read mapping and variant calling. These upcoming developments will directly benefit SV genotyping with vg.</p>
<h4 id="limitations">Limitations</h4>
<p>Copy number variants (CNVs) are currently represented as deletions or insertions. For this reason duplications are represented as additional sequence rather than encoded as a loop in the graph. While this is sufficient to represent single copy changes, such as deletions or single tandem duplications, CNVs with multi-copies states are not addressed by the current implementation. The genotyping algorithm would need to be extended to model copy number in order to assess these variants.</p>
<p>Near-breakpoint resolution is necessary. Simulations have shown that SV genotyping with vg is robust to errors up to 10 bp in breakpoint location. Variants with higher uncertainty in the breakpoint location, for example discovered through read coverage analysis, cannot be added to the graph.</p>
<p>The genotyping evaluation of inversions is limited by the lack of existing gold-standards. We showed that vg is capable of genotyping simple inversions from simulation or the few discovered in the SV catalog from Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span>. However, most inversions are complex and involve small insertions/deletions around their breakpoints<span class="citation" data-cites="1BciDpJfo">[<a href="#ref-1BciDpJfo">36</a>]</span>. While these complex variants are difficult to represent in the VCF format, we expect that de novo assembly alignment will help integrating these variants into the graph.</p>
<h2 id="supplementary-material">Supplementary Material</h2>
<a name="tbl:hgsvc"></a>
<table>
<caption><span>Table S1:</span> Genotyping evaluation on the HGSVC dataset. Precision, recall and F1 score for the call set with the best F1 score. The numbers in parenthesis corresponds to the results in non-repeat regions. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Experiment</th>
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Type</th>
<th style="text-align: left;">Precision</th>
<th style="text-align: left;">Recall</th>
<th style="text-align: left;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Simulated reads</td>
<td style="text-align: left;">vg</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.795 (0.885)</td>
<td style="text-align: left;">0.796 (0.883)</td>
<td style="text-align: left;">0.795 (0.884)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.869 (0.971)</td>
<td style="text-align: left;">0.771 (0.92)</td>
<td style="text-align: left;">0.817 (0.945)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">BayesTyper</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.91 (0.935)</td>
<td style="text-align: left;">0.835 (0.9)</td>
<td style="text-align: left;">0.871 (0.917)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.898 (0.981)</td>
<td style="text-align: left;">0.806 (0.929)</td>
<td style="text-align: left;">0.849 (0.954)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">svtyper</td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.809 (0.876)</td>
<td style="text-align: left;">0.328 (0.754)</td>
<td style="text-align: left;">0.467 (0.81)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">Delly</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.767 (0.866)</td>
<td style="text-align: left;">0.093 (0.225)</td>
<td style="text-align: left;">0.166 (0.358)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.696 (0.903)</td>
<td style="text-align: left;">0.707 (0.846)</td>
<td style="text-align: left;">0.701 (0.874)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Real reads</td>
<td style="text-align: left;">vg</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.431 (0.683)</td>
<td style="text-align: left;">0.541 (0.726)</td>
<td style="text-align: left;">0.48 (0.704)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.65 (0.886)</td>
<td style="text-align: left;">0.519 (0.708)</td>
<td style="text-align: left;">0.577 (0.787)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">BayesTyper</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.601 (0.747)</td>
<td style="text-align: left;">0.254 (0.433)</td>
<td style="text-align: left;">0.357 (0.549)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.627 (0.91)</td>
<td style="text-align: left;">0.325 (0.381)</td>
<td style="text-align: left;">0.428 (0.537)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">svtyper</td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.661 (0.733)</td>
<td style="text-align: left;">0.236 (0.551)</td>
<td style="text-align: left;">0.348 (0.629)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">Delly</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.516 (0.621)</td>
<td style="text-align: left;">0.068 (0.176)</td>
<td style="text-align: left;">0.12 (0.275)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.55 (0.838)</td>
<td style="text-align: left;">0.445 (0.547)</td>
<td style="text-align: left;">0.492 (0.662)</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:giab"></a>
<table>
<caption><span>Table S2:</span> Genotyping evaluation on the Genome in a Bottle dataset. Precision, recall and F1 score for the call set with the best F1 score. The numbers in parenthesis corresponds to the results in non-repeat regions. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Type</th>
<th style="text-align: left;">Precision</th>
<th style="text-align: left;">Recall</th>
<th style="text-align: left;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">vg</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.658 (0.774)</td>
<td style="text-align: left;">0.646 (0.735)</td>
<td style="text-align: left;">0.652 (0.754)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.68 (0.768)</td>
<td style="text-align: left;">0.643 (0.735)</td>
<td style="text-align: left;">0.661 (0.751)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BayesTyper</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.776 (0.879)</td>
<td style="text-align: left;">0.286 (0.379)</td>
<td style="text-align: left;">0.418 (0.53)</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.808 (0.886)</td>
<td style="text-align: left;">0.512 (0.696)</td>
<td style="text-align: left;">0.627 (0.779)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">svtyper</td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.742 (0.818)</td>
<td style="text-align: left;">0.342 (0.496)</td>
<td style="text-align: left;">0.468 (0.618)</td>
</tr>
<tr class="even">
<td style="text-align: left;">Delly</td>
<td style="text-align: left;">INS</td>
<td style="text-align: left;">0.822 (0.894)</td>
<td style="text-align: left;">0.177 (0.268)</td>
<td style="text-align: left;">0.291 (0.412)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">0.722 (0.822)</td>
<td style="text-align: left;">0.645 (0.768)</td>
<td style="text-align: left;">0.681 (0.794)</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:chmpd"></a>
<table>
<caption><span>Table S3:</span> Genotyping evaluation on the pseudo-diploid genome built from CHM cell lines in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span>. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Region</th>
<th style="text-align: left;">Type</th>
<th style="text-align: right;">Precision</th>
<th style="text-align: right;">Recall</th>
<th style="text-align: right;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">vg</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">0.665</td>
<td style="text-align: right;">0.661</td>
<td style="text-align: right;">0.663</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">0.688</td>
<td style="text-align: right;">0.500</td>
<td style="text-align: right;">0.579</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">0.806</td>
<td style="text-align: right;">0.784</td>
<td style="text-align: right;">0.795</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">0.869</td>
<td style="text-align: right;">0.762</td>
<td style="text-align: right;">0.812</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SMRT-SV2</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">0.757</td>
<td style="text-align: right;">0.536</td>
<td style="text-align: right;">0.628</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">0.848</td>
<td style="text-align: right;">0.630</td>
<td style="text-align: right;">0.723</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">0.880</td>
<td style="text-align: right;">0.680</td>
<td style="text-align: right;">0.767</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">0.971</td>
<td style="text-align: right;">0.824</td>
<td style="text-align: right;">0.891</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:svpop"></a>
<table>
<caption><span>Table S4:</span> Calling evaluation on the SVPOP dataset. Combined results for the HG00514, HG00733 and NA19240 individuals, 3 of the 15 individuals used to generate the high-quality SV catalog in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span>. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Region</th>
<th style="text-align: left;">Type</th>
<th style="text-align: right;">TP</th>
<th style="text-align: right;">FP</th>
<th style="text-align: right;">FN</th>
<th style="text-align: right;">Precision</th>
<th style="text-align: right;">Recall</th>
<th style="text-align: right;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">vg</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">25838</td>
<td style="text-align: right;">22042</td>
<td style="text-align: right;">15772</td>
<td style="text-align: right;">0.540</td>
<td style="text-align: right;">0.621</td>
<td style="text-align: right;">0.577</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">14545</td>
<td style="text-align: right;">6824</td>
<td style="text-align: right;">15425</td>
<td style="text-align: right;">0.681</td>
<td style="text-align: right;">0.485</td>
<td style="text-align: right;">0.567</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">173</td>
<td style="text-align: right;">0.509</td>
<td style="text-align: right;">0.135</td>
<td style="text-align: right;">0.213</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">8051</td>
<td style="text-align: right;">3258</td>
<td style="text-align: right;">1817</td>
<td style="text-align: right;">0.712</td>
<td style="text-align: right;">0.816</td>
<td style="text-align: right;">0.760</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">3769</td>
<td style="text-align: right;">623</td>
<td style="text-align: right;">818</td>
<td style="text-align: right;">0.858</td>
<td style="text-align: right;">0.822</td>
<td style="text-align: right;">0.840</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">75</td>
<td style="text-align: right;">0.613</td>
<td style="text-align: right;">0.202</td>
<td style="text-align: right;">0.304</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SMRT-SV2</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">16270</td>
<td style="text-align: right;">26031</td>
<td style="text-align: right;">25340</td>
<td style="text-align: right;">0.385</td>
<td style="text-align: right;">0.391</td>
<td style="text-align: right;">0.388</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">11793</td>
<td style="text-align: right;">10106</td>
<td style="text-align: right;">18177</td>
<td style="text-align: right;">0.539</td>
<td style="text-align: right;">0.393</td>
<td style="text-align: right;">0.455</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">4483</td>
<td style="text-align: right;">4659</td>
<td style="text-align: right;">5385</td>
<td style="text-align: right;">0.490</td>
<td style="text-align: right;">0.454</td>
<td style="text-align: right;">0.472</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">2928</td>
<td style="text-align: right;">930</td>
<td style="text-align: right;">1659</td>
<td style="text-align: right;">0.759</td>
<td style="text-align: right;">0.638</td>
<td style="text-align: right;">0.693</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:svpop-regions"></a>
<table>
<caption><span>Table S5:</span> Calling evaluation on the SVPOP dataset in different sets of regions for the HG5014 individual. </caption>
<thead>
<tr class="header">
<th style="text-align: left;">Method</th>
<th style="text-align: left;">Region</th>
<th style="text-align: left;">Type</th>
<th style="text-align: right;">TP</th>
<th style="text-align: right;">FP</th>
<th style="text-align: right;">FN</th>
<th style="text-align: right;">Precision</th>
<th style="text-align: right;">Recall</th>
<th style="text-align: right;">F1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">vg</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">8618</td>
<td style="text-align: right;">7237</td>
<td style="text-align: right;">5416</td>
<td style="text-align: right;">0.546</td>
<td style="text-align: right;">0.614</td>
<td style="text-align: right;">0.578</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">4762</td>
<td style="text-align: right;">2048</td>
<td style="text-align: right;">5145</td>
<td style="text-align: right;">0.696</td>
<td style="text-align: right;">0.481</td>
<td style="text-align: right;">0.569</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">54</td>
<td style="text-align: right;">0.579</td>
<td style="text-align: right;">0.169</td>
<td style="text-align: right;">0.262</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">6176</td>
<td style="text-align: right;">6923</td>
<td style="text-align: right;">4678</td>
<td style="text-align: right;">0.475</td>
<td style="text-align: right;">0.569</td>
<td style="text-align: right;">0.518</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">2428</td>
<td style="text-align: right;">1701</td>
<td style="text-align: right;">4542</td>
<td style="text-align: right;">0.584</td>
<td style="text-align: right;">0.348</td>
<td style="text-align: right;">0.436</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0.500</td>
<td style="text-align: right;">0.143</td>
<td style="text-align: right;">0.222</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">2677</td>
<td style="text-align: right;">987</td>
<td style="text-align: right;">514</td>
<td style="text-align: right;">0.731</td>
<td style="text-align: right;">0.839</td>
<td style="text-align: right;">0.781</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">1180</td>
<td style="text-align: right;">176</td>
<td style="text-align: right;">321</td>
<td style="text-align: right;">0.869</td>
<td style="text-align: right;">0.786</td>
<td style="text-align: right;">0.825</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">0.636</td>
<td style="text-align: right;">0.259</td>
<td style="text-align: right;">0.368</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;">called in SMRT-SV</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">3410</td>
<td style="text-align: right;">3789</td>
<td style="text-align: right;">2108</td>
<td style="text-align: right;">0.478</td>
<td style="text-align: right;">0.618</td>
<td style="text-align: right;">0.539</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">2544</td>
<td style="text-align: right;">1092</td>
<td style="text-align: right;">1518</td>
<td style="text-align: right;">0.699</td>
<td style="text-align: right;">0.626</td>
<td style="text-align: right;">0.661</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">INV</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">52</td>
<td style="text-align: right;">0.500</td>
<td style="text-align: right;">0.133</td>
<td style="text-align: right;">0.210</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">not called in SMRT-SV</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">4838</td>
<td style="text-align: right;">542</td>
<td style="text-align: right;">3678</td>
<td style="text-align: right;">0.899</td>
<td style="text-align: right;">0.568</td>
<td style="text-align: right;">0.696</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">2034</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">3723</td>
<td style="text-align: right;">0.987</td>
<td style="text-align: right;">0.353</td>
<td style="text-align: right;">0.520</td>
</tr>
<tr class="odd">
<td style="text-align: left;">SMRT-SV2</td>
<td style="text-align: left;">all</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">5245</td>
<td style="text-align: right;">8563</td>
<td style="text-align: right;">8789</td>
<td style="text-align: right;">0.394</td>
<td style="text-align: right;">0.374</td>
<td style="text-align: right;">0.384</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">3741</td>
<td style="text-align: right;">3382</td>
<td style="text-align: right;">6166</td>
<td style="text-align: right;">0.533</td>
<td style="text-align: right;">0.378</td>
<td style="text-align: right;">0.442</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">3848</td>
<td style="text-align: right;">7125</td>
<td style="text-align: right;">7006</td>
<td style="text-align: right;">0.368</td>
<td style="text-align: right;">0.354</td>
<td style="text-align: right;">0.361</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">1990</td>
<td style="text-align: right;">2832</td>
<td style="text-align: right;">4980</td>
<td style="text-align: right;">0.426</td>
<td style="text-align: right;">0.286</td>
<td style="text-align: right;">0.342</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">non-repeat</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">1396</td>
<td style="text-align: right;">1468</td>
<td style="text-align: right;">1795</td>
<td style="text-align: right;">0.493</td>
<td style="text-align: right;">0.438</td>
<td style="text-align: right;">0.464</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">901</td>
<td style="text-align: right;">308</td>
<td style="text-align: right;">600</td>
<td style="text-align: right;">0.745</td>
<td style="text-align: right;">0.600</td>
<td style="text-align: right;">0.665</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">called in SMRT-SV</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">4343</td>
<td style="text-align: right;">5595</td>
<td style="text-align: right;">1175</td>
<td style="text-align: right;">0.445</td>
<td style="text-align: right;">0.787</td>
<td style="text-align: right;">0.569</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">3227</td>
<td style="text-align: right;">2451</td>
<td style="text-align: right;">835</td>
<td style="text-align: right;">0.573</td>
<td style="text-align: right;">0.794</td>
<td style="text-align: right;">0.666</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">not called in SMRT-SV</td>
<td style="text-align: left;">INS</td>
<td style="text-align: right;">116</td>
<td style="text-align: right;">109</td>
<td style="text-align: right;">8400</td>
<td style="text-align: right;">0.551</td>
<td style="text-align: right;">0.014</td>
<td style="text-align: right;">0.026</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">DEL</td>
<td style="text-align: right;">206</td>
<td style="text-align: right;">16</td>
<td style="text-align: right;">5551</td>
<td style="text-align: right;">0.911</td>
<td style="text-align: right;">0.036</td>
<td style="text-align: right;">0.069</td>
</tr>
</tbody>
</table>
<hr />
<a name="tbl:simerror-bkpt"></a>
<table>
<caption><span>Table S6:</span> Breakpoint fine-tuning using graph augmentation from the read alignment. For deletions and inversions, either one or both breakpoints were shifted to introduce errors in the input VCF. For insertions, the insertion location and sequence contained errors. In all cases, the errors affected 1-10 bp. </caption>
<colgroup>
<col style="width: 10%" />
<col style="width: 15%" />
<col style="width: 13%" />
<col style="width: 10%" />
<col style="width: 13%" />
<col style="width: 17%" />
<col style="width: 19%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">SV type</th>
<th style="text-align: left;">Error type</th>
<th style="text-align: left;">Breakpoint</th>
<th style="text-align: right;">Variant</th>
<th style="text-align: right;">Proportion</th>
<th style="text-align: right;">Mean size (bp)</th>
<th style="text-align: right;">Mean error (bp)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">DEL</td>
<td style="text-align: left;">one end</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">220</td>
<td style="text-align: right;">0.219</td>
<td style="text-align: right;">422.655</td>
<td style="text-align: right;">6.095</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">784</td>
<td style="text-align: right;">0.781</td>
<td style="text-align: right;">670.518</td>
<td style="text-align: right;">5.430</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">both ends</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">811</td>
<td style="text-align: right;">0.814</td>
<td style="text-align: right;">826.070</td>
<td style="text-align: right;">6.275</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">185</td>
<td style="text-align: right;">0.186</td>
<td style="text-align: right;">586.676</td>
<td style="text-align: right;">2.232</td>
</tr>
<tr class="odd">
<td style="text-align: left;">INS</td>
<td style="text-align: left;">location/seq</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">123</td>
<td style="text-align: right;">0.062</td>
<td style="text-align: right;">428.724</td>
<td style="text-align: right;">6.667</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">1877</td>
<td style="text-align: right;">0.938</td>
<td style="text-align: right;">440.043</td>
<td style="text-align: right;">6.439</td>
</tr>
<tr class="odd">
<td style="text-align: left;">INV</td>
<td style="text-align: left;">one end</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">868</td>
<td style="text-align: right;">0.835</td>
<td style="text-align: right;">762.673</td>
<td style="text-align: right;">5.161</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">172</td>
<td style="text-align: right;">0.165</td>
<td style="text-align: right;">130.244</td>
<td style="text-align: right;">5.884</td>
</tr>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">both ends</td>
<td style="text-align: left;">incorrect</td>
<td style="text-align: right;">950</td>
<td style="text-align: right;">0.992</td>
<td style="text-align: right;">556.274</td>
<td style="text-align: right;">5.624</td>
</tr>
<tr class="even">
<td style="text-align: left;"></td>
<td style="text-align: left;"></td>
<td style="text-align: left;">fine-tuned</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">0.008</td>
<td style="text-align: right;">200.000</td>
<td style="text-align: right;">1.375</td>
</tr>
</tbody>
</table>
<hr />
<figure>
<img src="images/hgsvc-sim-geno.png" alt="Figure S1: Genotyping evaluation on the HGSVC dataset using simulated reads." id="fig:hgsvc-sim-geno" data-tag="S1" /><figcaption><span>Figure S1:</span> <strong>Genotyping evaluation on the HGSVC dataset using simulated reads.</strong></figcaption>
</figure>
<figure>
<img src="images/hgsvc-sim.png" alt="Figure S2: Calling evaluation on the HGSVC dataset using simulated reads." id="fig:hgsvc-sim" data-tag="S2" /><figcaption><span>Figure S2:</span> <strong>Calling evaluation on the HGSVC dataset using simulated reads.</strong></figcaption>
</figure>
<figure>
<img src="images/hgsvc-real-geno.png" alt="Figure S3: Genotyping evaluation on the HGSVC dataset using real reads. Combined results across the HG00514, HG00733 and NA19240." id="fig:hgsvc-real-geno" data-tag="S3" /><figcaption><span>Figure S3:</span> <strong>Genotyping evaluation on the HGSVC dataset using real reads.</strong> Combined results across the HG00514, HG00733 and NA19240.</figcaption>
</figure>
<figure>
<img src="images/hgsvc-real.png" alt="Figure S4: Calling evaluation on the HGSVC dataset using real reads. Combined results across the HG00514, HG00733 and NA19240." id="fig:hgsvc-real" data-tag="S4" /><figcaption><span>Figure S4:</span> <strong>Calling evaluation on the HGSVC dataset using real reads.</strong> Combined results across the HG00514, HG00733 and NA19240.</figcaption>
</figure>
<figure>
<img src="images/giab5-geno.png" alt="Figure S5: Genotyping evaluation on the Genome in a Bottle dataset. Predicted genotypes on HG002 were compared to the high-quality SVs from this same individual." id="fig:giab-geno" data-tag="S5" /><figcaption><span>Figure S5:</span> <strong>Genotyping evaluation on the Genome in a Bottle dataset.</strong> Predicted genotypes on HG002 were compared to the high-quality SVs from this same individual.</figcaption>
</figure>
<figure>
<img src="images/giab5.png" alt="Figure S6: Calling evaluation on the Genome in a Bottle dataset. Calls on HG002 were compared to the high-quality SVs from this same individual." id="fig:giab" data-tag="S6" /><figcaption><span>Figure S6:</span> <strong>Calling evaluation on the Genome in a Bottle dataset.</strong> Calls on HG002 were compared to the high-quality SVs from this same individual.</figcaption>
</figure>
<figure>
<img src="images/chmpd-geno.png" alt="Figure S7: Genotyping evaluation on the CHM pseudo-diploid dataset. The pseudo-diploid genome was built from CHM cell lines and used to train SMRT-SV2 in Audano et al.[10]" id="fig:chmpd-geno" data-tag="S7" /><figcaption><span>Figure S7:</span> <strong>Genotyping evaluation on the CHM pseudo-diploid dataset.</strong> The pseudo-diploid genome was built from CHM cell lines and used to train SMRT-SV2 in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span></figcaption>
</figure>
<figure>
<img src="images/chmpd.png" alt="Figure S8: Calling evaluation on the CHM pseudo-diploid dataset. The pseudo-diploid genome was built from CHM cell lines and used to train SMRT-SV2 in Audano et al.[10]" id="fig:chmpd" data-tag="S8" /><figcaption><span>Figure S8:</span> <strong>Calling evaluation on the CHM pseudo-diploid dataset.</strong> The pseudo-diploid genome was built from CHM cell lines and used to train SMRT-SV2 in Audano et al.<span class="citation" data-cites="3NNFS6U2">[<a href="#ref-3NNFS6U2">10</a>]</span></figcaption>
</figure>
<figure>
<img src="images/svpop.png" alt="Figure S9: Calling evaluation on the SVPOP dataset. Combined results across the HG00514, HG00733 and NA19240." id="fig:svpop" data-tag="S9" /><figcaption><span>Figure S9:</span> <strong>Calling evaluation on the SVPOP dataset.</strong> Combined results across the HG00514, HG00733 and NA19240.</figcaption>
</figure>
<figure>
<img src="images/svpop-regions.png" alt="Figure S10: Evaluation across different sets of regions in HG00514 (SVPOP dataset). Calling evaluation." id="fig:svpop-regions" data-tag="S10" /><figcaption><span>Figure S10:</span> <strong>Evaluation across different sets of regions in HG00514 (SVPOP dataset)</strong>. Calling evaluation.</figcaption>
</figure>
<figure>
<img src="images/simerror-bkpt-finetuning-vgcall.png" alt="Figure S11: Breakpoint fine-tuning using augmentation through “vg call”.. For deletions and inversions, either one or both breakpoints were shifted to introduce errors in the input VCF. For insertions, the insertion location and sequence contained errors. a) Proportion of variant for which breakpoints could be fine-tuned. b) Distribution of the amount of errors that could be corrected or not. c) Distribution of the size of the variants whose breakpoints could be fine-tuned or not." id="fig:simerror-bkpt" data-tag="S11" /><figcaption><span>Figure S11:</span> <strong>Breakpoint fine-tuning using augmentation through “vg call”.</strong>. For deletions and inversions, either one or both breakpoints were shifted to introduce errors in the input VCF. For insertions, the insertion location and sequence contained errors. a) Proportion of variant for which breakpoints could be fine-tuned. b) Distribution of the amount of errors that could be corrected or not. c) Distribution of the size of the variants whose breakpoints could be fine-tuned or not.</figcaption>
</figure>
<figure>
<img src="images/panel5.png" alt="Figure S12: Mapping comparison on graphs of all 12 strains. Short reads from all 12 yeast strains were aligned to both graphs. The fraction of reads mapped to the cactus graph (y-axis) and the construct graph (x-axis) are compared. a) Stratified by mapping quality threshold. b) Stratified by percent identity threshold. Colors and shapes represent the 12 strains and two clades, respectively." id="fig:panel5" data-tag="S12" /><figcaption><span>Figure S12:</span> <strong>Mapping comparison on graphs of all 12 strains.</strong> Short reads from all 12 yeast strains were aligned to both graphs. The fraction of reads mapped to the cactus graph (y-axis) and the construct graph (x-axis) are compared. a) Stratified by mapping quality threshold. b) Stratified by percent identity threshold. Colors and shapes represent the 12 strains and two clades, respectively.</figcaption>
</figure>
<figure>
<img src="images/panel6.png" alt="Figure S13: SV genotyping comparison on graphs of all 12 strains. Short reads from all 11 non-reference yeast strains were used to genotype SVs contained in both graphs. Subsequently, sample graphs were generated from the resulting SV callsets. The short reads were again aligned to the sample graphs and the quality of the alignments was used to ascertain genotyping performance. a) Average mapping quality of short reads aligned to the sample graphs derived from cactus graph (y-axis) and construct graph (x-axis). b) Average mapping identity of short reads aligned to the sample graphs derived from cactus graph (y-axis) and construct graph (x-axis). Colors and shapes represent the 11 non-reference strains and two clades, respectively." id="fig:panel6" data-tag="S13" /><figcaption><span>Figure S13:</span> <strong>SV genotyping comparison on graphs of all 12 strains.</strong> Short reads from all 11 non-reference yeast strains were used to genotype SVs contained in both graphs. Subsequently, sample graphs were generated from the resulting SV callsets. The short reads were again aligned to the sample graphs and the quality of the alignments was used to ascertain genotyping performance. a) Average mapping quality of short reads aligned to the sample graphs derived from <em>cactus graph</em> (y-axis) and <em>construct graph</em> (x-axis). b) Average mapping identity of short reads aligned to the sample graphs derived from <em>cactus graph</em> (y-axis) and <em>construct graph</em> (x-axis). Colors and shapes represent the 11 non-reference strains and two clades, respectively.</figcaption>
</figure>
<h2 id="references" class="page_break_before">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs">
<div id="ref-ebc66eBr">
<p>1. <strong>Phenotypic impact of genomic structural variation: insights from and for human disease</strong><br />
Joachim Weischenfeldt, Orsolya Symmons, François Spitz, Jan O. Korbel<br />
<em>Nature Reviews Genetics</em> (2013-02) <a href="https://doi.org/f4nhxh">https://doi.org/f4nhxh</a><br />
DOI: <a href="https://doi.org/10.1038/nrg3373">10.1038/nrg3373</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23329113">23329113</a></p>
</div>
<div id="ref-vQTymKCj">
<p>2. <strong>Multi-platform discovery of haplotype-resolved structural variation in human genomes</strong><br />
Mark J.P. Chaisson, Ashley D. Sanders, Xuefang Zhao, Ankit Malhotra, David Porubsky, Tobias Rausch, Eugene J. Gardner, Oscar Rodriguez, Li Guo, Ryan L. Collins, … Charles Lee<br />
<em>Cold Spring Harbor Laboratory</em> (2017-09-23) <a href="https://doi.org/gftxhc">https://doi.org/gftxhc</a><br />
DOI: <a href="https://doi.org/10.1101/193144">10.1101/193144</a></p>
</div>
<div id="ref-pJAv1D8R">
<p>3. <strong>Sequencing and de novo assembly of 150 genomes from Denmark as a population reference</strong><br />
Lasse Maretty, Jacob Malte Jensen, Bent Petersen, Jonas Andreas Sibbesen, Siyang Liu, Palle Villesen, Laurits Skov, Kirstine Belling, Christian Theil Have, Jose M. G. Izarzugaza, … Mikkel Heide Schierup<br />
<em>Nature</em> (2017-07-26) <a href="https://doi.org/gbpnnx">https://doi.org/gbpnnx</a><br />
DOI: <a href="https://doi.org/10.1038/nature23264">10.1038/nature23264</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28746312">28746312</a></p>
</div>
<div id="ref-qA6dWFP">
<p>4. <strong>An integrated map of structural variation in 2,504 human genomes</strong><br />
Peter H. SudmantTobias Rausch, Eugene J. Gardner, Robert E. Handsaker, Alexej Abyzov, John Huddleston, Yan Zhang, Kai Ye, Goo Jun, … Jan O. Korbel<br />
<em>Nature</em> (2015-10) <a href="https://doi.org/73c">https://doi.org/73c</a><br />
DOI: <a href="https://doi.org/10.1038/nature15394">10.1038/nature15394</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26432246">26432246</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4617611">PMC4617611</a></p>
</div>
<div id="ref-py6BC5kj">
<p>5. <strong>Whole-genome sequence variation, population structure and demographic history of the Dutch population</strong><br />
Laurent C FrancioliAndroniki Menelaou, Sara L Pulit, Freerk van Dijk, Pier Francesco Palamara, Clara C Elbers, Pieter BT Neerincx, Kai Ye, Victor Guryev, … Cisca Wijmenga<br />
<em>Nature Genetics</em> (2014-06-29) <a href="https://doi.org/f6bxm8">https://doi.org/f6bxm8</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3021">10.1038/ng.3021</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24974849">24974849</a></p>
</div>
<div id="ref-z91V6jjU">
<p>6. <strong>Resolving the complexity of the human genome using single-molecule sequencing</strong><br />
Mark J. P. Chaisson, John Huddleston, Megan Y. Dennis, Peter H. Sudmant, Maika Malig, Fereydoun Hormozdiari, Francesca Antonacci, Urvashi Surti, Richard Sandstrom, Matthew Boitano, … Evan E. Eichler<br />
<em>Nature</em> (2014-11-10) <a href="https://doi.org/w69">https://doi.org/w69</a><br />
DOI: <a href="https://doi.org/10.1038/nature13907">10.1038/nature13907</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25383537">25383537</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4317254">PMC4317254</a></p>
</div>
<div id="ref-rs7e40wC">
<p>7. <strong>Discovery and genotyping of structural variation from long-read haploid genome sequence data</strong><br />
John Huddleston, Mark J.P. Chaisson, Karyn Meltz Steinberg, Wes Warren, Kendra Hoekzema, David Gordon, Tina A. Graves-Lindsay, Katherine M. Munson, Zev N. Kronenberg, Laura Vives, … Evan E. Eichler<br />
<em>Genome Research</em> (2016-11-28) <a href="https://doi.org/f9x79h">https://doi.org/f9x79h</a><br />
DOI: <a href="https://doi.org/10.1101/gr.214007.116">10.1101/gr.214007.116</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27895111">27895111</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411763">PMC5411763</a></p>
</div>
<div id="ref-PRx3qEIm">
<p>8. <strong>Mapping and phasing of structural variation in patient genomes using nanopore sequencing</strong><br />
Mircea Cretu Stancu, Markus J. van Roosmalen, Ivo Renkens, Marleen M. Nieboer, Sjors Middelkamp, Joep de Ligt, Giulia Pregno, Daniela Giachino, Giorgia Mandrile, Jose Espejo Valle-Inclan, … Wigard P. Kloosterman<br />
<em>Nature Communications</em> (2017-11-06) <a href="https://doi.org/gftpt9">https://doi.org/gftpt9</a><br />
DOI: <a href="https://doi.org/10.1038/s41467-017-01343-4">10.1038/s41467-017-01343-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29109544">29109544</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5673902">PMC5673902</a></p>
</div>
<div id="ref-121OWcTA4">
<p>9. <strong>Genome-wide reconstruction of complex structural variants using read clouds</strong><br />
Noah Spies, Ziming Weng, Alex Bishara, Jennifer McDaniel, David Catoe, Justin M Zook, Marc Salit, Robert B West, Serafim Batzoglou, Arend Sidow<br />
<em>Nature Methods</em> (2017-07-17) <a href="https://doi.org/gbnhwk">https://doi.org/gbnhwk</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.4366">10.1038/nmeth.4366</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28714986">28714986</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5578891">PMC5578891</a></p>
</div>
<div id="ref-3NNFS6U2">
<p>10. <strong>Characterizing the Major Structural Variant Alleles of the Human Genome</strong><br />
Peter A. Audano, Arvis Sulovari, Tina A. Graves-Lindsay, Stuart Cantsilieris, Melanie Sorensen, AnneMarie E. Welch, Max L. Dougherty, Bradley J. Nelson, Ankeeta Shah, Susan K. Dutcher, … Evan E. Eichler<br />
<em>Cell</em> (2019-01) <a href="https://doi.org/gfthvz">https://doi.org/gfthvz</a><br />
DOI: <a href="https://doi.org/10.1016/j.cell.2018.12.019">10.1016/j.cell.2018.12.019</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30661756">30661756</a></p>
</div>
<div id="ref-6KbgcueR">
<p>11. <strong>Nanopore sequencing and assembly of a human genome with ultra-long reads</strong><br />
Miten Jain, Sergey Koren, Karen H Miga, Josh Quick, Arthur C Rand, Thomas A Sasani, John R Tyson, Andrew D Beggs, Alexander T Dilthey, Ian T Fiddes, … Matthew Loose<br />
<em>Nature Biotechnology</em> (2018-01-29) <a href="https://doi.org/gczffw">https://doi.org/gczffw</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.4060">10.1038/nbt.4060</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29431738">29431738</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5889714">PMC5889714</a></p>
</div>
<div id="ref-Pu6SY37C">
<p>12. <strong>Phased diploid genome assembly with single-molecule real-time sequencing</strong><br />
Chen-Shan Chin, Paul Peluso, Fritz J Sedlazeck, Maria Nattestad, Gregory T Concepcion, Alicia Clum, Christopher Dunn, Ronan O’Malley, Rosa Figueroa-Balderas, Abraham Morales-Cruz, … Michael C Schatz<br />
<em>Nature Methods</em> (2016-10-17) <a href="https://doi.org/f9fv4w">https://doi.org/f9fv4w</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.4035">10.1038/nmeth.4035</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27749838">27749838</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5503144">PMC5503144</a></p>
</div>
<div id="ref-Qa8mx6Ll">
<p>13. <strong>Genome graphs and the evolution of genome inference</strong><br />
Benedict Paten, Adam M. Novak, Jordan M. Eizenga, Erik Garrison<br />
<em>Genome Research</em> (2017-03-30) <a href="https://doi.org/f95nhd">https://doi.org/f95nhd</a><br />
DOI: <a href="https://doi.org/10.1101/gr.214155.116">10.1101/gr.214155.116</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28360232">28360232</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411762">PMC5411762</a></p>
</div>
<div id="ref-10jxt15v0">
<p>14. <strong>Variation graph toolkit improves read mapping by representing genetic variation in the reference</strong><br />
Erik Garrison, Jouni Sirén, Adam M Novak, Glenn Hickey, Jordan M Eizenga, Eric T Dawson, William Jones, Shilpa Garg, Charles Markello, Michael F Lin, … Richard Durbin<br />
<em>Nature Biotechnology</em> (2018-08-20) <a href="https://doi.org/gd2zqs">https://doi.org/gd2zqs</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.4227">10.1038/nbt.4227</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30125266">30125266</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6126949">PMC6126949</a></p>
</div>
<div id="ref-DuODeStx">
<p>15. <strong>Genome Graphs</strong><br />
Adam M Novak, Glenn Hickey, Erik Garrison, Sean Blum, Abram Connelly, Alexander Dilthey, Jordan Eizenga, M. A. Saleh Elmohamed, Sally Guthrie, André Kahles, … Benedict Paten<br />
<em>Cold Spring Harbor Laboratory</em> (2017-01-18) <a href="https://doi.org/gdcc74">https://doi.org/gdcc74</a><br />
DOI: <a href="https://doi.org/10.1101/101378">10.1101/101378</a></p>
</div>
<div id="ref-11Jy8B61m">
<p>16. <strong>Fast and accurate genomic analyses using genome graphs</strong><br />
Goran Rakocevic, Vladimir Semenyuk, Wan-Ping Lee, James Spencer, John Browning, Ivan J. Johnson, Vladan Arsenijevic, Jelena Nadj, Kaushik Ghose, Maria C. Suciu, … Deniz Kural<br />
<em>Nature Genetics</em> (2019-01-14) <a href="https://doi.org/gftd46">https://doi.org/gftd46</a><br />
DOI: <a href="https://doi.org/10.1038/s41588-018-0316-4">10.1038/s41588-018-0316-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30643257">30643257</a></p>
</div>
<div id="ref-ohTIiqfV">
<p>17. <strong>Graphtyper enables population-scale genotyping using pangenome graphs</strong><br />
Hannes P Eggertsson, Hakon Jonsson, Snaedis Kristmundsdottir, Eirikur Hjartarson, Birte Kehr, Gisli Masson, Florian Zink, Kristjan E Hjorleifsson, Aslaug Jonasdottir, Adalbjorg Jonasdottir, … Bjarni V Halldorsson<br />
<em>Nature Genetics</em> (2017-09-25) <a href="https://doi.org/gbx7v6">https://doi.org/gbx7v6</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3964">10.1038/ng.3964</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28945251">28945251</a></p>
</div>
<div id="ref-14Uxmwbxm">
<p>18. <strong>Accurate genotyping across variant classes and lengths using variant graphs</strong><br />
Jonas Andreas SibbesenLasse Maretty, Anders Krogh<br />
<em>Nature Genetics</em> (2018-06-18) <a href="https://doi.org/gdndnz">https://doi.org/gdndnz</a><br />
DOI: <a href="https://doi.org/10.1038/s41588-018-0145-5">10.1038/s41588-018-0145-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29915429">29915429</a></p>
</div>
<div id="ref-AltPnocw">
<p>19. <strong>SpeedSeq: ultra-fast personal genome analysis and interpretation</strong><br />
Colby Chiang, Ryan M Layer, Gregory G Faust, Michael R Lindberg, David B Rose, Erik P Garrison, Gabor T Marth, Aaron R Quinlan, Ira M Hall<br />
<em>Nature Methods</em> (2015-08-10) <a href="https://doi.org/gcpgfh">https://doi.org/gcpgfh</a><br />
DOI: <a href="https://doi.org/10.1038/nmeth.3505">10.1038/nmeth.3505</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26258291">26258291</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4589466">PMC4589466</a></p>
</div>
<div id="ref-nLvQCjXU">
<p>20. <strong>DELLY: structural variant discovery by integrated paired-end and split-read analysis</strong><br />
T. Rausch, T. Zichner, A. Schlattl, A. M. Stutz, V. Benes, J. O. Korbel<br />
<em>Bioinformatics</em> (2012-09-07) <a href="https://doi.org/f38r2c">https://doi.org/f38r2c</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/bts378">10.1093/bioinformatics/bts378</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22962449">22962449</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3436805">PMC3436805</a></p>
</div>
<div id="ref-14neTdqfN">
<p>21. <strong>Extensive sequencing of seven human genomes to characterize benchmark reference materials</strong><br />
Justin M. Zook, David Catoe, Jennifer McDaniel, Lindsay Vang, Noah Spies, Arend Sidow, Ziming Weng, Yuling Liu, Christopher E. Mason, Noah Alexander, … Marc Salit<br />
<em>Scientific Data</em> (2016-06-07) <a href="https://doi.org/f84nqc">https://doi.org/f84nqc</a><br />
DOI: <a href="https://doi.org/10.1038/sdata.2016.25">10.1038/sdata.2016.25</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27271295">27271295</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4896128">PMC4896128</a></p>
</div>
<div id="ref-16GvGhO20">
<p>22. <strong>Reproducible integration of multiple sequencing datasets to form high-confidence SNP, indel, and reference calls for five human genome reference materials</strong><br />
Justin Zook, Jennifer McDaniel, Hemang Parikh, Haynes Heaton, Sean A Irvine, Len Trigg, Rebecca Truty, Cory Y McLean, Francisco M De La Vega, Chunlin Xiao, … <br />
<em>Cold Spring Harbor Laboratory</em> (2018-03-13) <a href="https://doi.org/gfwsmj">https://doi.org/gfwsmj</a><br />
DOI: <a href="https://doi.org/10.1101/281006">10.1101/281006</a></p>
</div>
<div id="ref-7f5OKa5O">
<p>23. <strong>Contrasting evolutionary genome dynamics between domesticated and wild yeasts</strong><br />
Jia-Xing Yue, Jing Li, Louise Aigrain, Johan Hallin, Karl Persson, Karen Oliver, Anders Bergström, Paul Coupland, Jonas Warringer, Marco Cosentino Lagomarsino, … Gianni Liti<br />
<em>Nature Genetics</em> (2017-04-17) <a href="https://doi.org/f93kpp">https://doi.org/f93kpp</a><br />
DOI: <a href="https://doi.org/10.1038/ng.3847">10.1038/ng.3847</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28416820">28416820</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5446901">PMC5446901</a></p>
</div>
<div id="ref-krO7WgVi">
<p>24. <strong>Assemblytics: a web analytics tool for the detection of variants from an assembly</strong><br />
Maria Nattestad, Michael C. Schatz<br />
<em>Bioinformatics</em> (2016-06-17) <a href="https://doi.org/f9c485">https://doi.org/f9c485</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/btw369">10.1093/bioinformatics/btw369</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27318204">27318204</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6191160">PMC6191160</a></p>
</div>
<div id="ref-oVaXIwl5">
<p>25. <strong>Discovery, genotyping and characterization of structural variation and novel sequence at single nucleotide resolution from de novo genome assemblies on a population scale</strong><br />
Siyang LiuShujia Huang, Junhua Rao, Weijian Ye, Anders Krogh, Jun Wang<br />
<em>GigaScience</em> (2015-12) <a href="https://doi.org/f75r4n">https://doi.org/f75r4n</a><br />
DOI: <a href="https://doi.org/10.1186/s13742-015-0103-4">10.1186/s13742-015-0103-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26705468">26705468</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4690232">PMC4690232</a></p>
</div>
<div id="ref-172cJaw4Q">
<p>26. <strong>Minimap2: pairwise alignment for nucleotide sequences</strong><br />
Heng Li<br />
<em>Bioinformatics</em> (2018-05-10) <a href="https://doi.org/gdhbqt">https://doi.org/gdhbqt</a><br />
DOI: <a href="https://doi.org/10.1093/bioinformatics/bty191">10.1093/bioinformatics/bty191</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29750242">29750242</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6137996">PMC6137996</a></p>
</div>
<div id="ref-1FgS53pXi">
<p>27. <strong>Cactus: Algorithms for genome multiple sequence alignment</strong><br />
B. Paten, D. Earl, N. Nguyen, M. Diekhans, D. Zerbino, D. Haussler<br />
<em>Genome Research</em> (2011-06-10) <a href="https://doi.org/bk4697">https://doi.org/bk4697</a><br />
DOI: <a href="https://doi.org/10.1101/gr.123356.111">10.1101/gr.123356.111</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/21665927">21665927</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3166836">PMC3166836</a></p>
</div>
<div id="ref-faeC2cx0">
<p>28. <strong>Toil enables reproducible, open source, big biomedical data analyses</strong><br />
John Vivian, Arjun Arkal Rao, Frank Austin Nothaft, Christopher Ketchum, Joel Armstrong, Adam Novak, Jacob Pfeil, Jake Narkizian, Alden D Deran, Audrey Musselman-Brown, … Benedict Paten<br />
<em>Nature Biotechnology</em> (2017-04) <a href="https://doi.org/gfxbhs">https://doi.org/gfxbhs</a><br />
DOI: <a href="https://doi.org/10.1038/nbt.3772">10.1038/nbt.3772</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28398314">28398314</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5546205">PMC5546205</a></p>
</div>
<div id="ref-xJlNnKH2">
<p>29. <strong>Superbubbles, Ultrabubbles, and Cacti</strong><br />
Benedict Paten, Jordan M. Eizenga, Yohei M. Rosen, Adam M. Novak, Erik Garrison, Glenn Hickey<br />
<em>Journal of Computational Biology</em> (2018-07) <a href="https://doi.org/gdw582">https://doi.org/gdw582</a><br />
DOI: <a href="https://doi.org/10.1089/cmb.2017.0251">10.1089/cmb.2017.0251</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29461862">29461862</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6067107">PMC6067107</a></p>
</div>
<div id="ref-mH9pzoIn">
<p>30. <strong>Mash: fast genome and metagenome distance estimation using MinHash</strong><br />
Brian D. Ondov, Todd J. Treangen, Páll Melsted, Adam B. Mallonee, Nicholas H. Bergman, Sergey Koren, Adam M. Phillippy<br />
<em>Genome Biology</em> (2016-06-20) <a href="https://doi.org/gfx74q">https://doi.org/gfx74q</a><br />
DOI: <a href="https://doi.org/10.1186/s13059-016-0997-x">10.1186/s13059-016-0997-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27323842">27323842</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4915045">PMC4915045</a></p>
</div>
<div id="ref-10Jid8Wql">
<p>31. <strong>The Pancancer Analysis of Whole Genomes (PCAWG).</strong><a href="https://dcc.icgc.org/pcawg">https://dcc.icgc.org/pcawg</a></p>
</div>
<div id="ref-mWj2p7Xp">
<p>32. <strong>Genomics England 100,000 Genomes Project.</strong><a href="https://www.genomicsengland.co.uk">https://www.genomicsengland.co.uk</a></p>
</div>
<div id="ref-ir1O1h8n">
<p>33. <strong>Whole Genome Sequencing in the NHLBI Trans-Omics for Precision Medicine (TOPMed).</strong><a href="https://www.nhlbiwgs.org/">https://www.nhlbiwgs.org/</a></p>
</div>
<div id="ref-16glGyppX">
<p>34. <strong>Evaluation of computational genotyping of Structural Variations for clinical diagnoses.</strong><br />
Varuna Chander, Richard A Gibbs, Fritz J Sedlazeck<br />
<em>Cold Spring Harbor Laboratory</em> (2019-02-22) <a href="https://doi.org/gfwf66">https://doi.org/gfwf66</a><br />
DOI: <a href="https://doi.org/10.1101/558247">10.1101/558247</a></p>
</div>
<div id="ref-zRHJDqQ9">
<p>35. <strong>Nebula: Ultra-efficient mapping-free structural variant genotyper</strong><br />
Parsoa Khorsand, Fereydoun Hormozdiari<br />
<em>Cold Spring Harbor Laboratory</em> (2019-03-04) <a href="https://doi.org/gfwf67">https://doi.org/gfwf67</a><br />
DOI: <a href="https://doi.org/10.1101/566620">10.1101/566620</a></p>
</div>
<div id="ref-1BciDpJfo">
<p>36. <strong>Defining the diverse spectrum of inversions, complex structural variation, and chromothripsis in the morbid human genome</strong><br />
Ryan L. Collins, Harrison Brand, Claire E. Redin, Carrie Hanscom, Caroline Antolik, Matthew R. Stone, Joseph T. Glessner, Tamara Mason, Giulia Pregno, Naghmeh Dorrani, … Michael E. Talkowski<br />
<em>Genome Biology</em> (2017-03-06) <a href="https://doi.org/f9zjh2">https://doi.org/f9zjh2</a><br />
DOI: <a href="https://doi.org/10.1186/s13059-017-1158-6">10.1186/s13059-017-1158-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28260531">28260531</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5338099">PMC5338099</a></p>
</div>
</div>
<script>
// AnchorJS minified version below.
// Source https://github.com/bryanbraun/anchorjs/blob/064abdd0987f305933ec4982af6d0c1cf2fd0814/anchor.js

/**
 * AnchorJS - v4.0.0 - 2017-06-02
 * https://github.com/bryanbraun/anchorjs
 * Copyright (c) 2017 Bryan Braun; Licensed MIT
 */
!function(A,e){"use strict";"function"==typeof define&&define.amd?define([],e):"object"==typeof module&&module.exports?module.exports=e():(A.AnchorJS=e(),A.anchors=new A.AnchorJS)}(this,function(){"use strict";function A(A){function e(A){A.icon=A.hasOwnProperty("icon")?A.icon:"",A.visible=A.hasOwnProperty("visible")?A.visible:"hover",A.placement=A.hasOwnProperty("placement")?A.placement:"right",A.class=A.hasOwnProperty("class")?A.class:"",A.truncate=A.hasOwnProperty("truncate")?Math.floor(A.truncate):64}function t(A){var e;if("string"==typeof A||A instanceof String)e=[].slice.call(document.querySelectorAll(A));else{if(!(Array.isArray(A)||A instanceof NodeList))throw new Error("The selector provided to AnchorJS was invalid.");e=[].slice.call(A)}return e}function n(){if(null===document.head.querySelector("style.anchorjs")){var A,e=document.createElement("style");e.className="anchorjs",e.appendChild(document.createTextNode("")),void 0===(A=document.head.querySelector('[rel="stylesheet"], style'))?document.head.appendChild(e):document.head.insertBefore(e,A),e.sheet.insertRule(" .anchorjs-link {   opacity: 0;   text-decoration: none;   -webkit-font-smoothing: antialiased;   -moz-osx-font-smoothing: grayscale; }",e.sheet.cssRules.length),e.sheet.insertRule(" *:hover > .anchorjs-link, .anchorjs-link:focus  {   opacity: 1; }",e.sheet.cssRules.length),e.sheet.insertRule(" [data-anchorjs-icon]::after {   content: attr(data-anchorjs-icon); }",e.sheet.cssRules.length),e.sheet.insertRule(' @font-face {   font-family: "anchorjs-icons";   src: url(data:n/a;base64,AAEAAAALAIAAAwAwT1MvMg8yG2cAAAE4AAAAYGNtYXDp3gC3AAABpAAAAExnYXNwAAAAEAAAA9wAAAAIZ2x5ZlQCcfwAAAH4AAABCGhlYWQHFvHyAAAAvAAAADZoaGVhBnACFwAAAPQAAAAkaG10eASAADEAAAGYAAAADGxvY2EACACEAAAB8AAAAAhtYXhwAAYAVwAAARgAAAAgbmFtZQGOH9cAAAMAAAAAunBvc3QAAwAAAAADvAAAACAAAQAAAAEAAHzE2p9fDzz1AAkEAAAAAADRecUWAAAAANQA6R8AAAAAAoACwAAAAAgAAgAAAAAAAAABAAADwP/AAAACgAAA/9MCrQABAAAAAAAAAAAAAAAAAAAAAwABAAAAAwBVAAIAAAAAAAIAAAAAAAAAAAAAAAAAAAAAAAMCQAGQAAUAAAKZAswAAACPApkCzAAAAesAMwEJAAAAAAAAAAAAAAAAAAAAARAAAAAAAAAAAAAAAAAAAAAAQAAg//0DwP/AAEADwABAAAAAAQAAAAAAAAAAAAAAIAAAAAAAAAIAAAACgAAxAAAAAwAAAAMAAAAcAAEAAwAAABwAAwABAAAAHAAEADAAAAAIAAgAAgAAACDpy//9//8AAAAg6cv//f///+EWNwADAAEAAAAAAAAAAAAAAAAACACEAAEAAAAAAAAAAAAAAAAxAAACAAQARAKAAsAAKwBUAAABIiYnJjQ3NzY2MzIWFxYUBwcGIicmNDc3NjQnJiYjIgYHBwYUFxYUBwYGIwciJicmNDc3NjIXFhQHBwYUFxYWMzI2Nzc2NCcmNDc2MhcWFAcHBgYjARQGDAUtLXoWOR8fORYtLTgKGwoKCjgaGg0gEhIgDXoaGgkJBQwHdR85Fi0tOAobCgoKOBoaDSASEiANehoaCQkKGwotLXoWOR8BMwUFLYEuehYXFxYugC44CQkKGwo4GkoaDQ0NDXoaShoKGwoFBe8XFi6ALjgJCQobCjgaShoNDQ0NehpKGgobCgoKLYEuehYXAAAADACWAAEAAAAAAAEACAAAAAEAAAAAAAIAAwAIAAEAAAAAAAMACAAAAAEAAAAAAAQACAAAAAEAAAAAAAUAAQALAAEAAAAAAAYACAAAAAMAAQQJAAEAEAAMAAMAAQQJAAIABgAcAAMAAQQJAAMAEAAMAAMAAQQJAAQAEAAMAAMAAQQJAAUAAgAiAAMAAQQJAAYAEAAMYW5jaG9yanM0MDBAAGEAbgBjAGgAbwByAGoAcwA0ADAAMABAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAH//wAP) format("truetype"); }',e.sheet.cssRules.length)}}this.options=A||{},this.elements=[],e(this.options),this.isTouchDevice=function(){return!!("ontouchstart"in window||window.DocumentTouch&&document instanceof DocumentTouch)},this.add=function(A){var i,o,s,c,r,a,h,l,u,d,f,g,p=[];if(e(this.options),"touch"===(g=this.options.visible)&&(g=this.isTouchDevice()?"always":"hover"),A||(A="h2, h3, h4, h5, h6"),0===(i=t(A)).length)return this;for(n(),o=document.querySelectorAll("[id]"),s=[].map.call(o,function(A){return A.id}),r=0;r<i.length;r++)if(this.hasAnchorJSLink(i[r]))p.push(r);else{if(i[r].hasAttribute("id"))c=i[r].getAttribute("id");else if(i[r].hasAttribute("data-anchor-id"))c=i[r].getAttribute("data-anchor-id");else{u=l=this.urlify(i[r].textContent),h=0;do{void 0!==a&&(u=l+"-"+h),a=s.indexOf(u),h+=1}while(-1!==a);a=void 0,s.push(u),i[r].setAttribute("id",u),c=u}d=c.replace(/-/g," "),(f=document.createElement("a")).className="anchorjs-link "+this.options.class,f.href="#"+c,f.setAttribute("aria-label","Anchor link for: "+d),f.setAttribute("data-anchorjs-icon",this.options.icon),"always"===g&&(f.style.opacity="1"),""===this.options.icon&&(f.style.font="1em/1 anchorjs-icons","left"===this.options.placement&&(f.style.lineHeight="inherit")),"left"===this.options.placement?(f.style.position="absolute",f.style.marginLeft="-1em",f.style.paddingRight="0.5em",i[r].insertBefore(f,i[r].firstChild)):(f.style.paddingLeft="0.375em",i[r].appendChild(f))}for(r=0;r<p.length;r++)i.splice(p[r]-r,1);return this.elements=this.elements.concat(i),this},this.remove=function(A){for(var e,n,i=t(A),o=0;o<i.length;o++)(n=i[o].querySelector(".anchorjs-link"))&&(-1!==(e=this.elements.indexOf(i[o]))&&this.elements.splice(e,1),i[o].removeChild(n));return this},this.removeAll=function(){this.remove(this.elements)},this.urlify=function(A){var t=/[& +$,:;=?@"#{}|^~[`%!'<>\]\.\/\(\)\*\\]/g;return this.options.truncate||e(this.options),A.trim().replace(/\'/gi,"").replace(t,"-").replace(/-{2,}/g,"-").substring(0,this.options.truncate).replace(/^-+|-+$/gm,"").toLowerCase()},this.hasAnchorJSLink=function(A){var e=A.firstChild&&(" "+A.firstChild.className+" ").indexOf(" anchorjs-link ")>-1,t=A.lastChild&&(" "+A.lastChild.className+" ").indexOf(" anchorjs-link ")>-1;return e||t||!1}}return A});

// Enable links for selected headers
var anchors = new AnchorJS();
anchors.add("h2, h3, h4")
</script>
<!-- Enable Hypothesis annotation. https://web.hypothes.is/ -->
<script src="https://hypothes.is/embed.js" async></script>
</body>
</html>
